%--------------------------------------------------------%
%DIF LATEXDIFF DIFFERENCE FILE
%DIF DEL MiCoNE-pipeline-paper-old/main.tex   Fri Sep  2 10:21:30 2022
%DIF ADD MiCoNE-pipeline-paper/main.tex       Wed Sep 28 18:31:30 2022
% Journal Article Manuscript Template
%--------------------------------------------------------%

%!TEX root = ../main.tex

%--------------------------------------------------------%
% DOCUMENT CLASS
%--------------------------------------------------------%

  % Change "letterpaper" to "a4" if you use a4 paper size
  \documentclass[letterpaper,12pt]{article}

%--------------------------------------------------------%
% TITLE SECTION
%--------------------------------------------------------%

  %Abstract
  \usepackage{abstract} % Allows abstract customization
  % Set the "Abstract" text to bold
  \renewcommand{\abstractnamefont}{\normalfont\bfseries}
  % Set the abstract itself to small italic text
  \renewcommand{\abstracttextfont}{\normalfont\small\itshape}

  %Title
  \usepackage{titlesec} % Allows customization of titles

  %Authors
  \usepackage{authblk} % For multiple authors

  %Date
  \usepackage{datetime} % allows for including today's date
  % These two lines creates a new date format ``Month day(th), year''
  \newdateformat{usvardate}{
  \monthname[\THEMONTH] \ordinal{DAY}, \THEYEAR}

%--------------------------------------------------------%
% HEADERS & FOOTERS
%--------------------------------------------------------%

  %Footnotes
%DIF 42c42
%DIF <   \usepackage[bottom]{footmisc} % Makes footnotes stick to bottom of the page
%DIF -------
  % \usepackage[bottom]{footmisc} % Makes footnotes stick to bottom of the page %DIF > 
%DIF -------

  %Headers from page 2 on
%DIF 45-47c45-47
%DIF <   \usepackage{fancyhdr}
%DIF <   \pagestyle{fancy}
%DIF <   \fancyheadoffset{0cm}
%DIF -------
  % \usepackage{fancyhdr} %DIF > 
  \pagestyle{plain} %DIF > 
  % \fancyheadoffset{0cm} %DIF > 
%DIF -------
%   \setlength{\headheight}{15pt}

%--------------------------------------------------------%
% MACROS
%--------------------------------------------------------%

  % Define keywords macro command
  \providecommand{\keywords}[1]{\textbf{\textit{Keywords---}} #1}

%--------------------------------------------------------%
% MATH SUPPORT
%--------------------------------------------------------%

  % The amssymb package provides various useful mathematical symbols
  \usepackage{amssymb}
  % The amsthm package provides extended theorem environments
  \usepackage{amsthm}
  % The newtxmath package provides additional math symbol support
  % in Times New Roman symbols, etc.
  \usepackage{newtxmath}
%DIF 68a68-69
  \usepackage{mathtools} %DIF > 
  \usepackage{blkarray, bigstrut} %DIF > 
%DIF -------

%--------------------------------------------------------%
% FONTS
%--------------------------------------------------------%

  \usepackage{microtype} % Slightly tweak font spacing for aesthetics
  \usepackage[utf8]{inputenc}
  \usepackage{newtxtext} % Makes default font Adobe Times New Roman

%--------------------------------------------------------%
% LINES
%--------------------------------------------------------%

  % Spacing
  \usepackage{setspace} % See \doublespacing command at the top of content.tex
  % Numbering
%DIF 84c86-87
%DIF <   \usepackage{lineno,xcolor} 	% See \linenumbers at the top of content.tex
%DIF -------
  \usepackage{lineno} 	% See \linenumbers at the top of content.tex %DIF > 
  \usepackage[table,x11names]{xcolor} %DIF > 
%DIF -------
  % Lists
  \usepackage{enumitem}
  \setlist{nosep}
  \setlist[itemize]{leftmargin=*}

%--------------------------------------------------------%
% MARGINS
%--------------------------------------------------------%

  %NOTE: All spaces in this template are in inches, because it is
  % formatted for letterpaper (8.5 x 11 inch) paper. If you use a4
  % paper, choose different sizes in millimeters or centimeters.
  \usepackage[top=1.5in, bottom=1.5in, left=1in, right=1in]{geometry}

%--------------------------------------------------------%
% COMMENTS
%--------------------------------------------------------%

%DIF 103c106
%DIF <   \usepackage[colorinlistoftodos]{todonotes} % allows margin comments
%DIF -------
  % \usepackage[colorinlistoftodos]{todonotes} % allows margin comments %DIF > 
%DIF -------
  % See examples in content.tex, and here for manual:
  % http://www.ctan.org/pkg/todonotes
  \usepackage{soul} % allows for highlighting


%--------------------------------------------------------%
% ACRONYMS
%--------------------------------------------------------%

  \usepackage[nohyperlinks,nolist]{acronym} % Managing acronyms

%--------------------------------------------------------%
% GRAPHICS
%--------------------------------------------------------%

  \usepackage{graphicx,caption} % More advanced figure inclusion
  \graphicspath{{figures/}} % Set the default folder for images
  \usepackage{float} % For specifying table/figure locations, i.e. [ht!]

  % The printlen command allows the user to print the exact text width or height.
  % This is useful, when trying to create graphics (outside of LaTeX, of course)
  % with the optimal dimensions. See here for usage: http://www.ctan.org/pkg/printlen
  \usepackage{printlen}

  \usepackage[section]{placeins} % Used to ensure that figures do not go into the next section

%--------------------------------------------------------%
% TABLES
%--------------------------------------------------------%

  \usepackage{longtable} % For long tables that span multiple pages
  \newcommand{\sym}[1]{\rlap{#1}}% For symbols like *** in tables
  \usepackage{tabularx} % Allows advanced table features
  \newcolumntype{L}[1]{>{\raggedright\arraybackslash}p{#1}}
  \newcolumntype{C}[1]{>{\centering\arraybackslash}p{#1}}
  \newcolumntype{R}[1]{>{\raggedleft\arraybackslash}p{#1}}
  \usepackage{relsize} % Allows precise adjustment of font size,
  %useful for fitting tables to page width
  \usepackage{multirow}
%DIF 143a146-147
  %for horizontal tables %DIF > 
  \usepackage{lscape} %DIF > 
%DIF -------

%--------------------------------------------------------%
% REFERENCES
%--------------------------------------------------------%

  \usepackage{hyperref} % For hyperlinks in the PDF
  \usepackage{csquotes}
%DIF 150c155
%DIF <   \usepackage[style=numeric,backend=biber,sorting=none]{biblatex}
%DIF -------
  \usepackage[style=nature,url=false,backend=biber,sorting=none]{biblatex} %DIF > 
%DIF -------
  \bibliography{references/references.bib}
 % Edit preamble.tex to change the overall layout

% Header from Page Three on: Edit below for left and right headers
%DIF 155-156c160-161
%DIF < \lhead{}
%DIF < \rhead{}
%DIF -------
% \lhead{} %DIF > 
% \rhead{} %DIF > 
%DIF -------

%--------------------------------------------------------%
% BEGIN DOCUMENT
%--------------------------------------------------------%
%DIF PREAMBLE EXTENSION ADDED BY LATEXDIFF
%DIF UNDERLINE PREAMBLE %DIF PREAMBLE
\RequirePackage[normalem]{ulem} %DIF PREAMBLE
\RequirePackage{color}\definecolor{RED}{rgb}{1,0,0}\definecolor{BLUE}{rgb}{0,0,1} %DIF PREAMBLE
\providecommand{\DIFaddtex}[1]{{\protect\color{blue}\uwave{#1}}} %DIF PREAMBLE
\providecommand{\DIFdeltex}[1]{{\protect\color{red}\sout{#1}}}                      %DIF PREAMBLE
%DIF SAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddbegin}{} %DIF PREAMBLE
\providecommand{\DIFaddend}{} %DIF PREAMBLE
\providecommand{\DIFdelbegin}{} %DIF PREAMBLE
\providecommand{\DIFdelend}{} %DIF PREAMBLE
\providecommand{\DIFmodbegin}{} %DIF PREAMBLE
\providecommand{\DIFmodend}{} %DIF PREAMBLE
%DIF FLOATSAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddFL}[1]{\DIFadd{#1}} %DIF PREAMBLE
\providecommand{\DIFdelFL}[1]{\DIFdel{#1}} %DIF PREAMBLE
\providecommand{\DIFaddbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFaddendFL}{} %DIF PREAMBLE
\providecommand{\DIFdelbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFdelendFL}{} %DIF PREAMBLE
%DIF HYPERREF PREAMBLE %DIF PREAMBLE
\providecommand{\DIFadd}[1]{\texorpdfstring{\DIFaddtex{#1}}{#1}} %DIF PREAMBLE
\providecommand{\DIFdel}[1]{\texorpdfstring{\DIFdeltex{#1}}{}} %DIF PREAMBLE
\newcommand{\DIFscaledelfig}{0.5}
%DIF HIGHLIGHTGRAPHICS PREAMBLE %DIF PREAMBLE
\RequirePackage{settobox} %DIF PREAMBLE
\RequirePackage{letltxmacro} %DIF PREAMBLE
\newsavebox{\DIFdelgraphicsbox} %DIF PREAMBLE
\newlength{\DIFdelgraphicswidth} %DIF PREAMBLE
\newlength{\DIFdelgraphicsheight} %DIF PREAMBLE
% store original definition of \includegraphics %DIF PREAMBLE
\LetLtxMacro{\DIFOincludegraphics}{\includegraphics} %DIF PREAMBLE
\newcommand{\DIFaddincludegraphics}[2][]{{\color{blue}\fbox{\DIFOincludegraphics[#1]{#2}}}} %DIF PREAMBLE
\newcommand{\DIFdelincludegraphics}[2][]{% %DIF PREAMBLE
\sbox{\DIFdelgraphicsbox}{\DIFOincludegraphics[#1]{#2}}% %DIF PREAMBLE
\settoboxwidth{\DIFdelgraphicswidth}{\DIFdelgraphicsbox} %DIF PREAMBLE
\settoboxtotalheight{\DIFdelgraphicsheight}{\DIFdelgraphicsbox} %DIF PREAMBLE
\scalebox{\DIFscaledelfig}{% %DIF PREAMBLE
\parbox[b]{\DIFdelgraphicswidth}{\usebox{\DIFdelgraphicsbox}\\[-\baselineskip] \rule{\DIFdelgraphicswidth}{0em}}\llap{\resizebox{\DIFdelgraphicswidth}{\DIFdelgraphicsheight}{% %DIF PREAMBLE
\setlength{\unitlength}{\DIFdelgraphicswidth}% %DIF PREAMBLE
\begin{picture}(1,1)% %DIF PREAMBLE
\thicklines\linethickness{2pt} %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\framebox(1,1){}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\line( 1,1){1}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,1){\line(1,-1){1}}}% %DIF PREAMBLE
\end{picture}% %DIF PREAMBLE
}\hspace*{3pt}}} %DIF PREAMBLE
} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbegin}{\DIFaddbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddend}{\DIFaddend} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbegin}{\DIFdelbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelend}{\DIFdelend} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbegin}{\DIFOaddbegin \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbegin}{\DIFOdelbegin \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbeginFL}{\DIFaddbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddendFL}{\DIFaddendFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbeginFL}{\DIFdelbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelendFL}{\DIFdelendFL} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbeginFL}{\DIFOaddbeginFL \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbeginFL}{\DIFOdelbeginFL \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
%DIF COLORLISTINGS PREAMBLE %DIF PREAMBLE
\RequirePackage{listings} %DIF PREAMBLE
\RequirePackage{color} %DIF PREAMBLE
\lstdefinelanguage{DIFcode}{ %DIF PREAMBLE
%DIF DIFCODE_UNDERLINE %DIF PREAMBLE
  moredelim=[il][\color{red}\sout]{\%DIF\ <\ }, %DIF PREAMBLE
  moredelim=[il][\color{blue}\uwave]{\%DIF\ >\ } %DIF PREAMBLE
} %DIF PREAMBLE
\lstdefinestyle{DIFverbatimstyle}{ %DIF PREAMBLE
	language=DIFcode, %DIF PREAMBLE
	basicstyle=\ttfamily, %DIF PREAMBLE
	columns=fullflexible, %DIF PREAMBLE
	keepspaces=true %DIF PREAMBLE
} %DIF PREAMBLE
\lstnewenvironment{DIFverbatim}{\lstset{style=DIFverbatimstyle}}{} %DIF PREAMBLE
\lstnewenvironment{DIFverbatim*}{\lstset{style=DIFverbatimstyle,showspaces=true}}{} %DIF PREAMBLE
%DIF END PREAMBLE EXTENSION ADDED BY LATEXDIFF

\begin{document}

% COVER PAGE

%!TEX root = ../main.tex

\begin{titlepage}

  \newcommand{\HRule}{\rule{\linewidth}{0.5mm}} % Defines a new command for the horizontal lines, change thickness here

  \center % Center everything on the page


% HEADING SECTION

  %\textsc{\LARGE University Name}\\[1.5cm] % Name of your university/college
  % \textsc{\Large Manuscript Submission}\\[0.5cm] % Major heading such as course name
  % \textsc{\large The Journal of Blah Blah}\\[0.5cm] % Minor heading such as course title

% TITLE SECTION

  \vspace*{\fill}
  {\huge Inferring microbial co-occurrence networks from amplicon data: a systematic evaluation}\\[0.4cm]
  % {\huge  2. Investigating the best practices for inference of microbial co-occurrence networks from 16S data}\\[0.4cm] % Title of your document
  % {\huge  3. Attempting to find the best practice pipeline for inferring co-occurrence networks from 16S data}\\[0.4cm] % Title of your document
  % {\huge  4. Deciphering the complexities in co-occurrence network inference from 16S data}\\[0.4cm] % Title of your document

% AUTHOR SECTION

  \vspace{1.5 cm}
  Dileep Kishore\textsuperscript{\DIFdelbegin \DIFdel{1}\DIFdelend \DIFaddbegin \DIFadd{a}\DIFaddend ,\DIFdelbegin \DIFdel{2}\DIFdelend \DIFaddbegin \DIFadd{b}\DIFaddend },
  Gabriel Birzu\textsuperscript{\DIFdelbegin \DIFdel{3}\DIFdelend \DIFaddbegin \DIFadd{c}\DIFaddend ,\DIFdelbegin \DIFdel{6}\DIFdelend \DIFaddbegin \DIFadd{f}\DIFaddend },
  Zhenjun Hu\textsuperscript{\DIFdelbegin \DIFdel{1}\DIFdelend \DIFaddbegin \DIFadd{a}\DIFaddend },
  Charles DeLisi\textsuperscript{\DIFdelbegin \DIFdel{1}\DIFdelend \DIFaddbegin \DIFadd{a}\DIFaddend ,\DIFdelbegin \DIFdel{3}\DIFdelend \DIFaddbegin \DIFadd{c}\DIFaddend },
  Kirill S. Korolev\textsuperscript{\DIFdelbegin \DIFdel{$\dagger$1}\DIFdelend \DIFaddbegin \DIFadd{a}\DIFaddend ,\DIFdelbegin \DIFdel{2}\DIFdelend \DIFaddbegin \DIFadd{b}\DIFaddend ,\DIFdelbegin \DIFdel{3}\DIFdelend \DIFaddbegin \DIFadd{c}\DIFaddend }\DIFaddbegin \DIFadd{\#}\DIFaddend ,\\
  Daniel Segr\`{e}\textsuperscript{\DIFdelbegin \DIFdel{$\dagger$1}\DIFdelend \DIFaddbegin \DIFadd{a}\DIFaddend ,\DIFdelbegin \DIFdel{2}\DIFdelend \DIFaddbegin \DIFadd{b}\DIFaddend ,\DIFdelbegin \DIFdel{4}\DIFdelend \DIFaddbegin \DIFadd{d}\DIFaddend ,\DIFdelbegin \DIFdel{5}\DIFdelend \DIFaddbegin \DIFadd{e}\DIFaddend }\DIFaddbegin \DIFadd{\#}\DIFaddend \\
  \vspace{1cm}
  \textsuperscript{\DIFdelbegin \DIFdel{1}\DIFdelend \DIFaddbegin \DIFadd{a}\DIFaddend }Bioinformatics Program, Boston University, Boston, Massachusetts, USA\\
  \textsuperscript{\DIFdelbegin \DIFdel{2}\DIFdelend \DIFaddbegin \DIFadd{b}\DIFaddend }Biological Design Center, Boston University, Boston, Massachusetts, USA\\
  \textsuperscript{\DIFdelbegin \DIFdel{3}\DIFdelend \DIFaddbegin \DIFadd{c}\DIFaddend }Department of Physics, Boston University, Boston, Massachusetts, USA\\
  \textsuperscript{\DIFdelbegin \DIFdel{4}\DIFdelend \DIFaddbegin \DIFadd{d}\DIFaddend }Department of Biomedical Engineering, Boston University, Boston, Massachusetts, USA\\
  \textsuperscript{\DIFdelbegin \DIFdel{5}\DIFdelend \DIFaddbegin \DIFadd{e}\DIFaddend }Department of Biology, Boston University, Boston, Massachusetts, USA\\
  \textsuperscript{\DIFdelbegin \DIFdel{6}\DIFdelend \DIFaddbegin \DIFadd{f}\DIFaddend }Department of Applied Physics, Stanford University, Stanford, California, USA\\
  \textsuperscript{\DIFdelbegin \DIFdel{$\dagger$}\DIFdelend \DIFaddbegin \DIFadd{$\#$}\DIFaddend }Correspondence should be sent to \href{mailto:korolev@bu.edu}{korolev@bu.edu} or \href{mailto:dsegre@bu.edu}{dsegre@bu.edu}\\

  \DIFaddbegin \vspace{2cm}

\DIFaddend % % DATE SECTION

  % \vspace{1.5 cm}
  % {\large Submitted: \today}\\[3cm] % Date, change the \today to a set date if you want to be precise


  \vspace*{\fill} % Fill the rest of the page with whitespace

\end{titlepage}

%-----------------------------------------------------------------

\newpage
 % Comment out to remove cover page

\thispagestyle{empty} %DIF <  Removes header on page two. Only needed if there is a cover-page
%DIF >  Removes header on page two.

% NOTE: Comment out the lines below to remove line numbers
  % Running line numbers:
  \linenumbers
  \setlength\linenumbersep{15pt}
  \renewcommand\linenumberfont{\normalfont\footnotesize\sffamily\color{gray}}
  %\pagewiselinenumbers % Same, but that reset on every page:
  \modulolinenumbers[1] % Number only every line. Change for fewer.

%--------------------------------------------------------%
%   CONTENT
%--------------------------------------------------------%

% ABSTRACT
%!TEX root = ../main.tex

\begin{abstract}
  {
    \noindent
    Microbes \DIFdelbegin \DIFdel{tend to }\DIFdelend \DIFaddbegin \DIFadd{commonly }\DIFaddend organize into communities consisting of hundreds of species involved in complex interactions with each other.
    16S ribosomal RNA (16S rRNA) amplicon profiling provides snapshots that reveal the phylogenies and abundance profiles of these microbial communities.
    These snapshots, when collected from multiple samples, \DIFdelbegin \DIFdel{have the potential to reveal which microbesco-occur}\DIFdelend \DIFaddbegin \DIFadd{can reveal the co-occurrence of microbes}\DIFaddend , providing a glimpse into the network of associations in these communities.
    \DIFdelbegin \DIFdel{The }\DIFdelend \DIFaddbegin \DIFadd{However, the }\DIFaddend inference of networks from 16S data \DIFdelbegin \DIFdel{is prone to statistical artifacts.
    There are many tools for performing each step of the 16S analysis workflow, but }\DIFdelend \DIFaddbegin \DIFadd{involves numerous steps, each requiring specific tools and parameter choices.
    Moreover, }\DIFaddend the extent to which these steps affect the final network is still unclear.
    In this study, we perform a meticulous analysis of each step of a pipeline that can convert 16S sequencing data into a network of microbial associations.
    Through this process, we map how different choices of algorithms and parameters affect the co-occurrence network and \DIFdelbegin \DIFdel{estimate }\DIFdelend \DIFaddbegin \DIFadd{identify the }\DIFaddend steps that contribute \DIFdelbegin \DIFdel{most significantly }\DIFdelend \DIFaddbegin \DIFadd{substantially }\DIFaddend to the variance.
    We further determine the tools and parameters that generate \DIFdelbegin \DIFdel{the most accurate and }\DIFdelend robust co-occurrence networks \DIFdelbegin \DIFdel{based on comparison }\DIFdelend \DIFaddbegin \DIFadd{and develop consensus network algorithms based on benchmarks }\DIFaddend with mock and synthetic datasets.
    \DIFdelbegin \DIFdel{Ultimately, we develop a standardized pipeline }\DIFdelend \DIFaddbegin \DIFadd{The Microbial Co-occurrence Network Explorer or }\acs{micone} \DIFaddend (available at \href{https://github.com/segrelab/MiCoNE}{https://github.com/segrelab/MiCoNE})\DIFdelbegin \DIFdel{that }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend follows these default tools and parameters \DIFdelbegin \DIFdel{, but that can also }\DIFdelend \DIFaddbegin \DIFadd{and can }\DIFaddend help explore the outcome of \DIFdelbegin \DIFdel{any other combination of choices }\DIFdelend \DIFaddbegin \DIFadd{these combinations of choices on the inferred networks}\DIFaddend .
    We envisage that this pipeline could be used for integrating multiple \DIFdelbegin \DIFdel{data-sets}\DIFdelend \DIFaddbegin \DIFadd{datasets}\DIFaddend , and for generating comparative analyses and consensus networks that can \DIFdelbegin \DIFdel{help understand and control }\DIFdelend \DIFaddbegin \DIFadd{guide our understanding of }\DIFaddend microbial community assembly in different biomes.
  }
\end{abstract}

% Insert keywords here
\DIFdelbegin \DIFdel{\keywords{Microbiome, 16S rRNA, Pipeline, Interaction, Denoising, Taxonomy, Network Inference, Correlations, Qiime, Co-occurrence, Networks}
}\DIFdelend \DIFaddbegin \DIFadd{\keywords{Microbiome, 16S rRNA, Interaction, Denoising, Taxonomy, Network Inference, Correlations, QIIME2, Co-occurrence, Networks, Consensus algorithm, Pipeline, nextflow}
}\DIFaddend 

\DIFdelbegin \section*{\DIFdel{Importance}}
  %DIFAUXCMD
\DIFdelend %DIF >  \doublespacing

\DIFdelbegin \DIFdel{To understand and control the mechanisms that determine the structure and function of microbial communities, it is important to map the interrelationships between its constituent microbial species }\DIFdelend \DIFaddbegin \section*{\DIFadd{Importance}}
  \DIFadd{Mapping the interrelationships between different species in a microbial community is important for understanding and controlling their structure and function}\DIFaddend .
  The surge in the high-throughput sequencing of microbial communities has led to the creation of thousands of datasets containing information about microbial abundances.
  These abundances can be transformed into \DIFdelbegin \DIFdel{networks of co-occurrences across multiple samples}\DIFdelend \DIFaddbegin \DIFadd{co-occurrence networks}\DIFaddend , providing a glimpse into the \DIFdelbegin \DIFdel{structure of }\DIFdelend \DIFaddbegin \DIFadd{associations within }\DIFaddend microbiomes.
  However, processing these datasets to obtain co-occurrence information relies on several complex steps, each of which involves \DIFdelbegin \DIFdel{multiple }\DIFdelend \DIFaddbegin \DIFadd{numerous }\DIFaddend choices of tools and corresponding parameters.
  These multiple options pose questions about the \DIFdelbegin \DIFdel{accuracy }\DIFdelend \DIFaddbegin \DIFadd{robustness }\DIFaddend and uniqueness of the inferred networks.
  In this study, we address this workflow and provide a systematic analysis of how these choices of tools \DIFdelbegin \DIFdel{and parameters }\DIFdelend affect the final network, and \DIFdelbegin \DIFdel{on how to select those that are most appropriate }\DIFdelend \DIFaddbegin \DIFadd{guidelines on appropriate tool selection }\DIFaddend for a particular dataset.
  \DIFaddbegin \DIFadd{We also develop a consensus network algorithm that helps generate more robust co-occurrence networks based on benchmark synthetic datasets.
}\DIFaddend 


\doublespacing

% INTRODUCTION
%!TEX root = ../main.tex

\section*{Introduction}

Microbial communities are ubiquitous and play an important role in marine and terrestrial environments, urban ecosystems, \DIFdelbegin \DIFdel{metabolic engineering, }\DIFdelend and human health \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Ghoul2016,Thompson2017}}\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{lima-mendezDeterminantsCommunityStructure2015a,Thompson2017,royo-llonchCompendium530Metagenomeassembled2021,tedersooFungalBiogeographyGlobal2014,dankoGlobalMetagenomicMap2021,mclellanMicrobiomeUrbanWaters2015,HumanMicrobiomeProjectConsortium2012}}\hskip0pt%DIFAUXCMD
}\DIFaddend .
  These microbial communities, or microbiomes, often comprise several hundreds of different microbial strains interacting with each other and their environment, often through \DIFdelbegin \DIFdel{intricate }\DIFdelend \DIFaddbegin \DIFadd{complex }\DIFaddend metabolic and signaling relationships\DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{zelezniakMetabolicDependenciesDrive2015,Ghoul2016,coyteUnderstandingCompetitionCooperation2019,DSouza2018}}\hskip0pt%DIFAUXCMD
}\DIFaddend .
  Understanding how these interconnections shape community structure and \DIFdelbegin \DIFdel{functionalities }\DIFdelend \DIFaddbegin \DIFadd{function }\DIFaddend is a fundamental challenge in microbial ecology, \DIFdelbegin \DIFdel{with }\DIFdelend \DIFaddbegin \DIFadd{and has }\DIFaddend applications in the study of microbial ecosystems across different biomes.
  With the advancement in DNA sequencing technologies\DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Narihiro2017} }\hskip0pt%DIFAUXCMD
and data processing methods}\DIFdelend \DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{huNextgenerationSequencingTechnologies2021,buermansNextGenerationSequencing2014,Narihiro2017}}\hskip0pt%DIFAUXCMD
}\DIFaddend ,  more information can be extracted from these microbial community samples than ever before.
  In particular, high-throughput sequencing, including \DIFdelbegin \DIFdel{community }\DIFdelend metagenomic sequencing and sequencing of 16S rRNA gene amplicons \DIFdelbegin \DIFdel{, has the potential to }\DIFdelend \DIFaddbegin \DIFadd{(hereafter referred to as 16S data) of microbial communities, can }\DIFaddend help detect, identify and quantify a large portion of the constitutive microorganisms of a microbiome \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Jovel2016,Lloyd-Price2016}}\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{ju16SRRNAGene2015,Jovel2016,quinceShotgunMetagenomicsSampling2017,sedlarBioinformaticsStrategiesTaxonomy2017}}\hskip0pt%DIFAUXCMD
}\DIFaddend .
  These advances have led to large-scale data collection efforts involving \DIFdelbegin \DIFdel{environmental (}%DIFDELCMD < \acl{emp}%%%
\DIFdel{) \mbox{%DIFAUXCMD
\cite{Thompson2017}}\hskip0pt%DIFAUXCMD
, marine(Tara Oceans Project) \mbox{%DIFAUXCMD
\cite{Zhang2015} }\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{terrestrial~\mbox{%DIFAUXCMD
\cite{Thompson2017,gilbertMeetingReportTerabase2010,tedersooFungalBiogeographyGlobal2014}}\hskip0pt%DIFAUXCMD
, marine~\mbox{%DIFAUXCMD
\cite{lima-mendezDeterminantsCommunityStructure2015a,royo-llonchCompendium530Metagenomeassembled2021} }\hskip0pt%DIFAUXCMD
}\DIFaddend and human-associated microbiota\DIFdelbegin \DIFdel{(Human Microbiome Project) \mbox{%DIFAUXCMD
\cite{HumanMicrobiomeProjectConsortium2012}}\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{HumanMicrobiomeProjectConsortium2012,proctorIntegrativeHumanMicrobiome2019,Lloyd-Price2016}}\hskip0pt%DIFAUXCMD
}\DIFaddend .

 This wealth of information \DIFdelbegin \DIFdel{on the composition and functions of a community at different times and under different environmental conditions }\DIFdelend has the potential to help us understand how communities assemble and operate.
 \DIFdelbegin \DIFdel{A }\DIFdelend \DIFaddbegin \DIFadd{In particular, a }\DIFaddend powerful tool for translating microbiome \DIFaddbegin \DIFadd{composition }\DIFaddend data into knowledge is the construction of \DIFdelbegin \DIFdel{possible inter-dependence networksacross species.
 }\DIFdelend \DIFaddbegin \DIFadd{association (co-occurrence or correlation) networks, in which  microbial taxa are represented by nodes, and frequent co-occurrences (or negative correlations) across datasets are encoded as edges between nodes.
 While the relationship between directly measured interactions~\mbox{%DIFAUXCMD
\cite{lubbeExometabolomicAnalysisCrossFeeding2017,Jian2020,Hsu2019} }\hskip0pt%DIFAUXCMD
and statistically inferred co-occurrence is still poorly understood \mbox{%DIFAUXCMD
\cite{Zuniga2017,Rottjers2018}}\hskip0pt%DIFAUXCMD
, a significant amount of effort has gone into estimating correlations from large microbiome sequence datasets~\mbox{%DIFAUXCMD
\cite{faustMicrobialCooccurrenceRelationships2012,leeCrosskingdomCooccurrenceNetworks2022,faustMicrobialInteractionsNetworks2012a,maEarthMicrobialCooccurrence2020a}}\hskip0pt%DIFAUXCMD
.
}

 \DIFaddend The importance of these networks \DIFdelbegin \DIFdel{of relationships is two fold}\DIFdelend \DIFaddbegin \DIFadd{is two-fold}\DIFaddend : first, \DIFdelbegin \DIFdel{such networks }\DIFdelend \DIFaddbegin \DIFadd{they }\DIFaddend can serve as maps that help identify hubs of keystone species \cite{Menon2018,Rottjers2018}, \DIFdelbegin \DIFdel{or basic microbiome changes that occur as a consequence of }\DIFdelend \DIFaddbegin \DIFadd{and the community response to }\DIFaddend environmental perturbations or underlying host conditions \cite{Gilbert2016}; second, \DIFdelbegin \DIFdel{networks of inter-dependencies }\DIFdelend \DIFaddbegin \DIFadd{they }\DIFaddend can serve as a \DIFdelbegin \DIFdel{key }\DIFdelend bridge towards building mechanistic models of microbial communities, greatly enhancing our capacity to understand and control them.
 For example, multiple studies have shown the importance of specific microbial \DIFdelbegin \DIFdel{interactions }\DIFdelend \DIFaddbegin \DIFadd{associations }\DIFaddend in the healthy microbiome \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Lloyd-Price2016} }\hskip0pt%DIFAUXCMD
and others have shown how changes in these interactions can lead to }\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{Lloyd-Price2016,Wu2016,HumanMicrobiomeProjectConsortium2012} }\hskip0pt%DIFAUXCMD
and their role in }\DIFaddend dysbiosis \cite{Wang2017,Gilbert2016,Belizario2015}.
 In the context of terrestrial \DIFdelbegin \DIFdel{bio-geochemistry}\DIFdelend \DIFaddbegin \DIFadd{biogeochemistry}\DIFaddend , co-occurrence networks \DIFdelbegin \DIFdel{have been proposed as a valuable approach towards reconstructing the processes leading to microbiome assembly \mbox{%DIFAUXCMD
\cite{Fierer2017}}\hskip0pt%DIFAUXCMD
, and  understanding }\DIFdelend \DIFaddbegin \DIFadd{were shown to help understand microbiome assembly \mbox{%DIFAUXCMD
\cite{fiererEmbracingUnknownDisentangling2017}}\hskip0pt%DIFAUXCMD
, and  }\DIFaddend the response of microbial communities to environmental perturbations \cite{Jiao2019}.

\DIFdelbegin \DIFdel{Direct high-throughput measurement of interactions, e.g. through co-culture micro-droplet experiments \mbox{%DIFAUXCMD
\cite{Hsu2019,Jian2020}}\hskip0pt%DIFAUXCMD
, or spatial visualization of natural communities \mbox{%DIFAUXCMD
\cite{Wilbert2020} }\hskip0pt%DIFAUXCMD
is possible, but it requires specific technological capabilities, and has yet to be extensively used.
 In parallel, sequencing data across multiple samples can be used for estimating co-occurrence relationships between taxa.
 While the the relationship between directly measured interactions and statistically inferred co-occurrence is still poorly understood \mbox{%DIFAUXCMD
\cite{Zuniga2017}}\hskip0pt%DIFAUXCMD
, a significant amount of effort has gone into estimating correlations from large microbiome sequence datasets.
 Co-occurrence networks have microbial taxa as nodes, and edges that represent the frequent co-occurrence (or negative correlations) across different datasets. 
}%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend One of the most frequently used avenues for inferring co-occurrence networks is the parsing and analysis of 16S sequencing data \cite{Rottjers2018,Friedman2012}.
\DIFdelbegin \DIFdel{A large number of }\DIFdelend \DIFaddbegin \DIFadd{Numerous }\DIFaddend software tools and pipelines have been developed to analyze 16S sequencing data, \DIFdelbegin \DIFdel{often focused on addressing the many }\DIFdelend \DIFaddbegin \DIFadd{with a strong emphasis on the }\DIFaddend known limitations of this \DIFdelbegin \DIFdel{methodology}\DIFdelend \DIFaddbegin \DIFadd{method}\DIFaddend , including resolution, sequencing depth, compositional nature, sequencing errors\DIFaddbegin \DIFadd{, }\DIFaddend and copy number variations \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Bharti2019,Pollock2018}}\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{Bharti2019,pollockMadnessMicrobiomeAttempting2018}}\hskip0pt%DIFAUXCMD
}\DIFaddend .
Popular methods for different phases of the analysis of 16S data include tools for: (i) \DIFdelbegin \DIFdel{denoising and clustering sequencing reads\mbox{%DIFAUXCMD
\cite{Caporaso2010,Callahan2016}}\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{quality checking and trimming the sequencing reads}\DIFaddend ; (ii) \DIFaddbegin \DIFadd{denoising and clustering the trimmed reads \mbox{%DIFAUXCMD
\cite{Caporaso2010,Callahan2016,Amir2017}}\hskip0pt%DIFAUXCMD
; (iii) }\DIFaddend assigning taxonomy to the \DIFdelbegin \DIFdel{reads \mbox{%DIFAUXCMD
\cite{DeSantis2006,Quast2012}}\hskip0pt%DIFAUXCMD
; (iii}\DIFdelend \DIFaddbegin \DIFadd{denoised reads \mbox{%DIFAUXCMD
\cite{bokulichOptimizingTaxonomicClassification2018}}\hskip0pt%DIFAUXCMD
; (iv}\DIFaddend ) processing and transforming the taxonomy count matrices \cite{Weiss2015}; and (\DIFdelbegin \DIFdel{iv}\DIFdelend \DIFaddbegin \DIFadd{v}\DIFaddend ) inferring the co-occurrence network \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Cougoul2019,Kurtz2015}}\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{Watts2018,Kurtz2015,tackmannRapidInferenceDirect2019}}\hskip0pt%DIFAUXCMD
}\DIFaddend .
Different specific algorithms are often aggregated into popular \DIFaddbegin \DIFadd{online }\DIFaddend platforms (like MG-RAST\DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Keegan2016}}\hskip0pt%DIFAUXCMD
, Qiita\mbox{%DIFAUXCMD
\cite{qiita}}\hskip0pt%DIFAUXCMD
) and }\DIFdelend \DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{keeganMGRASTMetagenomicsService2016}}\hskip0pt%DIFAUXCMD
, Qiita~\mbox{%DIFAUXCMD
\cite{gonzalezQiitaRapidWebenabled2018}}\hskip0pt%DIFAUXCMD
) and software }\DIFaddend packages (such as \DIFdelbegin \DIFdel{QIIME \mbox{%DIFAUXCMD
\cite{Caporaso2010}}\hskip0pt%DIFAUXCMD
)that provide pipelines for 16S data analysis}\DIFdelend \DIFaddbegin \ac{qiime2}\DIFadd{~\mbox{%DIFAUXCMD
\cite{bolyenReproducibleInteractiveScalable2019}}\hskip0pt%DIFAUXCMD
)}\DIFaddend .
The different methods and tools \DIFdelbegin \DIFdel{developed to solve issues arising in 16S analysis }\DIFdelend can lead to vastly different inferences of community compositions and co-occurrence networks \cite{Golob2017,Weiss2016}, making it difficult to reliably compare networks across different publications and studies.
 This \DIFaddbegin \DIFadd{difference }\DIFaddend is partially due to the \DIFdelbegin \DIFdel{fact that existing platforms are typically focused }\DIFdelend \DIFaddbegin \DIFadd{focus of existing platforms }\DIFaddend on \ac{otu} \DIFaddbegin \DIFadd{or }\ac{esv} \DIFaddend generation and not on the effects of upstream statistical methods on the inferred co-occurrence networks.
 Furthermore, no organized framework currently \DIFdelbegin \DIFdel{exist to }\DIFdelend \DIFaddbegin \DIFadd{exists that can }\DIFaddend systematically analyze and compare \DIFdelbegin \DIFdel{existing components of the data analysis from amplicons to networks.
  More broadly, given the lack of comprehensive comparisons between directly observed microbial interactions (e.g. from co-culture experiments) and }\DIFdelend \DIFaddbegin \DIFadd{each step in the pipeline for processing amplicons into }\DIFaddend co-occurrence networks\DIFdelbegin \DIFdel{, there is no straightforward way to determine which set of tools or methods generate the most accurate networks}\DIFdelend .

 In this study, we present a standardized 16S data analysis pipeline called \ac{micone} that produces robust and reproducible co-occurrence networks from \DIFdelbegin \DIFdel{community }\DIFdelend 16S sequence data \DIFdelbegin \DIFdel{, and allow }\DIFdelend \DIFaddbegin \DIFadd{of microbial communities, and enable }\DIFaddend users to interactively explore how the network would change upon using different alternative tools and parameters at each step.
 %DIF <  TODO: Link or describe MIND here (?)
  Our pipeline is coupled to an online integrative tool for the organization, visualization\DIFaddbegin \DIFadd{, }\DIFaddend and analysis of inter-microbial networks \DIFdelbegin \DIFdel{.
  In addition to making this tool freely available, we implemented }\DIFdelend \DIFaddbegin \DIFadd{called }\ac{mind}\DIFadd{~\mbox{%DIFAUXCMD
\cite{huResourceComparisonIntegration2022}}\hskip0pt%DIFAUXCMD
, which is available at }\href{http://microbialnet.org/}{\DIFadd{http://microbialnet.org/}}\DIFadd{.
Through }\DIFaddend a systematic comparative analysis\DIFdelbegin \DIFdel{to }\DIFdelend \DIFaddbegin \DIFadd{, we }\DIFaddend determine which steps of the \DIFaddbegin \ac{micone} \DIFaddend pipeline have the largest influence on the final network, and \DIFdelbegin \DIFdel{what }\DIFdelend \DIFaddbegin \DIFadd{which }\DIFaddend choice seems to \DIFdelbegin \DIFdel{provide best }\DIFdelend \DIFaddbegin \DIFadd{have the optimal }\DIFaddend agreement with the tested mock and synthetic datasets.
\DIFdelbegin \DIFdel{We believe that these steps will }\DIFdelend \DIFaddbegin \DIFadd{These steps together with our default settings }\DIFaddend ensure better reproducibility and easier comparison of co-occurrence networks across datasets.
We expect that our tool will also be useful for benchmarking future alternative methods, and for ensuring a transparent evaluation of the possible biases introduced by the use of specific tools.


% RESULTS
%!TEX root = ../main.tex

\section*{Results}

  \subsection*{\acl{micone} (\acs{micone})}

  We \DIFdelbegin \DIFdel{have }\DIFdelend developed \ac{micone}, a flexible and modular pipeline for \DIFdelbegin \DIFdel{16S amplicon sequencing rRNA data (hereafter mentioned simply as 16S data) analysis, that allows us to infer microbial }\DIFdelend \DIFaddbegin \DIFadd{the inference of }\DIFaddend co-occurrence networks \DIFdelbegin \DIFdel{.
  It }\DIFdelend \DIFaddbegin \DIFadd{from 16S data.
  }\ac{micone} \DIFaddend incorporates various popular, publicly available tools as well as custom Python modules \DIFdelbegin \DIFdel{and scripts to facilitate inference of co-occurrence networks from }\DIFdelend \DIFaddbegin \DIFadd{for }\DIFaddend 16S data \DIFdelbegin \DIFdel{(see }\DIFdelend \DIFaddbegin \DIFadd{analysis and network inference (}\DIFaddend Methods).
  \DIFdelbegin \DIFdel{Using }\DIFdelend \DIFaddbegin \DIFadd{The different steps that are a part of the }\DIFaddend \ac{micone} \DIFdelbegin \DIFdel{one can obtain }\DIFdelend co-occurrence \DIFdelbegin \DIFdel{networks by applying to 16S data (or to already processed taxonomic count matrices) any combination of the available tools.
  The effects of changing any of the intermediate step can be monitored and evaluated in terms of its final network outcome, as well as on any of the intermediate metrics and data outputs.
  The }%DIFDELCMD < \ac{micone} %%%
\DIFdel{pipeline workflow is shown in }\DIFdelend \DIFaddbegin \DIFadd{network inference workflow (}\DIFaddend Figure~\ref{fig:figure1}\DIFdelbegin \DIFdel{.
  The different steps for going from 16S data to co-occurrence networks }\DIFdelend \DIFaddbegin \DIFadd{) }\DIFaddend can be grouped into \DIFdelbegin \DIFdel{four }\DIFdelend \DIFaddbegin \DIFadd{five }\DIFaddend major modules; (i) \DIFdelbegin \DIFdel{the denoising and clustering (DC) step, which handles denoising of the raw 16S sequencing data into representative sequences}\DIFdelend \DIFaddbegin \ac{sp}\DIFaddend ; (ii) \DIFdelbegin \DIFdel{the taxonomy assignment (TA) step that assigns taxonomic labels to the representative sequences}\DIFdelend \DIFaddbegin \ac{dc}\DIFaddend ; (iii) \DIFdelbegin \DIFdel{the }%DIFDELCMD < \ac{otu} %%%
\DIFdel{processing (OP) step that filters and transforms the taxonomy abundance table; and finally (}\DIFdelend \DIFaddbegin \ac{ta}\DIFadd{; (}\DIFaddend iv) \DIFdelbegin \DIFdel{the network inferences (NI) step which infers the microbial co-occurrence network}\DIFdelend \DIFaddbegin \ac{op}\DIFadd{; and (v) }\ac{ni}\DIFaddend .
  Each process in the pipeline \DIFdelbegin \DIFdel{supports alternate tools for performing the same task }\DIFdelend \DIFaddbegin \DIFadd{is implemented through multiple tools }\DIFaddend (see Methods and Figure~\ref{fig:figure1}).
  \DIFdelbegin \DIFdel{A centralized configuration file contains all the specifications for what modules are used in the pipeline , and can be modified by the user to choose the desired set of tools .
  In what follows, we perform }\DIFdelend \DIFaddbegin \DIFadd{The effects of changing any intermediate step of the pipeline can be evaluated in terms of the final network outcome, as well as on any of the intermediate metrics and data outputs.
  The choice of tools and parameters is encoded in a configuration file (with parameters as shown in Tables S2-S6 at }\href{https://github.com/segrelab/MiCoNE-pipeline-paper}{\DIFadd{https://github.com/segrelab/MiCoNE-pipeline-paper}}\DIFadd{).
  Through }\DIFaddend a systematic analysis of \DIFaddbegin \DIFadd{tool combinations at }\DIFaddend each step of the pipeline\DIFdelbegin \DIFdel{to estimate }\DIFdelend \DIFaddbegin \DIFadd{, we estimated }\DIFaddend how much the final co-occurrence network depends on the possible choices at each step.
\DIFdelbegin \DIFdel{We also evaluate a large number of tool combinations to determine a set of recommended default options for the pipeline and provide the users with a set of guidelines to facilitate tool selection as appropriate for their data.
}\DIFdelend 

  Our analysis \DIFdelbegin \DIFdel{involves }\DIFdelend \DIFaddbegin \DIFadd{involved }\DIFaddend two types of data: The first type \DIFdelbegin \DIFdel{consists of sets of }\DIFdelend \DIFaddbegin \DIFadd{consisted of }\DIFaddend 16S sequencing data from \DIFdelbegin \DIFdel{real communities sampled from human Stool and Oral microbiomes }\DIFdelend \DIFaddbegin \DIFadd{samples of human stool microbiomes from a fecal microbiome transplant (FMT) study of autism~\mbox{%DIFAUXCMD
\cite{Kang2017}}\hskip0pt%DIFAUXCMD
}\DIFaddend .
  The second \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{type was a collection of }\DIFaddend datasets synthetically or artificially created for the specific goal of \DIFdelbegin \DIFdel{helping evaluate }\DIFdelend \DIFaddbegin \DIFadd{evaluating }\DIFaddend computational analysis tools\DIFdelbegin \DIFdel{(see Methods)}\DIFdelend .
  In particular, in order to \DIFdelbegin \DIFdel{objectively compare, to the extent possible, how well }\DIFdelend \DIFaddbegin \DIFadd{benchmark }\DIFaddend each step in \ac{micone}\DIFdelbegin \DIFdel{best captures the underlying data, we use }\DIFdelend \DIFaddbegin \DIFadd{, we used }\DIFaddend both mock data (\DIFdelbegin \DIFdel{labelled }\DIFdelend \DIFaddbegin \DIFadd{labeled }\DIFaddend mock4, mock12\DIFaddbegin \DIFadd{, }\DIFaddend and mock16) from mockrobiota~\cite{Bokulich2016} \DIFdelbegin \DIFdel{as well as, synthetically generated reads from an Illumina read simulator called ART~\mbox{%DIFAUXCMD
\cite{Huang2012}}\hskip0pt%DIFAUXCMD
.
  These mock datasets consist of fake sequencing reads generated from reads obtained from synthetic microbial isolates mixed in know proportions. They contain the expected compositions along with the reference sequences for the organisms in the mock community.
  The synthetic reads were simulated using three different taxonomy distribution profiles, namely soil and water microbiomes obtained }%DIFDELCMD < \ac{emp}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Thompson2017} }\hskip0pt%DIFAUXCMD
and Stool microbiome that is used in our real community analysis~\mbox{%DIFAUXCMD
\cite{Kang2017}}\hskip0pt%DIFAUXCMD
.
  Reference sequences were generated using }%DIFDELCMD < \ac{ncbi} %%%
\DIFdel{and the Decard package~\mbox{%DIFAUXCMD
\cite{Golob2017} }\hskip0pt%DIFAUXCMD
for these taxonomy profiles.
  Detailed information on the mock communities and the settings used to generate the synthetic data are provided in the Methodssection}\DIFdelend \DIFaddbegin \DIFadd{and synthetic networks generated using the NorTA~\mbox{%DIFAUXCMD
\cite{Kurtz2015} }\hskip0pt%DIFAUXCMD
and seqtime~\mbox{%DIFAUXCMD
\cite{Rottjers2018} }\hskip0pt%DIFAUXCMD
approaches (See Methods)}\DIFaddend .

  \FloatBarrier

  \subsection*{\DIFdelbegin \DIFdel{The choice }\DIFdelend \DIFaddbegin \DIFadd{DC: Denoising and clustering methods differ in their identification }\DIFaddend of \DIFdelbegin \DIFdel{reference database has the biggest impact on inferred networks}\DIFdelend \DIFaddbegin \DIFadd{sequences that are low in abundance}\DIFaddend }

  \DIFdelbegin \DIFdel{In order to analyze the effect of different statistical methods on the inferred co-occurrence networks, we generated co-occurrence networks using all possible combinations of methods and estimated the variability in the networks due to each choice (Figure \ref{fig:figure1}).
  This analysis is performed while keeping the network inference algorithm (NI step) the same throughout the analysis.
  The effects of various steps on the final co-occurrence network is estimated by building a linear model of the edges of the network as a function the various step in the analysis pipeline (see Methods).
  Figure \ref{fig:figure2}B, shows the fraction of total variation among the co-occurrence networks due to the first three steps of the pipeline. In other words, each point corresponds to a different combination of tools, and captures how much the final network is affected by such choice.
  The 16S reference database contributes the most ($\sim25\%$) to variation in the networks. This is also reflected in the fact that the networks can be clearly separated based on the database used (Figure \ref{fig:figure2}B).
  This indicates that the taxonomy assigned to the reference sequences drastically alters the co-occurrence network.
  In fact the variability induced by taxonomy assignment is much more significant than that due to the variability induced based on how the reference sequences themselves are identified }\DIFdelend \DIFaddbegin \DIFadd{The }\ac{dc} \DIFadd{step is commonly carried out to generate representative sequences }\DIFaddend (in the \DIFdelbegin \DIFdel{DC step).
  The grouping of the networks by taxonomy assignment into clusters (Figure~\ref{fig:figure2}B) seems to derive from the mislabelling of constitutive taxa that are present in high abundance in the community, which drastically alter the nodes and hence the underlying network topology.
  The residual variation (Figure \ref{fig:figure2}A) can be seen as an artifact that arises when multiple steps are changed at the same time.
  Another interesting observation (elaborated in detail in the denoising and clustering section) is that the dissimilarity between the networks decreases when the low abundance }%DIFDELCMD < \ac{otu}%%%
\DIFdel{s are removed from the network.
  These results suggest that the most important criterion for accurate comparative analyses of co-occurrence networks is the taxonomy reference database.
}%DIFDELCMD < 

%DIFDELCMD <   \FloatBarrier
%DIFDELCMD < 

%DIFDELCMD <   %%%
\subsection*{\DIFdel{Denoising and clustering methods differ in their identification of less common reference sequences}}
%DIFAUXCMD
%DIFDELCMD < 

%DIFDELCMD <   %%%
\DIFdel{Denoising and clustering are commonly carried out to generate representative sequences from the raw }\DIFdelend \DIFaddbegin \DIFadd{form of the }\acs{otu}\DIFadd{/}\acs{esv} \DIFadd{tables) from the demultiplexed and trimmed }\DIFaddend 16S sequencing data\DIFdelbegin \DIFdel{and to obtain the }%DIFDELCMD < \ac{otu}%%%
\DIFdel{/}%DIFDELCMD < \ac{esv} %%%
\DIFdel{tables (counts of these representative sequences for each sample)}\DIFdelend .
  In order to compare the \DIFdelbegin %DIFDELCMD < \ac{otu} %%%
\DIFdelend \DIFaddbegin \DIFadd{count }\DIFaddend tables generated by different tools\DIFaddbegin \DIFadd{, }\DIFaddend we processed the \DIFdelbegin \DIFdel{same }\DIFdelend 16S sequencing reads (\DIFdelbegin \DIFdel{healthy samples from a fecal microbiome transplant }\DIFdelend \DIFaddbegin \DIFadd{from the FMT }\DIFaddend study~\cite{Kang2017}) using 5 different methods: open-reference clustering, closed-reference clustering, \DIFdelbegin \DIFdel{denovo }\DIFdelend \DIFaddbegin \DIFadd{de novo }\DIFaddend clustering, \ac{dada2}~\cite{Callahan2016} and Deblur~\cite{Amir2017}.
  The first three methods are from the \DIFdelbegin %DIFDELCMD < \ac{qiime1}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Caporaso2010} }\hskip0pt%DIFAUXCMD
package.
  We find that there is good agreement in the }%DIFDELCMD < \ac{otu}%%%
\DIFdel{/}%DIFDELCMD < \ac{esv} %%%
\DIFdel{tables when different combinations of methods are used to generate them (Supplementary Figure~\ref{fig:figureS1}).
  }\DIFdelend \DIFaddbegin \DIFadd{vsearch plugin from }\ac{qiime2}\DIFadd{~\mbox{%DIFAUXCMD
\cite{bolyenReproducibleInteractiveScalable2019}}\hskip0pt%DIFAUXCMD
.
  The closed and open reference methods in this analysis use the }\acl{gg}\DIFadd{~\mbox{%DIFAUXCMD
\cite{DeSantis2006} }\hskip0pt%DIFAUXCMD
database for reference sequence alignment.
}\DIFaddend 

  \DIFdelbegin \DIFdel{To compare the representative sequences generated by these methods we employ }\DIFdelend \DIFaddbegin \DIFadd{A comparison of the different methods was carried out by calculating the mean UniFrac distances across all samples (Figure~\ref{fig:figure2}).
  The analysis was performed using }\DIFaddend both the weighted \DIFaddbegin \DIFadd{UniFrac}\DIFaddend ~\cite{Lozupone2007} (Figure~\DIFdelbegin \DIFdel{\ref{fig:figure3}A) and unweighted UniFrac method~\mbox{%DIFAUXCMD
\cite{Lozupone2005} }\hskip0pt%DIFAUXCMD
(Figure~\ref{fig:figure3}B).
  The weighted UniFrac distance metric}\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure2}A) distance metric, which }\DIFaddend takes into account the counts of the representative sequences, \DIFdelbegin \DIFdel{whereas }\DIFdelend \DIFaddbegin \DIFadd{and }\DIFaddend the unweighted UniFrac\DIFdelbegin \DIFdel{distance metric does not and hence }\DIFdelend \DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{Lozupone2005} }\hskip0pt%DIFAUXCMD
(Figure~\ref{fig:figure2}B) distance metric, which }\DIFaddend gives equal weights to each sequence.
\DIFdelbegin \DIFdel{From Figure~\ref{fig:figure3}A one can see }\DIFdelend \DIFaddbegin 

  \DIFadd{The first main message emerging from this analysis is }\DIFaddend that the representative sequences generated by the different methods\DIFaddbegin \DIFadd{, with the exception of Deblur, }\DIFaddend are similar to each other when weighted by their abundance \DIFdelbegin \DIFdel{.
  Figure~\ref{fig:figure3}B on the other hand shows an increase in dissimilarity between each pair of methods suggesting that the methods might differ in the treatment }\DIFdelend \DIFaddbegin \DIFadd{(Figure~\ref{fig:figure2}A).
  A second message is that the different methods differ mainly in the assignment }\DIFaddend of sequences of \DIFdelbegin \DIFdel{low abundance.
  In order to verify this claim, for each of these methods we use the }%DIFDELCMD < \ac{gg} %%%
\DIFdel{taxonomy database to assign taxonomies to the representative sequences.
  We then correlate the abundances of matching taxonomies between a pair of DC methods (Figure\ref{fig:figureS1}A and B) .
  The }%DIFDELCMD < \ac{esv} %%%
\DIFdel{tables generated by methods that perform denoising are very similar to each other ($\sim0.91$) and the }%DIFDELCMD < \ac{otu} %%%
\DIFdel{tables generated by the clustering methods are very similar to each other ($\sim0.9$), but results of denoising and clustering are highly uncorrelated with each other ($\sim0.4$) (Figure \ref{fig:figureS1}C}\DIFdelend \DIFaddbegin \DIFadd{lower abundance.
  This can be inferred from the unweighted comparison (Figure~\ref{fig:figure2}B) which shows an increase in dissimilarity between each pair of methods (see additional details in Supplementary and Figure \ref{fig:figure_s2}}\DIFaddend ).

  These comparisons only elucidate the \DIFdelbegin \DIFdel{pairwise similarity or dissimilarity of }\DIFdelend \DIFaddbegin \DIFadd{similarity between }\DIFaddend a pair of methods.
  \DIFdelbegin \DIFdel{In order to determine the tool that }\DIFdelend \DIFaddbegin \DIFadd{To determine which tool }\DIFaddend most accurately recapitulates the reference sequences in the samples, we \DIFdelbegin \DIFdel{used the 16S sequences from the mock datasets.
  In particular, we used the pipeline to process mock community datasets using each of the possible methods included for this step.
  We next compared }\DIFdelend \DIFaddbegin \DIFadd{applied the same pipeline step to process the mock datasets (mock4, mock12, and mock16) and compared the }\DIFaddend predicted representative sequences with \DIFdelbegin \DIFdel{expected representative }\DIFdelend \DIFaddbegin \DIFadd{the true }\DIFaddend sequences and their distribution.
  The results (Figure~\DIFdelbegin \DIFdel{\ref{fig:figure3}C and }\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure2}C and \ref{fig:figure2}}\DIFaddend D) show that \DIFdelbegin \DIFdel{, for the mock datasets, the different methods perform similar to each other, exactly as observed in the case of the real dataset. However, the mock }\DIFdelend \DIFaddbegin \DIFadd{the }\DIFaddend predicted sequence distributions are \DIFdelbegin \DIFdel{substantially }\DIFdelend \DIFaddbegin \DIFadd{overall }\DIFaddend different from the expected \DIFdelbegin \DIFdel{sequence distribution.
  This result is more exaggerated in the case of the unweighted UniFrac metric, where some of the datasets show a very high deviation from the expected sequences.
  These high deviations are primarily in two of the three datasets that were analyzed and show that }\DIFdelend \DIFaddbegin \DIFadd{ones.
  The variation across datasets indicates that }\DIFaddend the datasets themselves play a big role in \DIFdelbegin \DIFdel{the performanceof these methods.
  This can be clearly seen in the performance (weighted UniFrac distance) of }%DIFDELCMD < \ac{dada2} %%%
\DIFdel{and Deblur on mock12 and mock16 datasets, where, Deblur outperforms }%DIFDELCMD < \ac{dada2} %%%
\DIFdel{on mock12 but the under-performs on mock16}\DIFdelend \DIFaddbegin \DIFadd{method performance}\DIFaddend .
  \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD <   %%%
\DIFdel{There }\DIFdelend \DIFaddbegin \DIFadd{We note that there }\DIFaddend is no method that \DIFdelbegin \DIFdel{clearly }\DIFdelend outperforms the rest in all datasets \DIFaddbegin \DIFadd{(see Supplementary for an extended discussion)}\DIFaddend .
  Based on their slightly better performance on the mock datasets, their \DIFdelbegin \textit{\DIFdel{de novo}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{de novo }\DIFaddend error correcting nature and \DIFdelbegin \DIFdel{other previous studies}\DIFdelend \DIFaddbegin \DIFadd{previous independent evaluation}\DIFaddend ~\cite{Nearing2018}, \ac{dada2} and Deblur \DIFdelbegin \DIFdel{seem to be in general }\DIFdelend \DIFaddbegin \DIFadd{appear to be }\DIFaddend the most reliable.
  \DIFdelbegin \DIFdel{Given the unexpected poor performance of Deblur on the synthetic data, the default algorithm in the pipeline was chosen to be }%DIFDELCMD < \ac{dada2} %%%
\DIFdel{(Supplementary Figure~\ref{fig:figureS3}).
}\DIFdelend \DIFaddbegin \DIFadd{This is because the open-reference and de novo clustering methods return a much larger number of }\ac{otu}\DIFadd{s compared to the other pipelines and would affect the accuracy of the network inference step if stringent filtering is not performed.
  Overall, since }\ac{dada2} \DIFadd{as compared to Deblur, displays better performance on all the mock datasets on the weighted UniFrac metric, we set this tool as the default for the DC step of the pipeline.
  However, if comparison across studies that have sequenced different 16S regions is required, closed-reference and open-reference might be a better option.
}\DIFaddend 

  \DIFaddbegin \DIFadd{After the denoising, the sequences are subject to Chimera Checking (CC).
  The }\ac{micone} \DIFadd{pipeline supports two different chimera checking methods, ``uchime-denovo"~\mbox{%DIFAUXCMD
\cite{bolyenReproducibleInteractiveScalable2019}}\hskip0pt%DIFAUXCMD
, and ``remove bimera"~\mbox{%DIFAUXCMD
\cite{Callahan2016}}\hskip0pt%DIFAUXCMD
.
  We did not notice any notable difference between the two methods (Figure~\ref{fig:figure_s3}), implying that they identify and remove mostly the same set of sequences as chimeras.
  Since the remove bimera method was originally developed in conjunction with dada2 we use this method as the default.
  The DC step thus results in a reduced set of unique sequences, which will be referred to as representative sequences in the subsequent steps.
}

  \DIFaddend \FloatBarrier

  \subsection*{\DIFaddbegin \DIFadd{TA: }\DIFaddend Taxonomy databases vary widely in taxonomy \DIFdelbegin \DIFdel{hierarchy and update frequency}\DIFdelend \DIFaddbegin \DIFadd{assignments beyond Order level}\DIFaddend }

  Taxonomy databases are used to assign taxonomic identities to the representative sequences obtained after the DC step.
  \DIFdelbegin \DIFdel{In order to compare the assigned taxonomies from different databases, we use the same reference sequences and assign taxonomies to them using different taxonomy reference databases.
  }\DIFdelend The three 16S taxonomic reference databases used in this study are SILVA~\cite{Quast2012}, \ac{gg}~\cite{DeSantis2006} and \ac{ncbi} RefSeq~\cite{Sayers2009} \DIFdelbegin \DIFdel{.
  SILVA and }%DIFDELCMD < \ac{gg} %%%
\DIFdel{are two popular 16S databases used for taxonomy identification.
  The }%DIFDELCMD < \ac{ncbi} %%%
\DIFdel{RefSeq nucleotide database contains 16S rRNA sequences as a part of two BioProjects - 33175 and 33317.
  The three databases vastly differ in terms of their last update status - }%DIFDELCMD < \ac{gg} %%%
\DIFdel{was last updated on May 2013, SILVA was last updated on December 2017 at the time of writing and }%DIFDELCMD < \ac{ncbi} %%%
\DIFdel{is updated as new sequences are curated.
  Since updates to taxonomic classifications are frequent, these databases vary significantly }\DIFdelend \DIFaddbegin \DIFadd{(Methods).
  These databases vary substantially }\DIFaddend in terms of taxonomy hierarchies\DIFaddbegin \DIFadd{, }\DIFaddend including species names and phylogenetic relationships~\cite{Balvociute2017}.
  \DIFaddbegin \DIFadd{Assignment using a particular database also requires a query tool.
  We used the ``Naive Bayes'' classifier from }\ac{qiime2} \DIFadd{for the }\ac{gg} \DIFadd{and SILVA databases and the ``BLAST'' tool (included as a }\ac{qiime2} \DIFadd{plugin) for the }\ac{ncbi} \DIFadd{database.
  These tools have been well quantified and optimized~\mbox{%DIFAUXCMD
\cite{bokulichOptimizingTaxonomicClassification2018}}\hskip0pt%DIFAUXCMD
, hence, we made use of the default parameters in our analyses.
}\DIFaddend 

  The representative sequences obtained \DIFdelbegin \DIFdel{from the }%DIFDELCMD < \ac{dada2} %%%
\DIFdel{method in }\DIFdelend \DIFaddbegin \DIFadd{using the default settings of the }\DIFaddend DC step were used for taxonomic assignment using the three reference databases.
  Figure~\DIFdelbegin \DIFdel{\ref{fig:figure4}}\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure3}}\DIFaddend A depicts a flow diagram that shows how the top 50 representative sequences (sorted by abundance) are assigned a Genus according to the three \DIFdelbegin \DIFdel{different databases.
  We observe that not only does }\DIFdelend \DIFaddbegin \DIFadd{databases.
  The different databases lead to assignments that qualitatively display similar distributions. However, }\DIFaddend the assigned Genus \DIFdelbegin \DIFdel{composition vary significantly, but }\DIFdelend \DIFaddbegin \DIFadd{compositions also display clear differences, as does }\DIFaddend the percentage of unassigned representative sequences (\DIFdelbegin \DIFdel{gray)also differ.
  Even the most abundant }\DIFdelend \DIFaddbegin \DIFadd{pink).
  Some of the differences in Genus composition have a clear explanation, for example, abundant Genera like Bacteroides and Escherichia are assigned to different representative sequences.
  The large percentage of unassigned sequences is due to the large fraction of the }\DIFaddend representative \DIFdelbegin \DIFdel{sequence is assigned to }\DIFdelend \DIFaddbegin \DIFadd{sequences assigned to }\DIFaddend an "unknown" \DIFdelbegin \DIFdel{Genus in two of the three databases.
  A representative sequence might be assigned an "unknown" }\DIFdelend Genus \DIFdelbegin \DIFdel{for one of two reasons: the first is if the taxonomy identifier associated with the sequence in the database did not contain a Genus; the second (more likely)reason is that the database contains multiple sequences that are very similar to the query (representative) sequence and the consensus algorithm (from }%DIFDELCMD < \ac{qiime2}%%%
\DIFdel{) is unable to assign one particular Genus at the required confidence.
After assigning all the representative sequences to taxonomies we perform }\DIFdelend \DIFaddbegin \DIFadd{during the assignment process (Methods).
}

  \DIFadd{After the assignment, we performed }\DIFaddend a pairwise comparison of the similarity between \DIFdelbegin \DIFdel{assignments }\DIFdelend \DIFaddbegin \DIFadd{the top 100 assignments (by abundance) }\DIFaddend from different databases at every taxonomic level (Figure~\DIFdelbegin \DIFdel{\ref{fig:figure4}}\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure3}}\DIFaddend B).
  The \DIFdelbegin \DIFdel{assignments beyond Family }\DIFdelend \DIFaddbegin \DIFadd{comparisons of the assignments below the Order }\DIFaddend level (Family, Genus\DIFaddbegin \DIFadd{, }\DIFaddend and Species) \DIFdelbegin \DIFdel{are very dissimilar with $<70\%$ }\DIFdelend \DIFaddbegin \DIFadd{show less than $45\%$ }\DIFaddend similarity between any pair of databases.
  \DIFdelbegin \DIFdel{There are no two reference databases that are more similar than the other pairs, with }%DIFDELCMD < \ac{gg} %%%
\DIFdel{and SILVA producing only marginally similar assignments compared to }%DIFDELCMD < \ac{ncbi}%%%
\DIFdel{.
  }\DIFdelend This implies that the taxonomy assignments from each reference database are fairly unique\DIFdelbegin \DIFdel{and are largely responsible for the differences observed in the co-occurrence networks generated from different taxonomy databases.
  }%DIFDELCMD < 

%DIFDELCMD <   %%%
\DIFdel{Supplementary Figure~\ref{fig:figureS4} shows that the top 20 most abundant genera in the three resulting taxonomy composition tables are different.
  For example, }\DIFdelend \DIFaddbegin \DIFadd{.
  The comparison of all assigned genera (Figure~\ref{fig:figure_s4}), instead of just the top 100, contains a higher percentage of mismatches.
  This suggests that, comparatively, }\DIFaddend the most abundant \DIFdelbegin \DIFdel{genus in the }%DIFDELCMD < \ac{gg} %%%
\DIFdel{taxonomy table was }\textit{\DIFdel{Escherichia}} %DIFAUXCMD
\DIFdel{whereas in the SILVA taxonomy table it was }\textit{\DIFdel{Escherichia-Shigella}}%DIFAUXCMD
\DIFdel{.
  Although these are minor differences, when comparing a large number of taxonomy composition tables these problems are hard to diagnose.
%DIF <    The comparison of all assigned genera instead of the just the top 20 contains the same percentage of matches and mismatches, implying that there does not seem to exist a correlation between abundance and mismatch.
%DIF <    This suggests that the most abundant sequences are not necessarily the ones that are consistently matched to the same taxonomies in the different reference databases.
}\DIFdelend \DIFaddbegin \DIFadd{sequences are more consistently matched to the same taxonomies, at least for the dataset tested in the current analysis.
}\DIFaddend 

  \DIFdelbegin \DIFdel{As in the previous section, these comparisons only indicate similarity or dissimilarity between methods.
In order to }\DIFdelend \DIFaddbegin \DIFadd{To }\DIFaddend obtain an absolute measure of \DIFaddbegin \DIFadd{the }\DIFaddend accuracy of the taxonomic assignments\DIFdelbegin \DIFdel{we use the expected reference }\DIFdelend \DIFaddbegin \DIFadd{, we used the representative }\DIFaddend sequences from the \DIFaddbegin \DIFadd{DC step for }\DIFaddend mock datasets as the query sequences \DIFdelbegin \DIFdel{for the databases }\DIFdelend and the expected taxonomic composition as the standard to compare against\DIFaddbegin \DIFadd{.
  We used the Bray-Curtis distance metric~\mbox{%DIFAUXCMD
\cite{virtanenSciPyFundamentalAlgorithms2020} }\hskip0pt%DIFAUXCMD
to calculate the distance between the predicted and expected taxonomic distribution }\DIFaddend (Figure~\DIFdelbegin \DIFdel{\ref{fig:figure4}}\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure3}}\DIFaddend C).
  \DIFdelbegin \DIFdel{Again, we observe }\DIFdelend \DIFaddbegin \DIFadd{We find }\DIFaddend that none of the databases perform better than the others in absolute terms \DIFaddbegin \DIFadd{and that the dissimilarity with the expected composition is high ($>0.5$ for Family and Genus and $>0.9$ for Species), indicating that all the databases have some limitations when trying to recapture the expected taxonomic composition}\DIFaddend .

  \DIFdelbegin \DIFdel{Given that }\DIFdelend \DIFaddbegin \DIFadd{Since }\DIFaddend no database performs better than others against mock datasets, \DIFdelbegin \DIFdel{and that databases are almost equally distant from each other in terms of final output, }\DIFdelend the choice of which database to use \DIFdelbegin \DIFdel{should }\DIFdelend \DIFaddbegin \DIFadd{could }\DIFaddend be driven by other \DIFdelbegin \DIFdel{reason.
  One user-specific way to choose, would be based on the known representation of taxa for the microbiome of interest (see also Discussion).
  Another reason }\DIFdelend \DIFaddbegin \DIFadd{reasons (see Supplementary discussion).
  One reason to choose a particular database }\DIFaddend could be the frequency of updates and the potential for future growth\DIFdelbegin \DIFdel{, which prompted us to set }\DIFdelend \DIFaddbegin \DIFadd{.
  Both }\ac{gg}\DIFadd{, due to its frequent use in the literature~\mbox{%DIFAUXCMD
\cite{Balvociute2017}}\hskip0pt%DIFAUXCMD
, and }\DIFaddend \ac{ncbi}\DIFdelbegin \DIFdel{as the }%DIFDELCMD < \ac{micone} %%%
\DIFdel{standard }\DIFdelend \DIFaddbegin \DIFadd{, due to its regular revision and maintenance, could be good choices }\DIFaddend for taxonomy assignment.
  In \DIFdelbegin \DIFdel{addition to being regularly maintained and updated the }%DIFDELCMD < \ac{ncbi} %%%
\DIFdel{database already has the advantage that its accuracy of assignments is still comparable to the SILVA and }\DIFdelend \DIFaddbegin \DIFadd{our default pipeline, we choose }\DIFaddend \ac{gg} \DIFdelbegin \DIFdel{reference databases that are routinely used as reference databases}\DIFdelend \DIFaddbegin \DIFadd{as the default method.
}

  \DIFadd{The TA step results in a taxonomic counts table that is used as input to the subsequent steps of the pipeline.
  Note that the count tables at different levels can be obtained through aggregation; for example, Genus count tables were obtained by summing up the counts of the lower taxonomy levels (Species and }\ac{otu}\DIFadd{) that map to the same higher taxonomy level entity}\DIFaddend .

  \FloatBarrier

  \subsection*{\DIFdelbegin \DIFdel{Networks generated using different }\DIFdelend \DIFaddbegin \DIFadd{NI: Different }\DIFaddend network inference methods \DIFdelbegin \DIFdel{show notable difference in }\DIFdelend \DIFaddbegin \DIFadd{drastically affect }\DIFaddend edge-density and connectivity}

   %DIF <  TODO: Talk about the difference between correlations and associations
  \DIFdelbegin \DIFdel{The six different }\DIFdelend \DIFaddbegin \DIFadd{The ten }\DIFaddend network inference methods \DIFaddbegin \DIFadd{we }\DIFaddend used in this \DIFdelbegin \DIFdel{study are }%DIFDELCMD < \ac{magma}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Cougoul2019}}\hskip0pt%DIFAUXCMD
, }%DIFDELCMD < \ac{mldm}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Yang2017}}\hskip0pt%DIFAUXCMD
, }%DIFDELCMD < \ac{spieceasi}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Kurtz2015}}\hskip0pt%DIFAUXCMD
, }%DIFDELCMD < \ac{sparcc}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Friedman2012}}\hskip0pt%DIFAUXCMD
, Spearman and Pearson.
  These network inference methods }\DIFdelend \DIFaddbegin \DIFadd{step }\DIFaddend fall into two groups\DIFdelbegin \DIFdel{, }\DIFdelend \DIFaddbegin \DIFadd{: }\DIFaddend the first set of methods (Pearson, Spearman, \DIFdelbegin %DIFDELCMD < \ac{sparcc}%%%
\DIFdelend \DIFaddbegin \acs{sparcc}\DIFadd{~\mbox{%DIFAUXCMD
\cite{Friedman2012,Watts2018}}\hskip0pt%DIFAUXCMD
, and propr~\mbox{%DIFAUXCMD
\cite{quinnProprRpackageIdentifying2017}}\hskip0pt%DIFAUXCMD
}\DIFaddend ) infer pairwise correlations while the second set \DIFdelbegin \DIFdel{infer direct associations (}%DIFDELCMD < \ac{spieceasi}%%%
\DIFdel{, }%DIFDELCMD < \ac{mldm}%%%
\DIFdel{, }%DIFDELCMD < \ac{magma}%%%
\DIFdel{) }\DIFdelend \DIFaddbegin \DIFadd{(}\acs{spieceasi}\DIFadd{~\mbox{%DIFAUXCMD
\cite{Kurtz2015}}\hskip0pt%DIFAUXCMD
, FlashWeave~\mbox{%DIFAUXCMD
\cite{tackmannRapidInferenceDirect2019}}\hskip0pt%DIFAUXCMD
, }\acs{cozine}\DIFadd{~\mbox{%DIFAUXCMD
\cite{haCompositionalZeroinflatedNetwork2020a}}\hskip0pt%DIFAUXCMD
, }\acs{harmonies}\DIFadd{~\mbox{%DIFAUXCMD
\cite{jiangHARMONIESHybridApproach2020}}\hskip0pt%DIFAUXCMD
, }\acs{spring}\DIFadd{~\mbox{%DIFAUXCMD
\cite{yoonMicrobialNetworksSPRING2019}}\hskip0pt%DIFAUXCMD
, and }\acs{mldm}\DIFadd{~\mbox{%DIFAUXCMD
\cite{Yang2017}}\hskip0pt%DIFAUXCMD
) infer direct associations.
   Note that while Pearson and Spearman methods are included in the pipeline for completeness, they tend to generate a large number of spurious edges as they are not intended for compositional datasets.
   Thus, they are not included in subsequent quantitative analyses}\DIFaddend .
\DIFdelbegin \DIFdel{Pairwise correlation methods involve calculating the correlation coefficient between every pair of }%DIFDELCMD < \ac{otu}%%%
\DIFdel{/}%DIFDELCMD < \ac{esv}%%%
\DIFdel{s leading to the detection of spurious indirect connections.
   On the other hand, direct association methods use conditional independence to avoid the detection of correlated but indirectly connected }%DIFDELCMD < \ac{otu}%%%
\DIFdel{s~\mbox{%DIFAUXCMD
\cite{Kurtz2015,Menon2018}}\hskip0pt%DIFAUXCMD
.
}\DIFdelend 

  \DIFdelbegin \DIFdel{For the analysis presented in this section, we used the taxonomy composition }\DIFdelend \DIFaddbegin \DIFadd{Filtered (see }\ac{op} \DIFadd{step in Methods) genus-level counts }\DIFaddend table obtained using the \DIFdelbegin %DIFDELCMD < \ac{ncbi} %%%
\DIFdel{reference database as the input for algorithms that infer co-occurrence associations between the microbes.
  Figure~\ref{fig:figure5}Ashows the networks inferred from this dataset using the different inference algorithms.
  The different }\DIFdelend \DIFaddbegin \DIFadd{default settings in the previous steps were used as input for the different network inference algorithms (Figure~\ref{fig:figure4}).
  Even from a visual inspection (Figure~\ref{fig:figure4}A), one can see that the different }\DIFaddend networks differ vastly in their edge-density and connectivity\DIFdelbegin \DIFdel{; even some of the edges in common to these networks have their signs inverted . Note, however, that some of these comparisons depend on the threshold that has to be applied to the pairwise correlations methods (currently 0.3, based on~\mbox{%DIFAUXCMD
\cite{Friedman2012}}\hskip0pt%DIFAUXCMD
).
  To get a more quantitative picture of }\DIFdelend \DIFaddbegin \DIFadd{, with common edges often displaying inverted signs.
}

  \DIFadd{To quantify }\DIFaddend the differences between the \DIFdelbegin \DIFdel{inferred }\DIFdelend networks, we \DIFdelbegin \DIFdel{checked }\DIFdelend \DIFaddbegin \DIFadd{analyzed }\DIFaddend the distribution of common nodes and edges (Figure~\DIFdelbegin \DIFdel{\ref{fig:figure5}B }\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure4} B and \ref{fig:figure4}C}\DIFaddend ) using UpSet plots~\DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Lex} }\hskip0pt%DIFAUXCMD
(only }%DIFDELCMD < \ac{magma}%%%
\DIFdel{, }%DIFDELCMD < \ac{mldm}%%%
\DIFdel{, }%DIFDELCMD < \ac{spieceasi}%%%
\DIFdel{, }%DIFDELCMD < \ac{sparcc} %%%
\DIFdel{are used in the comparison since Pearson and Spearman add a large number of spurious edges since they are not intended for compositional datasets).
  The results for the node intersections show }\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{lexUpSetVisualizationIntersecting2014}}\hskip0pt%DIFAUXCMD
.
  The node intersection analysis shows }\DIFaddend that the networks have \DIFdelbegin \DIFdel{a large number of nodes in common ($63$ out of $67$ nodes in the smallest network - }%DIFDELCMD < \ac{magma}%%%
\DIFdel{) and }\DIFdelend \DIFaddbegin \DIFadd{$33$ out of $68$ total unique nodes in common and that }\DIFaddend no network possesses \DIFdelbegin \DIFdel{any }\DIFdelend \DIFaddbegin \DIFadd{a }\DIFaddend unique node.
  \DIFdelbegin \DIFdel{The edge }\DIFdelend \DIFaddbegin \DIFadd{Edge }\DIFaddend intersections in contrast show that only \DIFdelbegin \DIFdel{$19$ }\DIFdelend \DIFaddbegin \DIFadd{$8$ }\DIFaddend edges (out of \DIFdelbegin \DIFdel{$98$ edgesin the smallest network - }%DIFDELCMD < \ac{magma}%%%
\DIFdelend \DIFaddbegin \DIFadd{$202$ total unique edges}\DIFaddend ) are in common between all the methods and each network has \DIFdelbegin \DIFdel{a large number of }\DIFdelend \DIFaddbegin \DIFadd{many }\DIFaddend unique edges.
  These results \DIFdelbegin \DIFdel{indicate that there is }\DIFdelend \DIFaddbegin \DIFadd{showed }\DIFaddend a substantial rewiring of connections in \DIFdelbegin \DIFdel{the }\DIFdelend \DIFaddbegin \DIFadd{different inferred networks and prompted us to identify associations robust across methods, through consensus algorithms.
}

  \FloatBarrier

  \subsection*{\DIFadd{NI: The scaled-sum consensus method shows high precision on benchmark datasets}}

 \DIFadd{Inspired by previous approaches~\mbox{%DIFAUXCMD
\cite{bustinceFuzzySetsTheir2008,tsarevApplicationMajorityVoting2018}}\hskip0pt%DIFAUXCMD
, we developed two methods that take into consideration the evidence offered by each network inference algorithm and generate a consensus network that contains the common edges among the }\DIFaddend inferred networks.

  \DIFdelbegin \DIFdel{Unlike the }\DIFdelend \DIFaddbegin \DIFadd{Both of our approaches - simple voting (SV) and scaled-sum (SS) - combine appropriately filtered networks inferred from correlation-based and direct association methods (see Methods).
  We chose the scaled-sum method as the pipeline default since this method takes into account the weights of the associations in the determination of the final consensus.
  The pipeline enables the selection of any subset of methods for the consensus calculation. Currently, by default, all direct methods are used, together with }\acs{sparcc} \DIFadd{and propr.
}

  \DIFadd{Similar to what was done for the }\DIFaddend previous steps of the pipeline, \DIFdelbegin \DIFdel{where were }\DIFdelend \DIFaddbegin \DIFadd{and in analogy with previous estimations of network inference accuracy~\mbox{%DIFAUXCMD
\cite{Kurtz2015,Weiss2016}}\hskip0pt%DIFAUXCMD
, }\DIFaddend we evaluated the \DIFdelbegin \DIFdel{performance of methods on mock datasets, there is no equivalent dataset that contain a set of known interactions for }\DIFdelend \DIFaddbegin \DIFadd{network inference algorithms and the final consensus network using synthetic interaction data.
  For this purpose, we generated synthetic interaction data using the ``NorTA''~\mbox{%DIFAUXCMD
\cite{Kurtz2015} }\hskip0pt%DIFAUXCMD
and ``seqtime''~\mbox{%DIFAUXCMD
\cite{faustSignaturesEcologicalProcesses2018} }\hskip0pt%DIFAUXCMD
methods (see Methods).
  For each method, an }\ac{otu} \DIFadd{counts table was generated based on the selected parameters and abundance distributions.
  This counts table was used as the input to the }\ac{micone} \DIFadd{pipeline to generate predicted associations.
  The interaction network used to generate the counts table was used as the source of true interactions to calculate the precision (Figure \ref{fig:figure5}) and sensitivity (Figure \ref{fig:figure_s5} and Figure \ref{fig:figure_s6}) for each network inference algorithm.
  As shown in Figure \ref{fig:figure5} the consensus algorithm, especially the scaled-sum method, captures true associations with high precision (through the removal of edges that are either not present in most of the inference methods or whose association strength is low across methods).
  Overall, }\DIFaddend the \DIFdelbegin \DIFdel{evaluation of the network inference algorithms.
  Therefore, we propose the construction of a consensus network (Figure~\ref{fig:figure5}C)involving }%DIFDELCMD < \ac{magma}%%%
\DIFdel{, }%DIFDELCMD < \ac{mldm}%%%
\DIFdel{, }%DIFDELCMD < \ac{spieceasi} %%%
\DIFdel{and }%DIFDELCMD < \ac{sparcc}%%%
\DIFdel{.
 This consensus network is built by merging the p-values generated from bootstraps of the original taxonomy composition table using the Browns p-value combining method~\mbox{%DIFAUXCMD
\cite{Poole} }\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{scaled-sum method for $p=1.000$ performs the best (precision = $1.000$ for both NorTA and seqtime).
  The scaled-sum method for $p=0.333$ (default option in the pipeline) shows a high precision ($0.956$ with NorTA; $0.688$ with seqtime), without displaying significant reduction in sensitivity (Figure~\ref{fig:figure_s5} and Figure~\ref{fig:figure_s6}).
  However, if higher precision is required $p>0.5$ can be considered.
  Therefore, the consensus networks provide the means to obtain a short list of associations that would have a high likelihood of being present in the real association network.
}

  \FloatBarrier

  \subsection*{\DIFadd{Impact of different pipeline steps on co-occurrence networks}}

  \DIFadd{In order to analyze the effect of different processing methods on the inferred co-occurrence networks (before consensus estimation), we generated networks using all possible combinations of methods and quantified the variability due to each choice (Figure \ref{fig:figure6}A).
 This was achieved by building a linear model of the edges of the network as a function of the various steps in the pipeline workflow }\DIFaddend (see Methods\DIFdelbegin \DIFdel{section).
  Based on this approach, }%DIFDELCMD < \ac{micone} %%%
\DIFdel{reports as default output the consensus network , annotated with weights (correlations for }%DIFDELCMD < \ac{sparcc} %%%
\DIFdel{and direct associations for the other methods) for all four methods}\DIFdelend \DIFaddbegin \DIFadd{)}\DIFaddend .
  \DIFaddbegin \DIFadd{Figure \ref{fig:figure6}A, shows the percentage of total variation among the co-occurrence networks due to the different steps of the pipeline.
  The }\ac{ta} \DIFadd{step, or more specifically the choice of 16S reference database, contributes the most ($65.4\%$) to the variation in the networks, followed by the }\ac{op} \DIFadd{step ($26.8\%$).
  This result highlights the importance of the taxonomy assignment step in the 16S data analysis workflow, implying that a change in the reference database will result in drastically different inferred networks.
  This is likely due to the differential assignment of representative sequences to taxonomic entities (Figure~\ref{fig:figure3} and Figure~\ref{fig:figure_s4}), which drastically alter the nodes and hence the underlying network topology.
}\DIFaddend 

  \DIFaddbegin \DIFadd{The effects of the different steps of the pipeline on the inferred networks can be visualized through dimensionality reduction.
  The PCA in Figure \ref{fig:figure6}B shows all the above networks, colored by the tools used in the DC, TA, OP, and NI steps in each subfigure.
  The major effect of the TA step choice, shown before in Figure \ref{fig:figure6}A, is also reflected in the PCA plot, where networks segregate based on the database used (Figure~\ref{fig:figure6}B and Figure~\ref{fig:figure_s1}).
  Additionally, the plot also shows that the variation between the networks decreases when the low abundance }\ac{otu}\DIFadd{s are removed from the network.
  It is also evident that in the NI step, some networks, especially those inferred using the direct association network inference methods, are much closer in the PCA plot regardless of the reference database used.
  These results suggest that the most important criterion for accurate comparative analysis of co-occurrence networks is the taxonomy reference database followed by the level of filtering of the taxonomy tables and the network inference algorithm used.
}

  \DIFaddend \FloatBarrier

  \DIFdelbegin \subsubsection*{\DIFdel{The default pipeline}}
  %DIFAUXCMD
\DIFdelend \DIFaddbegin \subsection*{\DIFadd{The default pipeline}}
\DIFaddend 

  The systematic analyses \DIFdelbegin \DIFdel{performed }\DIFdelend in the previous sections \DIFdelbegin \DIFdel{clearly show }\DIFdelend \DIFaddbegin \DIFadd{illustrate }\DIFaddend that the choice of tools and parameters can have a big impact on the final \DIFaddbegin \DIFadd{consensus }\DIFaddend co-occurrence network.
  \DIFdelbegin \DIFdel{For some of these choices (e.g. }%DIFDELCMD < \ac{dada2} %%%
\DIFdel{vs. deblur) there is no clear metric to establish a best protocol.
  For other choices}\DIFdelend \DIFaddbegin \DIFadd{However}\DIFaddend , the mock communities \DIFaddbegin \DIFadd{and synthetic data }\DIFaddend provide an opportunity to select \DIFdelbegin \DIFdel{combination of parameters that yield more }\DIFdelend \DIFaddbegin \DIFadd{combinations of tools that yield the most }\DIFaddend accurate and robust results.
  \DIFdelbegin \DIFdel{Despite this partial degree of assessment, we wish to suggest a combination }\DIFdelend \DIFaddbegin \DIFadd{As highlighted in the above sections for individual steps, we propose a set }\DIFaddend of tools and parameters \DIFdelbegin \DIFdel{that produce networks that are derived from the combination of tools which performed best on the mock communities, and displayed highest robustness to switching to alternative methods.
  These tools and parameters are chosen }\DIFdelend as the defaults for the pipeline \DIFdelbegin \DIFdel{and are given in Table~\ref{tab:default_options}}\DIFdelend \DIFaddbegin \DIFadd{(Table~\ref{tab:micone_tools})}\DIFaddend .

  \DIFdelbegin \DIFdel{The recommended tool for the }%DIFDELCMD < \ac{dc} %%%
\DIFdel{step (}%DIFDELCMD < \ac{dada2} %%%
\DIFdel{or Deblur) were chosen based on their accuracy in recapitulating the reference sequences in mock communities and synthetic data.
  The choice of }\DIFdelend \DIFaddbegin \DIFadd{Figure~\ref{fig:figure7} shows the co-occurrence networks inferred for the healthy subjects (control) and subjects with autism specific disorder (ASD) in }\DIFaddend the \DIFdelbegin \DIFdel{taxonomy reference database in the }%DIFDELCMD < \ac{ta} %%%
\DIFdel{step is dictated largely by the species expected to be present in the sample as well the database used in similar studies if comparison is a goal.
  Nevertheless, we suggest }%DIFDELCMD < \ac{ncbi} %%%
\DIFdel{RefSeq along with blast+ as the query tool since the database is updated regularly and has a broad collection of taxonomies.
  The abundance threshold at the }%DIFDELCMD < \ac{op} %%%
\DIFdel{step is determined automatically based on the number of samplesand the required statistical power.
  Finally, we use the Browns p-value combining method on the networksgenerated using }%DIFDELCMD < \ac{magma}%%%
\DIFdel{, }%DIFDELCMD < \ac{mldm}%%%
\DIFdel{, }%DIFDELCMD < \ac{spieceasi} %%%
\DIFdel{and }%DIFDELCMD < \ac{sparcc} %%%
\DIFdel{to obtain a final consensus networkin the }%DIFDELCMD < \ac{ni} %%%
\DIFdel{step}\DIFdelend \DIFaddbegin \DIFadd{fecal microbiome transplant study~\mbox{%DIFAUXCMD
\cite{Kang2017} }\hskip0pt%DIFAUXCMD
(constructed using the default tools and parameters from Table~\ref{tab:micone_tools}).
  This figure demonstrates a typical use case of comparative analysis of networks using the }\ac{micone} \DIFadd{pipeline.
  As a consequence of using the consensus network algorithm, the final co-occurrence networks are sparse and can be visually compared and examined.
}

  \DIFadd{The analysis of the rewiring of associations in the ASD samples with respect to the control provides a guide for the identification of key genera that could be linked to dysbiosis.
  We observed 22 unique links in the network for control samples, 12 unique links in the network for ASD subjects, and 7 edges in common between the two networks.
  Although these unique associations do not imply actual interactions, they can still serve as potential starting points for literature surveys and further experimental exploration of mechanistic processes underlying dysbiosis.
  For example, }\textit{\DIFadd{Prevotella}} \DIFadd{and }\textit{\DIFadd{Porphyromonas}}\DIFadd{, genera previously implicated in ASD~\mbox{%DIFAUXCMD
\cite{Kang2017,hoGutMicrobiotaChanges2020} }\hskip0pt%DIFAUXCMD
and cognitive impairment~\mbox{%DIFAUXCMD
\cite{chiPorphyromonasGingivalisInducedCognitive2021} }\hskip0pt%DIFAUXCMD
display modified connectivity in our network, suggesting that the observed associations may be relevant for understanding the role of these bacteria in disease.
  Additional visualization and comparison of networks can be performed using the }\acf{mind}\DIFadd{~\mbox{%DIFAUXCMD
\cite{huResourceComparisonIntegration2022}}\hskip0pt%DIFAUXCMD
}\DIFaddend .

  Figure~\DIFdelbegin \DIFdel{\ref{fig:figure6}A shows }\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure_s7} shows a sensitivity analysis in which we compared }\DIFaddend the default network \DIFdelbegin \DIFdel{compared }\DIFdelend against networks generated by altering one of the steps of the pipeline \DIFdelbegin \DIFdel{from }\DIFdelend \DIFaddbegin \DIFadd{relative to }\DIFaddend the default.
  \DIFdelbegin \DIFdel{These results indicate that the biggest differences in networks occur when the reference database or the network inference algorithm are changed.
  Furthermore, the L1 distance of networks generated by altering one of the steps of the pipeline from the default against the default network }\DIFdelend \DIFaddbegin \DIFadd{This result, both visually }\DIFaddend (Figure~\DIFdelbegin \DIFdel{\ref{fig:figure6}B)  shows that the biggest deviations from the default network }\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure_s7} A), and quantitatively (Figure~\ref{fig:figure_s7} B)  suggests that the most significant changes }\DIFaddend occur when the \DIFaddbegin \ac{op} \DIFadd{or }\DIFaddend \ac{ta} \DIFdelbegin \DIFdel{and }%DIFDELCMD < \ac{ni} %%%
\DIFdelend steps are changed \DIFdelbegin \DIFdel{, reinforcing the same results observed in Figure~\ref{fig:figure2}. Figure~\ref{fig:figure7} shows the co-occurrence networks inferred for the hard palate for healthy subjects in a periodontal disease study~\mbox{%DIFAUXCMD
\cite{Chen2018} }\hskip0pt%DIFAUXCMD
and the healthy stool microbiome in fecal microbial transplant study~\mbox{%DIFAUXCMD
\cite{Kang2017}}\hskip0pt%DIFAUXCMD
. These consensus networks were generated using the default tools and parameters from Table~\ref{tab:default_options}}\DIFdelend \DIFaddbegin \DIFadd{from the default value}\DIFaddend .



% DISCUSSION
%!TEX root = ../main.tex

\section*{Discussion}

  %DIF <  General statements
\DIFdelbegin \DIFdel{Co-occurrence associations in microbial communities help identify important interactions that drive microbial community structure and organization.
  Our analysis shows }\DIFdelend \DIFaddbegin \subsection*{\DIFadd{Why }\ac{micone}\DIFadd{?}}

  \DIFadd{A myriad of tools and methods have been developed for different parts of the workflow for inference of co-occurrence networks from 16S rRNA data.
  Our analyses have shown }\DIFaddend that networks generated using different combinations of tools and approaches can \DIFdelbegin \DIFdel{look significantly }\DIFdelend \DIFaddbegin \DIFadd{be substantially }\DIFaddend different from each other, highlighting the \DIFdelbegin \DIFdel{importance of a clear assessment }\DIFdelend \DIFaddbegin \DIFadd{need for a clear evaluation }\DIFaddend of the source of variability and \DIFdelbegin \DIFdel{of }\DIFdelend \DIFaddbegin \DIFadd{for }\DIFaddend tools that provide the most robust and accurate results.
  Our newly developed \DIFdelbegin \DIFdel{integrated software}\DIFdelend \DIFaddbegin \DIFadd{software, }\ac{micone}\DIFadd{, is a customizable pipeline }\DIFaddend for the inference of co-occurrence networks from 16S rRNA data \DIFdelbegin \DIFdel{, }%DIFDELCMD < \ac{micone}%%%
\DIFdel{, constitutes a freely customizable and user friendly pipeline that allows users to easily test combinations of tools and to }\DIFdelend \DIFaddbegin \DIFadd{that enables users to }\DIFaddend compare networks generated by multiple possible \DIFdelbegin \DIFdel{choices (see Methods)}\DIFdelend \DIFaddbegin \DIFadd{combinations of tools and parameters}\DIFaddend .
  Importantly, in addition to revisiting the test cases presented in this work, users will be able to explore the effect of various tool combinations on their own datasets of interest.
  The \ac{micone} pipeline \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{has been }\DIFaddend built in a modular fashion\DIFdelbegin \DIFdel{.
Its }\DIFdelend \DIFaddbegin \DIFadd{; its }\DIFaddend plug-and-play architecture \DIFdelbegin \DIFdel{will make it possible for }\DIFdelend \DIFaddbegin \DIFadd{enables }\DIFaddend users to add new tools and steps, either \DIFdelbegin \DIFdel{from existing packages , or from packages that were not }\DIFdelend \DIFaddbegin \DIFadd{using existing packages that have not been }\DIFaddend examined in the present work \DIFdelbegin \DIFdel{, as well as futureones.
  }%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{The main outcome of this work is thus two-fold: on one hand we transparently reveal }\DIFdelend \DIFaddbegin \DIFadd{or those developed in the future.
  The }\ac{micone} \DIFadd{Python package provides functions and methods to perform a detailed analysis of the count matrices and the co-occurrence networks.
  The inferred networks are exported to a custom JSON format (see Supplementary) by default, but can also be exported to Cytoscape~\mbox{%DIFAUXCMD
\cite{shannonCytoscapeSoftwareEnvironment2003}}\hskip0pt%DIFAUXCMD
, GML~\mbox{%DIFAUXCMD
\cite{himsoltGMLPortableGraph2010}}\hskip0pt%DIFAUXCMD
, and many other popular formats via the Python package.
}

  \DIFadd{While several tools/workflows such as }\ac{qiime2}\DIFadd{~\mbox{%DIFAUXCMD
\cite{bolyenReproducibleInteractiveScalable2019} }\hskip0pt%DIFAUXCMD
and NetCoMi~\mbox{%DIFAUXCMD
\cite{peschelNetCoMiNetworkConstruction2020} }\hskip0pt%DIFAUXCMD
can be used to generate co-occurrence networks from 16S sequencing data, no single tool exist that integrates the complete process of inferring microbial interaction networks from 16S sequencing reads.
  }\ac{micone} \DIFadd{is unique as it offers this functionality packaged in a workflow that can be run locally, on the compute cluster, or in the cloud.
}

  \subsection*{\DIFadd{The default pipeline and recommended tools}}

  \DIFadd{Through }\ac{micone}\DIFadd{, in addition to transparently revealing }\DIFaddend the dependence of co-occurrence networks on tool and parameter choices \DIFdelbegin \DIFdel{, making it possible to more rigorously assess and compare existing networks.
On the other hand, we take }\DIFdelend \DIFaddbegin \DIFadd{(see Discussion in Supplementary Text for details on the DC, TA and OP steps), we have taken }\DIFaddend advantage of our spectrum of computational options and the availability of mock and synthetic datasets, to suggest a default standard setting\DIFdelbegin \DIFdel{, and }\DIFdelend \DIFaddbegin \DIFadd{.
  Additionally, we have developed }\DIFaddend a consensus approach, \DIFdelbegin \DIFdel{likely to yield }\DIFdelend \DIFaddbegin \DIFadd{that can reliably generate }\DIFaddend networks that are \DIFaddbegin \DIFadd{fairly }\DIFaddend robust across multiple tool \DIFdelbegin \DIFdel{/parameter choices.
  }%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend \DIFaddbegin \DIFadd{choices.
  }\DIFaddend An important caveat related to \DIFdelbegin \DIFdel{this last point is the fact that }\DIFdelend \DIFaddbegin \DIFadd{these results is that due to the lack of a universal standard for microbial interaction data, }\DIFaddend our conclusions are based on the specific datasets used in our analysis.
  While our \DIFdelbegin \DIFdel{datasets cover a relatively broad spectrum of biomes and sequencing pipelines}\DIFdelend \DIFaddbegin \DIFadd{analysis is based on several mock and synthetic datasets that cover a diverse range of abundance distributions and network topologies}\DIFaddend , datasets that have drastically different distributions may require a re-assessment of the best settings\DIFdelbegin \DIFdel{through our pipeline}\DIFdelend .

  \DIFdelbegin \DIFdel{It is worth pointing out some additional more specific conclusions stemming from the individual steps of our analysis.
}%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{The different denoising/clustering methods differ mostly in their identification of sequences that are in low abundances.
  Hence, they do not have much of an impact on the inferred co-occurrence networks when the sequences of low abundance are removed.
However, comparison of inferred and expected reference sequences and their abundances in mock community datasets has allowed us to identify }%DIFDELCMD < \ac{dada2} %%%
\DIFdel{as the method which best recapitulates the expected sequence composition.
For the current work we have decided to focus on the tools most widely used at the time of the analysis.
 Some tools that we recently published (e.g. dbOTU3~\mbox{%DIFAUXCMD
\cite{Olesen2017}}\hskip0pt%DIFAUXCMD
) as well as older popular methods like mothur~\mbox{%DIFAUXCMD
\cite{Schloss2009} }\hskip0pt%DIFAUXCMD
have not been included in the study, but could be added into the pipelines in future updated analyses.
  }%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{The choice of taxonomy database was found to be the most important factor in the inference of a microbial co-occurrence network, contributing $\sim20\%$ of the total variance.
}\DIFdelend The \DIFdelbegin \DIFdel{frequent changes in the taxonomy nomenclature coupled with the frequency of updates to the various 16S reference databases create inherent differences \mbox{%DIFAUXCMD
\cite{Balvociute2017} }\hskip0pt%DIFAUXCMD
in taxonomy hierarchies in these databases.
Our analysis revealed that no particular reference database performs better than the others across all scenarios.
  We suggest that that choice of the database should be made based on possible reported or inferred biases in the representation of given biomes in a specific databases \mbox{%DIFAUXCMD
\cite{Balvociute2017}}\hskip0pt%DIFAUXCMD
.
The default reference database in the pipeline is the }%DIFDELCMD < \ac{ncbi} %%%
\DIFdel{16S RefSeq database as it is more frequently updated and is most compatible with the blast+ query tool.
We also enable users to use custom databases \mbox{%DIFAUXCMD
\cite{Ritari2015} }\hskip0pt%DIFAUXCMD
with the blast+ and naive bayes classifiers that are incorporated into the pipeline (from }%DIFDELCMD < \ac{qiime2}%%%
\DIFdel{).
  }\DIFdelend \DIFaddbegin \DIFadd{networks generated by different network inference methods show considerable differences in edge-density and connectivity, partially due to the underlying assumptions regarding sparsity, distribution, and compositionality.
  To address this issue, we have developed two consensus algorithms (simple voting and scaled-sum method) that generate networks whose links have evidence based on multiple inference algorithms.
}\DIFaddend 

 \DIFdelbegin \DIFdel{Filtering out taxa that are present in low abundances in all samples did not increase (in most datasets tested) the proportion of taxa in common between taxonomy tables generated using different reference databases.
  However, we do observe that the reduction in the number of taxa leads to better agreement in the networks inferred through different methods.
  Moreover, filtering is necessary in order to increase the power in tests of significance when the number of taxa is much greater than the number of samples}\DIFdelend \DIFaddbegin \DIFadd{We find that the scaled-sum method performs the best on synthetic datasets, and is therefore chosen as the default for the }\ac{ni} \DIFadd{step of the pipeline.
 Notably, the consensus network displays a higher precision and returns a concise list of robust associations which represent a valuable set for experimental validation follow-up}\DIFaddend .

  \DIFdelbegin \DIFdel{The networks generated by different network inference methods show considerable differences in edge-density and connectivity.
One reason for this is the underlying assumptions regarding sparsity, distribution and compositionality that the algorithms make.
The consensus network created by merging the networks inferred using the different network inference methods enables the creation of a network whose links have evidence based on multiple inferencealgorithms.
  }\DIFdelend \DIFaddbegin \subsection*{\DIFadd{Future directions}}
\DIFaddend 

  %DIF <  \hl{Other factors that play an important role in network inference} \\
%DIF <  Differences in environments of the samples could lead to the inference of spurious interactions [ref].
\DIFaddbegin \DIFadd{Future work building upon our current results could enhance the network inference process in multiple ways.
  The current analyses make use of one fecal microbiome transplant dataset with healthy and ASD samples, three mock community datasets, and several datasets generated by two synthetic interaction methods.
  Incorporating datasets from a broad spectrum of biomes with varying microbial distributions into }\ac{micone} \DIFadd{will likely increase the robustness and generalizability of the results from these analyses.
}

  \DIFadd{The network analyses in this study are primarily at the Genus level, wherein the lowest resolution of a node is a Genus and if an entity cannot be resolved to the Genus level, the next lowest taxonomic level is used (for example, Family).
  As a consequence, two entities belonging to the same lineage where one entity is resolved to the Genus level and another is resolved to the Family level are treated as two different nodes in the network.
  Thus, the development of a metric of overlap to compare nodes with shared lineages within and across networks could enable more biologically and phylogenetically relevant comparisons.
}\DIFaddend 

  \DIFdelbegin \DIFdel{Exploring the effects of these combinations of methods on the resultant networks is difficult and inconvenient since different tools differ in their input and output formats and require inter-converting between the various formats.
The pipeline facilitates this comparative exploration by providing a variety of modules for inter-conversion between various formats, and by allowing easy incorporation of new tools as modules.
}\DIFdelend \DIFaddbegin \DIFadd{Although direct comparisons between co-occurrence networks and directly measured interactions are difficult to interpret and highly debated~\mbox{%DIFAUXCMD
\cite{hiranoDifficultyInferringMicrobial2019,gobernaCautionaryNotesUse2022}}\hskip0pt%DIFAUXCMD
.
  Further, benchmarking of co-occurrence networks could also be pursued through the use of literature-based interactions~\mbox{%DIFAUXCMD
\cite{lima-mendezDeterminantsCommunityStructure2015a} }\hskip0pt%DIFAUXCMD
or biological benchmark interaction data~\mbox{%DIFAUXCMD
\cite{sungGlobalMetabolicInteraction2017a}}\hskip0pt%DIFAUXCMD
.
  Additionally, }\ac{micone} \DIFadd{could be extended to enable the processing of metagenomics sequencing data, facilitating the analysis of a much larger and diverse range of datasets and domains of life.
}\DIFaddend 

  \DIFaddbegin \DIFadd{Although in the current analysis, we have only used default parameter values recommended by the tool creators, the }\ac{micone} \DIFadd{pipeline could be used in the future to explore any combinations of parameters and to optimize these values for improved network inference.
  Overall, there likely is no ``best method'' for the various steps of 16S data analysis, and hence, }\ac{micone} \DIFadd{is intended to help researchers to identify the methods and algorithms that are most suitable for their datasets in an easy-to-use and reproducible manner.
}

  \DIFaddend We envision that \ac{micone}, and \DIFdelbegin \DIFdel{the }\DIFdelend \DIFaddbegin \DIFadd{its }\DIFaddend underlying tools and databases\DIFdelbegin \DIFdel{that help process amplicon sequencing data into co-occurrence networks}\DIFdelend , will be increasingly useful \DIFdelbegin \DIFdel{towards }\DIFdelend \DIFaddbegin \DIFadd{for }\DIFaddend building large comparative analyses across studies.
  \DIFdelbegin \DIFdel{By having a unified transparent tool to compute networks, it will be possible to reprocess available 16S datasets to obtain networks that are directly comparable to each other.
Furthermore, even in the analysis of published networks across studies and processing methods, }%DIFDELCMD < \ac{micone} %%%
\DIFdel{could help understand underlying biases of each network , which could in turn be taken into account upon making cross-study comparisons}\DIFdelend \DIFaddbegin \DIFadd{It enables rapid, configurable, and reproducible inference of microbial networks and furthers the formulation of hypotheses about the role of these interactions on community composition and stability.
  These comparative analyses will require coupled network analysis and visualization tools (such as }\ac{mind}\DIFadd{~\mbox{%DIFAUXCMD
\cite{huResourceComparisonIntegration2022}}\hskip0pt%DIFAUXCMD
) and need systematic access to datasets, shared in accordance with FAIR standards~\mbox{%DIFAUXCMD
\cite{pachecoFAIRRepresentationsMicrobial2022}}\hskip0pt%DIFAUXCMD
}\DIFaddend .


%DIF <  TODO: Uses in MIND (?)
\DIFdelbegin %DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend % METHODS
%!TEX root = ../main.tex

\section*{Materials and Methods}

  \subsection*{\DIFdelbegin \DIFdel{Datasets}\DIFdelend \DIFaddbegin \DIFadd{16S rRNA sequencing datasets}\DIFaddend }
  \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD <   %%%
\DIFdelend \vspace{-5mm}
  \DIFdelbegin \DIFdel{The study uses three kinds }\DIFdelend \DIFaddbegin \DIFadd{This study utilized two types }\DIFaddend of 16S rRNA sequencing datasets: \DIFdelbegin \DIFdel{real datasets , mock datasets and }\DIFdelend \DIFaddbegin \DIFadd{biological datasets and mock/}\DIFaddend synthetic datasets.
  \DIFdelbegin \DIFdel{Real }\DIFdelend \DIFaddbegin \DIFadd{Biological }\DIFaddend datasets are collections of sequencing reads obtained from naturally occurring microbial community samples.
  The current \DIFdelbegin \DIFdel{study used healthy }\DIFdelend \DIFaddbegin \DIFadd{analysis used }\DIFaddend stool samples from a fecal microbiome transplant study \DIFdelbegin \DIFdel{~\mbox{%DIFAUXCMD
\cite{Kang2017} }\hskip0pt%DIFAUXCMD
and healthy saliva samples from a periodontal disease study ~\mbox{%DIFAUXCMD
\cite{Chen2018} }\hskip0pt%DIFAUXCMD
as real datasets for analysis.
  }\DIFdelend \DIFaddbegin \DIFadd{of autism~\mbox{%DIFAUXCMD
\cite{Kang2017} }\hskip0pt%DIFAUXCMD
as the biological dataset.
  This dataset was chosen because the sequences were easily accessible on Qiita~\mbox{%DIFAUXCMD
\cite{qiita} }\hskip0pt%DIFAUXCMD
and optimally pre-processed according to the }\ac{emp}\DIFadd{~\mbox{%DIFAUXCMD
\cite{Thompson2017} }\hskip0pt%DIFAUXCMD
protocol, allowing them to be used directly as input to the }\ac{micone} \DIFadd{pipeline.
  The study was composed of multiple sequencing runs.
  The runs that contained paired-end reads (run 2 (10M reads), run 3 (750K reads) and run 4 (16M reads)), were downloaded from Qiita~\mbox{%DIFAUXCMD
\cite{qiita} }\hskip0pt%DIFAUXCMD
(study ID 10532) and used as input sequences for the }\ac{micone} \DIFadd{pipeline.
  Sequences from both control (212 samples including neurotypical and donors) and autism spectrum disorder (ASD) (126 samples) patients were included in the analyses.
  All the network analyses in the study, unless explicitly mentioned, were performed on the healthy and ASD samples in the fecal microbiome transplant study.
  }\DIFaddend The mock community 16S datasets are \DIFdelbegin \DIFdel{real }\DIFdelend \DIFaddbegin \DIFadd{experimental }\DIFaddend sequencing data obtained for artificially assembled collections of \DIFaddbegin \DIFadd{DNA of }\DIFaddend species in known proportions.
  The mock datasets used for this study, obtained from mockrobiota~\cite{Bokulich2016}, are \DIFdelbegin \DIFdel{labelled }\DIFdelend \DIFaddbegin \DIFadd{labeled }\DIFaddend mock4, mock12\DIFaddbegin \DIFadd{, }\DIFaddend and mock16.
  The mock4 community is composed of 21 bacterial strains.
  Two replicate samples from mock4 contain all species in equal abundances, and two additional replicate samples contain the same species in unequal abundances.
  The mock12 community is composed of 27 bacterial strains that include closely related taxa with some pairs having only one to two nucleotide \DIFdelbegin \DIFdel{difference from }\DIFdelend \DIFaddbegin \DIFadd{differences from one }\DIFaddend another.
  The mock16 community is composed of 49 bacteria and 10 \DIFdelbegin \DIFdel{archea}\DIFdelend \DIFaddbegin \DIFadd{Archaea}\DIFaddend , all represented in equal \DIFdelbegin \DIFdel{amount.
The synthetic datasets were generated using an artificial read simulator called ART~\mbox{%DIFAUXCMD
\cite{Huang2012}}\hskip0pt%DIFAUXCMD
.
  Three different microbial composition profiles were used as input; reads were generated using a soil and water microbiome composition profiles from the }%DIFDELCMD < \ac{emp}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Thompson2017} }\hskip0pt%DIFAUXCMD
and healthy gut microbiome project from the fecal microbiome transplant study~\mbox{%DIFAUXCMD
\cite{Kang2017}}\hskip0pt%DIFAUXCMD
.
  The reads are simulated using the NCBI RefSeq database as the reference sequence pool and the "art\_illumina" sequence profile with a mutation rate of 2\%.
  The scripts used to generate the synthetic data are in the scripts folder of the repository (}\href{https://github.com/segrelab/MiCoNE-pipeline-paper}{\DIFdel{https://github.com/segrelab/MiCoNE-pipeline-paper}}%DIFAUXCMD
\DIFdel{).
}\DIFdelend \DIFaddbegin \DIFadd{amounts.
}\DIFaddend 

  \subsection*{\ac{micone}}
  \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD <   %%%
\DIFdelend \vspace{-5mm}
  The flowchart describing the workflow of \ac{micone} (\acl{micone}), our complete 16S data-analysis pipeline, is shown in Figure \ref{fig:figure1}.
  The pipeline integrates many publicly available tools as well as custom R or Python modules and scripts to extract co-occurrence associations from 16S sequence data.
  Each of these tools corresponds to a distinct \DIFdelbegin \DIFdel{R or python }\DIFdelend module that recapitulates the relevant analyses.
  All such individual modules are available as part of the \ac{micone} package.
  The inputs to the pipeline by default are \DIFdelbegin \DIFdel{the raw community }\DIFdelend \DIFaddbegin \DIFadd{raw untrimmed }\DIFaddend 16S rRNA sequence reads, but the software can be alternatively configured to use trimmed sequences, \ac{otu} tables and other types of intermediate data \DIFdelbegin \DIFdel{.
  The }\DIFdelend \DIFaddbegin \DIFadd{(see documentation).
  The configuration and modular nature of the }\ac{micone} \DIFadd{package enables users to start and end the pipeline at any point in the workflow, and to run parts of the pipeline in isolation.
  The pipeline supports both paired-end and single-end reads, and additionally supports independently processing reads from multiple runs and merging the OTU tables in the DC step.
  The }\DIFaddend final output of the pipeline is the inferred network of co-occurrence relationships among the microbes present in the samples.

  The \ac{micone} pipeline provides both a Python API \DIFdelbegin \DIFdel{as well as }\DIFdelend \DIFaddbegin \DIFadd{together with }\DIFaddend a command-line interface and only \DIFdelbegin \DIFdel{requires }\DIFdelend \DIFaddbegin \DIFadd{uses }\DIFaddend a single configuration file \DIFdelbegin \DIFdel{.
  The configuration file }\DIFdelend \DIFaddbegin \DIFadd{(}\texttt{\DIFadd{nextflow.config}}\DIFadd{) to encode the configuration parameters.
  The }\ac{micone} \DIFadd{Python API provides several }\ac{otu} \DIFadd{table and network-related functions and methods, enabling detailed comparison of counts tables and inferred networks if desired.
  Exploring the effects of these combinations of methods on the resultant networks is difficult and inconvenient since different tools differ in their input and output formats and require interconversions between the various formats.
  The pipeline facilitates this comparative exploration by providing a variety of modules for interconversion between various formats, and by allowing for easy incorporation of new tools as modules.
  It also contains helper functions that can help in parsing taxonomies and communicate with the }\ac{ncbi} \DIFadd{taxonomy database to query taxonomy by name or taxonomic IDs.
  The configuration file along with the run file (}\texttt{\DIFadd{main.nf}}\DIFadd{) }\DIFaddend lists the inputs, output\DIFaddbegin \DIFadd{, }\DIFaddend and the steps to be performed during runtime, along with the parameters to be used (if different from defaults) for the various steps.
  \DIFaddbegin \DIFadd{The default settings of the pipeline are shown in Table~\ref{tab:micone_tools} (with default parameter values shown in Tables S2-S6 at }\href{https://github.com/segrelab/MiCoNE-pipeline-paper}{\DIFadd{https://github.com/segrelab/MiCoNE-pipeline-paper}}\DIFadd{).
  }\DIFaddend Since the entire pipeline \DIFdelbegin \DIFdel{run-through }\DIFdelend \DIFaddbegin \DIFadd{run }\DIFaddend is stored in the form of a text file (the configuration file), subsequent runs are highly reproducible and changes can be easily tracked using version control.
  \DIFdelbegin \DIFdel{It uses }\DIFdelend \DIFaddbegin \DIFadd{The pipeline makes use of }\DIFaddend the nextflow workflow manager~\cite{Tommaso2015} under the hood, making it readily usable on \DIFdelbegin \DIFdel{local machines, cluster}\DIFdelend \DIFaddbegin \DIFadd{the local machine, cluster, }\DIFaddend or cloud with minimal configuration change.
  It also allows for automatic parallelization of all possible processes, both within and across samples.
  The pipeline is designed to be modular: each tool or method is organized into modules \DIFdelbegin \DIFdel{which }\DIFdelend \DIFaddbegin \DIFadd{that }\DIFaddend can be easily modified or replaced.
  This modular architecture simplifies the process of adding new tools (refer to \DIFaddbegin \DIFadd{the }\DIFaddend modules section in the \ac{micone} documentation).
  %DIF <    In addition to the Python package, the entire pipeline has been containerized into a Docker~\cite{Merkel1994} image (\hl{dockerhub link}) for easy deployment and setup.
  The main components of the pipeline are detailed in the subsequent sections.

  \subsection*{\DIFdelbegin \DIFdel{Denoising and Clustering }\DIFdelend \DIFaddbegin \DIFadd{Sequence Processing }\DIFaddend (\DIFdelbegin \DIFdel{DC}\DIFdelend \DIFaddbegin \DIFadd{SP}\DIFaddend )}
  \vspace{-5mm}
  This module deals with processing the raw \DIFaddbegin \DIFadd{multiplexed }\DIFaddend 16S sequence data into \DIFdelbegin %DIFDELCMD < \ac{otu} %%%
\DIFdel{or }%DIFDELCMD < \ac{esv} %%%
\DIFdel{count tables}\DIFdelend \DIFaddbegin \DIFadd{demultiplexed, quality-controlled, trimmed sequences}\DIFaddend .
  It consists of the \DIFdelbegin \DIFdel{following processes: quality control, denoising (or clustering) and chimera checking.
  The quality control process handles the }\DIFdelend demultiplexing and \DIFaddbegin \DIFadd{trimming processes.
  The demultiplexing process deals with separating the multiplexed sequences into individual samples based on barcodes.
  The trimming process handles the }\DIFaddend quality control steps such as trimming adapters and \DIFdelbegin \DIFdel{trimming }\DIFdelend low-quality nucleotide stretches from the sequences.
  The \DIFaddbegin \DIFadd{parameters and tools in this process are fixed and are not available for user customization.
  The various tools used for the processes were adapted from }\ac{qiime2} \DIFadd{v2021.8.0~\mbox{%DIFAUXCMD
\cite{bolyenReproducibleInteractiveScalable2019}}\hskip0pt%DIFAUXCMD
.
  The list of tools used in this step, along with their modules and references are provided in Table~\ref{tab:micone_tools}.
}

  \subsection*{\DIFadd{Denoising and Clustering (DC)}}
  \vspace{-5mm}
  \DIFadd{This module deals with processing the quality-controlled, trimmed 16S sequence data into }\ac{otu} \DIFadd{or }\ac{esv} \DIFadd{count tables.
  It consists of the following processes: denoising (or clustering) and chimera checking.
  The }\DIFaddend denoise/cluster process handles the conversion of the demultiplexed, trimmed sequences into \ac{otu} or \ac{esv} count tables (some methods, like closed reference and open reference clustering, \DIFdelbegin \DIFdel{perform clustering and taxonomy assignment in the same step}\DIFdelend \DIFaddbegin \DIFadd{make use of a taxonomy reference database for clustering}\DIFaddend ).
  The chimera checking process handles the removal of chimeric sequences created during the \ac{pcr} step.
  The output of this module is a matrix of counts, that describes the number of reads of a particular \ac{otu} or \ac{esv} (rows of the matrix) present in each sample (columns of the matrix).
  The options currently available in the pipeline for denoising and clustering are: open reference clustering, closed reference clustering and de novo clustering methods from \DIFdelbegin %DIFDELCMD < \ac{qiime1} %%%
\DIFdel{v1.9.1~\mbox{%DIFAUXCMD
\cite{Caporaso2010} }\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{the vsearch plugin of }\ac{qiime2} \DIFadd{v2021.8.0~\mbox{%DIFAUXCMD
\cite{bolyenReproducibleInteractiveScalable2019} }\hskip0pt%DIFAUXCMD
}\DIFaddend and denoising methods from \ac{dada2} v1.14~\cite{Callahan2016} \DIFaddbegin \DIFadd{(from the }\ac{dada2} \DIFadd{R package) }\DIFaddend and Deblur v1.1.0~\cite{Amir2017} \DIFaddbegin \DIFadd{(from the deblur plugin of }\ac{qiime2}\DIFadd{)}\DIFaddend .
  The quality filtering and chimera checking tools are derived from those used in \ac{qiime2} \DIFdelbegin \DIFdel{v2019.10}\DIFdelend \DIFaddbegin \DIFadd{v2021.8}\DIFaddend .0 \DIFaddbegin \DIFadd{(uchime-denovo method) }\DIFaddend and \ac{dada2} \DIFdelbegin \DIFdel{. 
}\DIFdelend \DIFaddbegin \DIFadd{(remove bimera method).
  The list of tools used in this step, along with their modules and references are provided in Table~\ref{tab:micone_tools}.
}\DIFaddend 

  \DIFaddbegin \DIFadd{For the UniFrac analysis in Figure~\ref{fig:figure2}, we had set a count threshold of 10, such that if the count of the representative sequences in a particular sample is less than the threshold, it is omitted from the analysis.
  Additionally, for Figure~\ref{fig:figure2}C and \ref{fig:figure2}D, the expected sequences from the mock communities were trimmed to the V4 region before being subject to UniFrac analyses.
}

  \DIFaddend \subsection*{Taxonomy Assignment (TA)}
  \vspace{-5mm}
  This module deals with assigning taxonomies to \DIFdelbegin \DIFdel{either }\DIFdelend the representative sequences \DIFdelbegin \DIFdel{of the }\DIFdelend \DIFaddbegin \DIFadd{(}\DIFaddend \ac{otu}s or \DIFdelbegin \DIFdel{directly to the }\DIFdelend \ac{esv}s\DIFaddbegin \DIFadd{)}\DIFaddend .
  In order to assign taxonomies to a particular sequence\DIFdelbegin \DIFdel{we need }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend a taxonomy database and a query tool \DIFaddbegin \DIFadd{are necessary}\DIFaddend .
  The taxonomy database contains \DIFdelbegin \DIFdel{the }\DIFdelend \DIFaddbegin \DIFadd{a }\DIFaddend collection of 16S sequences of \DIFdelbegin \DIFdel{micro-organisms of interest }\DIFdelend \DIFaddbegin \DIFadd{microorganisms }\DIFaddend and the query tool allows one to compare a sequence of interest to all the sequences in the database to identify the best matches.
  Finally, a consensus method is used to identify the most probable match from the list of best matches.
  The pipeline incorporates \ac{gg} 13\_8~\cite{DeSantis2006} \DIFdelbegin \DIFdel{, SILVA 132~\mbox{%DIFAUXCMD
\cite{Quast2012} }\hskip0pt%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{(99\% identity), SILVA 138~\mbox{%DIFAUXCMD
\cite{Quast2012} }\hskip0pt%DIFAUXCMD
(99\% identity) }\DIFaddend and the \ac{ncbi} (16S RefSeq as of Oct \DIFdelbegin \DIFdel{2019}\DIFdelend \DIFaddbegin \DIFadd{2021}\DIFaddend )~\cite{Sayers2009} databases for taxonomy assignment\DIFaddbegin \DIFadd{.
  SILVA and }\ac{gg} \DIFadd{are two popular 16S databases used for taxonomy identification and the }\ac{ncbi} \DIFadd{RefSeq nucleotide database contains 16S rRNA sequences as a part of two BioProjects - 33175 }\DIFaddend and \DIFdelbegin \DIFdel{the }\DIFdelend \DIFaddbegin \DIFadd{33317.
  The three databases vastly differ in terms of their last update status - }\ac{gg} \DIFadd{was last updated on May 2013, SILVA was last updated on August 2020 at the time of writing and }\ac{ncbi} \DIFadd{is updated regularly as new sequences are curated.
  These databases were downloaded and built using the RESCRIPt }\ac{qiime2} \DIFadd{plugin~\mbox{%DIFAUXCMD
\cite{iiRESCRIPtReproducibleSequence2021}}\hskip0pt%DIFAUXCMD
.
  The }\DIFaddend Naive Bayes classifier \DIFdelbegin \DIFdel{from }%DIFDELCMD < \ac{qiime2} %%%
\DIFdel{and }\DIFdelend \DIFaddbegin \DIFadd{and the }\DIFaddend \ac{ncbi} blast \DIFaddbegin \DIFadd{used }\DIFaddend as the query tools \DIFdelbegin \DIFdel{(from }\DIFdelend \DIFaddbegin \DIFadd{in this study were from the }\DIFaddend \ac{qiime2} \DIFdelbegin \DIFdel{)}\DIFdelend \DIFaddbegin \DIFadd{package and the parameters used were the defaults of the package}\DIFaddend .
  The consensus algorithm used is the default method used by the classifiers in \ac{qiime2}.
  \DIFaddbegin \DIFadd{During the assignment, a representative sequence might be assigned an "unknown" Genus for one of two reasons: the first is if the taxonomy identifier associated with the sequence in the database did not contain a given Genus; the second, more likely reason, is that the database contains multiple sequences that are very similar to the query (representative) sequence and the consensus algorithm (from }\ac{qiime2}\DIFadd{) is unable to assign one particular Genus at the required confidence.
  The assignments in SILVA were originally substantially different from the other two databases ($40\%$ mismatch) even at the Phylum level.
  However, this was corrected via minor adjustments to the taxonomic names, such as changing Bacteroidota to Bacteroidetes in the SILVA Phylum assignments.
  The full list of changes can be found in }\texttt{\DIFadd{figure4ab\_data.py}} \DIFadd{in the data and scripts  repository.
  The list of tools used in this step, along with their modules and references are provided in Table~\ref{tab:micone_tools}.
}\DIFaddend 

  %DIF <  TODO: Add references and basic equations or details
  \subsection*{OTU and ESV Processing (OP)}
  \vspace{-5mm}
  This module deals with normalization, filtering\DIFaddbegin \DIFadd{, forking, grouping, }\DIFaddend and applying transformations to the \ac{otu} or \ac{esv} counts matrix.
  \DIFdelbegin \DIFdel{Rarefaction }\DIFdelend \DIFaddbegin \DIFadd{Normalization of the count matrix involves converting the count matrix of read counts into a count matrix containing relative abundances.
  The module also supports rarefaction, which }\DIFaddend is a normalization technique used to overcome the bias that might arise due to variable sampling depth in different samples.
  This is performed either by sub-sampling or by normalization of the matrix to the lowest sampling depth\DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Weiss2015}}\hskip0pt%DIFAUXCMD
.
  Rarefaction is usually followed by filtering, which }\DIFdelend \DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{Weiss2015}}\hskip0pt%DIFAUXCMD
.
  However, due to the potential biases and false positives~\mbox{%DIFAUXCMD
\cite{mcmurdieWasteNotWant2014,chaoCoveragebasedRarefactionExtrapolation2012} }\hskip0pt%DIFAUXCMD
that might arise during the process, the rarefaction module is disabled by default and can be enabled in the configuration if needed.
  Hence, although the pipeline supports normalization and rarefaction, these modules are turned off by default, and the analyses reported in the paper do not normalize or rareify the counts matrices.
  However, most of the network inference methods perform normalization and other transformation operations on the counts matrix as a part of their workflow.
  Filtering, }\DIFaddend is performed to remove samples or features (\ac{otu}s or \ac{esv}s) from the \DIFdelbegin \DIFdel{count }\DIFdelend \DIFaddbegin \DIFadd{counts }\DIFaddend matrix that are sparse.
  \DIFdelbegin \DIFdel{In order to determine the filtering threshold we fix the number of samples and correlation detection power needed and determine the number of features to be used}\DIFdelend \DIFaddbegin \DIFadd{By default, when the OP module is ``on'', the samples are filtered out if the total reads in a sample are less than 500 and features are filtered out if the relative abundance is less than 1\%, prevalence (percentage of samples containing feature) is less than 5\% and count sum across all the samples is less than 100.
  When the OP module is ``off'', the filtering is still performed but threshold parameters are much more relaxed.
  The parameters used are given in Table~\ref{tab:micone_tools}.
  The forking operation splits the count matrix into multiple matrices based on sample metadata column, this is useful for example to compare case vs. control.
  The group operation transforms the }\ac{otu} \DIFadd{or }\ac{esv} \DIFadd{count matrix into a taxonomic count matrix at the requested level by adding up counts that map to the same taxonomy and is carried out at the end of the OP step}\DIFaddend .
  Finally, transformations are performed in order to correct for and overcome the compositional bias that is inherent in \DIFdelbegin \DIFdel{a }\DIFdelend \DIFaddbegin \DIFadd{the }\DIFaddend counts matrix (in \DIFdelbegin \DIFdel{most cases this is }\DIFdelend \DIFaddbegin \DIFadd{the analysis performed in the study these were disabled and directly }\DIFaddend handled by the network inference algorithm).
  \DIFaddbegin \DIFadd{All the modules in this step were implemented using functions from the }\href{http://biom-format.org/}{\DIFadd{biom-format}} \DIFadd{Python package~\mbox{%DIFAUXCMD
\cite{mcdonaldBiologicalObservationMatrix2012}}\hskip0pt%DIFAUXCMD
.
}\DIFaddend 

  \subsection*{Network Inference (NI)}
  \vspace{-5mm}
  This module deals with the inference of co-occurrence associations from the \DIFaddbegin \DIFadd{processed }\DIFaddend \ac{otu} or \ac{esv} counts matrix.
  \DIFdelbegin \DIFdel{These }\DIFdelend \DIFaddbegin \DIFadd{The input count matrices are collapsed to the Genus level (or any other required taxonomy level) using the group module at the OP step.
  These collapsed matrices are used as input to the network inference methods to produce association matrices at the appropriate taxonomy level.
  These }\DIFaddend associations can be represented as a network, with nodes representing \DIFaddbegin \DIFadd{the }\DIFaddend taxonomies of the \DIFdelbegin \DIFdel{micro-organisms }\DIFdelend \DIFaddbegin \DIFadd{microorganisms }\DIFaddend and edges representing the \DIFdelbegin \DIFdel{association }\DIFdelend \DIFaddbegin \DIFadd{associations }\DIFaddend between them.
\DIFaddbegin 

  \DIFadd{The pipeline includes 4 methods for pairwise correlation metrics, and 6 methods for direct association metrics (refer to Table~\ref{tab:micone_tools}).
  Pairwise correlation methods involve the calculation of the correlation coefficient between each pair of nodes (taxonomic entity like Genera) leading to the inclusion of spurious indirect connections.
  On the other hand, direct association methods use conditional independence to avoid the detection of correlated but indirectly connected taxonomic entities~\mbox{%DIFAUXCMD
\cite{Kurtz2015,Menon2018}}\hskip0pt%DIFAUXCMD
.
  }\DIFaddend A null model is created by re-sampling and bootstrapping the \DIFdelbegin \DIFdel{correlation/interaction matrix and is }\DIFdelend \DIFaddbegin \DIFadd{counts matrix and recalculating the correlations (see next section for details on network analysis and statistics).
  These bootstrapped association matrices are }\DIFaddend used to calculate the significance of the inferred \DIFdelbegin \DIFdel{associations }\DIFdelend \DIFaddbegin \DIFadd{correlations }\DIFaddend by calculating the p-values against this null model\DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{Watts2018}}\hskip0pt%DIFAUXCMD
.
  The pipeline includes Pearson, Spearman and FastSpar v0.0.10 (}\DIFdelend \DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{Watts2018}}\hskip0pt%DIFAUXCMD
.
  Brown's p-value merging method~\mbox{%DIFAUXCMD
\cite{brown_400_1975} }\hskip0pt%DIFAUXCMD
is used for combining p-values from the pairwise correlations methods to obtain a consensus p-value, which can be used to filter for significance.
  The bootstrapping and p-value calculations are only performed on the correlations-based methods.
  In the final module of this step, the consensus algorithms are used to create the final consensus network using associations from all the network inference methods (except Pearson and Spearman, by default).
  The outputs of this step are co-occurrence association networks encoded in the JSON format (refer to Supplementary section) and which can also be exportable to a variety of network formats.
  The list of tools used in this step, along with their modules and references are provided in Table~\ref{tab:micone_tools}.
}

  \subsection*{\DIFadd{Consensus network and p-value merging}}
  \DIFadd{The consensus methods combine networks inferred from both correlation-based and direct association methods.
  First, for the correlation-based methods, we calculate p-values using null models and then merge the p-values using Brown's p-value merging method~\mbox{%DIFAUXCMD
\cite{Poole_Gibbs_Shmulevich_Bernard_Knijnenburg_2016,faustCoNetAppInference2016} }\hskip0pt%DIFAUXCMD
(see Methods).
  Second, we filter all the inferred networks based on an association strength threshold of 0.1 and }\DIFaddend a \DIFdelbegin \DIFdel{faster implementation of }%DIFDELCMD < \ac{sparcc}%%%
\DIFdel{)~\mbox{%DIFAUXCMD
\cite{Watts2018} }\hskip0pt%DIFAUXCMD
as the pairwise correlation metrics, and }%DIFDELCMD < \ac{spieceasi} %%%
\DIFdel{v1.0.7~\mbox{%DIFAUXCMD
\cite{Kurtz2015}}\hskip0pt%DIFAUXCMD
, }%DIFDELCMD < \ac{mldm} %%%
\DIFdel{v1.1~\mbox{%DIFAUXCMD
\cite{Yang2017} }\hskip0pt%DIFAUXCMD
and }%DIFDELCMD < \ac{magma}%%%
\DIFdel{~\mbox{%DIFAUXCMD
\cite{Cougoul2019} }\hskip0pt%DIFAUXCMD
as the direct association metrics.
  The empirical Browns method~\mbox{%DIFAUXCMD
\cite{Poole} }\hskip0pt%DIFAUXCMD
is used for combining }\DIFdelend \DIFaddbegin \DIFadd{p-value cutoff of 0.05.
  Finally, we apply the consensus algorithms we have developed on these filtered networks.
  These steps are elaborated on in the subsequent sections.
}

 \subsubsection*{\DIFadd{Notation}}
  \vspace{-5mm}
 \DIFadd{This section defines the notation used below to describe the consensus network algorithm used in the }\ac{micone} \DIFadd{pipeline.
 Note that all networks to be compared were updated to have the same number of nodes.
}

  \DIFadd{$w$, the number of co-occurrence networks to be integrated into the consensus network (by default, is equal to the total number of network inference methods excluding Spearman and Pearson, 8)
}

  \DIFadd{$q$, the number of unique nodes across all $w$ co-occurrence networks
}

  \DIFadd{$N^i$, the matrix of edge weights for the $i^{th}$ co-occurrence network.
  This is a $q \times q$ matrix, where $i \in \{1,\dots,w\}$.
  $N^i_{a,b}$ represents edge $(a,b)$ in network $i$
}

  \DIFadd{$P^i$, the matrix of }\DIFaddend p-values \DIFdelbegin \DIFdel{from the various methodsto obtain a consensus p-value, which is used to create the consensus network}\DIFdelend \DIFaddbegin \DIFadd{for all edges of the $i^{th}$ co-occurrence network.
  This is a $q \times q$ matrix, where $i \in \{1,\dots,w\}$
}

  \DIFadd{$\bar{N}^i$, the ``flattened'' version of the adjacency matrix $N^i$ into a $q^2 \times 1$ column vector, where all columns are stacked onto each other into a $q^2$ long vector.
  Element $\bar{N}^i_j$ corresponds to the $j^{th}$ edge in the $i^{th}$ network.
}

  \DIFadd{$\bar{P}^i$, the ``flattened'' version of the adjacency matrix $P^i$ into a $q^2 \times 1$ column vector, where all columns are stacked onto each other into a $q^2$ long vector.
}

  \subsubsection*{\DIFadd{Bootstrapping and p-value calculation}}
  \vspace{-5mm}
  \DIFadd{For all correlation-based methods $k \le w$, $1000$ permutations of the original }\ac{otu} \DIFadd{counts data were generated~\mbox{%DIFAUXCMD
\cite{Watts2018}}\hskip0pt%DIFAUXCMD
.
  The correlations in the permuted }\ac{otu} \DIFadd{tables are recalculated using the different correlation-based algorithms.
  Finally, the p-value is determined based on how often a more extreme association is observed for randomly permuted data.
  Note that, all the direct association-based methods used in the study have their own regularization methods built in and hence do not need to undergo bootstrapping.
}

  \subsubsection*{\DIFadd{p-value merging}}
  \vspace{-5mm}
  \DIFadd{The next step in the consensus algorithm workflow is to merge the p-values for the networks generated by the correlation-based methods.
  This step is performed using the Brown's p-value merging method~\mbox{%DIFAUXCMD
\cite{Poole_Gibbs_Shmulevich_Bernard_Knijnenburg_2016,faustCoNetAppInference2016}}\hskip0pt%DIFAUXCMD
.
}

  \DIFadd{As described in more detail in the Supplementary section and in the original reference~\mbox{%DIFAUXCMD
\cite{Poole_Gibbs_Shmulevich_Bernard_Knijnenburg_2016}}\hskip0pt%DIFAUXCMD
, the final combined p-value is given by:
  }\begin{equation}
    \DIFadd{\begin{aligned}
        & \hat{P}_j = 1.0 - \Phi_{2f}\left( \psi / c \right) \\
        \text{where},~ &\psi = -2 \sum_{i=1}^k \log(\bar{P}^i_j) ~~~\text{and}~~~ \Phi_{2f} = \mathrm{CDF}\left( \chi^2_{2f} \right)
    \end{aligned}
    \label{eqn:pvalue-combined}
  }\end{equation}
  \DIFadd{where, $\hat{P}_j$ is the combined p-value for the edge $j$, $f$ is the number of degrees of freedom, and $c$ is a scale factor.
}

  \DIFadd{Note that we do not use Pearson and Spearman methods in the p-value merging step to determine the consensus network.
  These methods are only used for demonstration and comparison.
  The combined p-values are used to threshold for significance right before the consensus algorithm is applied to the inferred networks.
}

  \subsubsection*{\DIFadd{Consensus methods}}
  \vspace{-5mm}
  \DIFadd{The consensus algorithm was designed to increase the precision (number of true positives) at the end of the network inference step.
  For this purpose, we developed two simple algorithms that combine the edges reported by the different network inference tools.
  Both the algorithms make use of a user-defined parameter $p$ ($0 \leq p \leq 1$), in order to threshold the edges from the individual methods.
  The inputs to both the algorithms are the co-occurrence networks (association matrices) $\bar{N}^i$ (flattened version of $N^{i}$) generated by each method $i$, and the threshold parameter $p$.
  Here, the $\bar{N}^{i}$ each have the same set of nodes $q$ and only differ by the value of the association inferred between every pair.
  Networks that do not have a particular node, are updated such that the node is added as an isolated component.
  In this manner, $\bar{N}^{i}_j$ represents edge $j$ in network $i$}\DIFaddend .

    \DIFaddbegin \DIFadd{Note that the consensus method is only used to filter relevant interactions.
    If a given pair of nodes is inferred to have edges that satisfy the consensus requirements, all corresponding edges from the $w$ networks will be returned by the algorithm, as a multigraph.
    Based on this approach, }\ac{micone} \DIFadd{reports as the default output, the consensus network where each edge is annotated with weights (correlations for the correlation-based methods and direct associations for the other methods) from all the methods used in the consensus algorithm.
}

  \paragraph*{\DIFadd{Algorithm 1 - Simple voting:}}
  \DIFadd{The simple voting method performs a voting-based consensus to determine whether an edge will exist between a given node-pair in the final consensus network~\mbox{%DIFAUXCMD
\cite{bustinceFuzzySetsTheir2008,tsarevApplicationMajorityVoting2018}}\hskip0pt%DIFAUXCMD
.
  For each pair of nodes, we determine the number of network inference methods that report an edge $j$ between them, i.e. $\bar{N}^{i}_{j}, \forall i \in \{1,\dots,w\}$.
  Each node-pair will have an edge in the final consensus network if the number of reported edges is larger than the threshold (Equation~\ref{eqn:simple-voting2}).
}

 \DIFadd{The number of reported edges is computed as follows:
}

 \DIFadd{For each edge $j$, we obtain $M_j$ which represents the number of networks in which edge $j$ is reported.
 Formally, $M_j$ is calculated as the following function:
  }\begin{equation}
      \DIFadd{M_j = f(g(\bar{N}^{i=1}_j), \dots, g(\bar{N}^{i=w}_j)) }\\
    \DIFadd{\label{eq:simple-voting}
  }\end{equation}

\DIFadd{where, $g$ and $f$ are defined as follows:
}

  \begin{equation*}
    \DIFadd{g(x) =
    \begin{cases}
       & 0, \text{ if } x=0, \\
       & -1, \text{ if } x<0, \\
       & 1, \text{ if } x>0
    \end{cases}
  }\end{equation*}
  \DIFadd{and
  }\begin{equation*}
    \DIFadd{f(x_1,\dots,x_w) = max \left( \#(i \mid x_i=-1),\#(i \mid x_i=1)
       \right)
  }\end{equation*}
  \DIFadd{where, $\#$ refers to the cardinality of the set.
}

 \DIFadd{The edge $j$ is selected to be present in the final consensus network if the number of networks in which $j$ appears is greater than a threshold, i.e:
  }\begin{equation}
     \DIFadd{M_j \geq \lfloor p \times w \rfloor
    \label{eqn:simple-voting2}
  }\end{equation}
  \DIFadd{where, $p$ is the user-defined threshold parameter.
}

  \DIFadd{The simple voting method returns the union of the networks when $0 \leq p \leq \frac{1}{w}$ and will return the intersection when $\frac{(w - 1)}{w} \leq p \leq 1$.
  In general, if $\frac{(n - 1)}{w} \leq p \leq \frac{n}{w}$, this algorithm will report an edge in the consensus network when at least $n$ network inference methods report this edge.
}


  \paragraph*{\DIFadd{Algorithm 2 - Scaled-sum method:}}

  \DIFadd{This algorithm generates a consensus network based on the sum of all edges (weights of associations) reported between a pair of nodes~\mbox{%DIFAUXCMD
\cite{bustinceFuzzySetsTheir2008,tsarevApplicationMajorityVoting2018}}\hskip0pt%DIFAUXCMD
.
  Since in generating a consensus network using this method we sum the edges reported by direct association methods with those from correlation-based methods, summing of the edges is preceded by a pre-processing step, in which all networks are re-scaled.
}

  \DIFadd{First, the network generated by each network inference method ($\bar{N}^i$) is re-scaled into a normalized version ($\bar{S}^i$), as follows:
  }\begin{equation}
    \DIFadd{\bar{S}^{i} = \frac{\bar{N}^{i}}{\max(\mid \bar{N}^{i} \mid)},~~\forall i \in }{\DIFadd{1, \dots, w}}
    \DIFadd{\label{eqn:scaled-sum-rescaling}
  }\end{equation}

    \DIFadd{In this way, it is guaranteed that $\max(\mid \bar{S}^i \mid) = 1$.
}


  \DIFadd{Next, for each edge $j$, we sum the weights of all reported edges from the different networks.
  }\begin{equation}
    \DIFadd{s_j = \sum_{i=1}^{w} \bar{S}^i_j
    \label{eqn:scaled-sum}
  }\end{equation}


  \DIFadd{An edge $j$ will be included in the consensus network if $s_j$ passes a threshold.
  }\begin{equation}
    \DIFadd{\mid s_j \mid > (w - 1) \times p
    \label{eq:scaled-sum2}
  }\end{equation}

  \DIFadd{The advantage of this method over the simple voting method is that it also takes into account the strength of the association reported for that particular node in the inferred networks.
}

  \DIFaddend \subsection*{Network \DIFdelbegin \DIFdel{Variability}\DIFdelend \DIFaddbegin \DIFadd{variability}\DIFaddend }
\DIFaddbegin 

  \subsubsection*{\DIFadd{Notation}}
  \DIFaddend \vspace{-5mm}
  \DIFaddbegin \DIFadd{This section defines the notation used for the network variability analysis performed for Figure~\ref{fig:figure6}.
}

  \DIFadd{$W$, the number of co-occurrence networks generated from all possible combinations of tools and parameters in the workflow.
  Note that this is different from $w$, which counted only the different network inference modules.
}

  \DIFadd{$Q$, the number of unique nodes across all $W$ networks.
}

  \DIFadd{$N^i$, the edge weights of the $i^{th}$ co-occurrence network represented as a $Q \times Q$ adjacency matrix, where $i \in {1, \dots, W}$.
  $N^i_{a,b}$ represents the edge $(a,b)$ in network $i$
}

  \DIFadd{$\bar{N}^i$, the ``flattened'' version of the adjacency matrix $N^i$ into a $Q^2 \times 1$ column vector, where all columns are stacked onto each other into a $Q^2$ long vector.
}


  \subsubsection*{\DIFadd{Principal Component Analysis and variability calculation}}
  \vspace{-5mm}
   \DIFaddend In order to compare across different networks \DIFdelbegin \DIFdel{, and analyze }\DIFdelend \DIFaddbegin \DIFadd{and calculate }\DIFaddend the degree of variability induced by the choice of different modules\DIFdelbegin \DIFdel{and parameters}\DIFdelend , we organized multiple networks into a single mathematical structure that we could use for linear regression.
   \DIFdelbegin \DIFdel{In particular, we transformed the adjacency matrix of each }\DIFdelend \DIFaddbegin \DIFadd{First, we obtained the }\DIFaddend co-occurrence network \DIFdelbegin \DIFdel{into a vector}\DIFdelend \DIFaddbegin \DIFadd{$\bar{N}^i$ for each of the $W$ possible tool and parameter combinations in the workflow}\DIFaddend .
   We then \DIFdelbegin \DIFdel{merged the networks generated from all possible combinations of tools into a table (N, see below) in which each column represents one network.
   }%DIFDELCMD < 

%DIFDELCMD <   %%%
\begin{displaymath}
    \DIFdel{N = \begin{bmatrix}
     edge_{1,1} & edge_{2,1} & \cdots  & edge_{n, 1} \\ 
     edge_{1,2} & edge_{2,2} & \cdots  & edge_{n, 2} \\ 
     \vdots & \vdots & \vdots  & \vdots \\ 
     edge_{1,n} & edge_{2,n} & \cdots  & edge_{n, n}
    \end{bmatrix}      
  }\end{displaymath}%DIFAUXCMD
%DIFDELCMD <   

%DIFDELCMD <   %%%
\DIFdel{In other words, $N$ is the merged table, each column $N_i$ is the vector representation of }\DIFdelend \DIFaddbegin \DIFadd{constructed a matrix $\mathbf{\bar{N}}$ whose $i^{th}$ column is the flattened version of the $i^{th}$ network, i.e. the column vector $\bar{N}^i$.
   Therefore, $\bar{N}^i_j$ is the weight of edge $j$ in the network $i$.
   $\bar{N}^i_j$ is assigned a value of 0 if edge $j$ did not exist in network $i$ but was present in }\DIFaddend one of the \DIFdelbegin \DIFdel{networks, and each row $L_i$ represents one particular edge in all networks (assigned 0 if the edge does not exist in the network)}\DIFdelend \DIFaddbegin \DIFadd{other networks.
   Note that row $j$ of $\mathbf{\bar{N}}$, $\bar{N}_j$ is the vector that encodes the values of edge $j$ across all the networks}\DIFaddend .

  \DIFaddbegin \begin{equation*}
   \DIFadd{\mathbf{\bar{N}} =
      \begin{bmatrix}
       \bar{N}^1_1 & \bar{N}^2_1 & \cdots  & \bar{N}^{W}_1  \\
       \bar{N}^1_2 & \bar{N}^2_2 & \cdots  & \bar{N}^{W}_2 \\
       \vdots & \vdots & \vdots  & \vdots \\
       \bar{N}^1_{Q^2} & \bar{N}^2_{Q^2} & \cdots  & \bar{N}^{W}_{Q^2}
      \end{bmatrix}
  }\end{equation*}

  \DIFadd{To infer the variability contributed due to the different steps in the pipeline we can perform a linear regression on each edge in $\mathbf{\bar{N}}$ and a subsequent ANOVA to extract the within-group variances.
  One issue with this approach is that the possibility of correlations existing between the edges of the network could lead to inaccurate estimates of the variance if a linear model were used to directly model the relationships between edges and steps in the workflow.
  In order to remedy this issue, we performed a PCA (Principal Component Analysis) on the matrix $\mathbf{\bar{N}}$ to obtain the $\mathbf{C}$ matrix ($W \times c$) of components for each network, such that we reduce the dimensions from the $Q^2$ dimensional edge space to a $c$ dimensional component space.
}

  \DIFaddend We \DIFaddbegin \DIFadd{then }\DIFaddend use linear regression to express each \DIFdelbegin \DIFdel{link $L_i$ }\DIFdelend \DIFaddbegin \DIFadd{component $C_j$ (where $j \in 1:c$) }\DIFaddend as a linear function of categorical variables that describe the possible options in each of the \DIFdelbegin \DIFdel{first three }\DIFdelend steps of the pipeline.

  %DIF <  TODO: Explain the categorical linear model and ANOVA better (text)
  In particular, we infer parameters \DIFdelbegin \DIFdel{$\alpha_i$ }\DIFdelend \DIFaddbegin \DIFadd{$\alpha_j$ }\DIFaddend such that:
  \DIFdelbegin \begin{displaymath}
       \DIFdel{L_i = \sum_{j=1}^5 \left( \alpha^{DC(j)}_i.\delta^{DC(j)}_i \right) +
             \sum_{j=1}^3 \left( \alpha^{TA(j)}_i.\delta^{TA(j)}_i \right) +
             \sum_{j=1}^2 \left( \alpha^{OP(j)}_i.\delta^{OP(j)}_i \right) +
             \epsilon_i
   }\end{displaymath}%DIFAUXCMD
\DIFdelend \DIFaddbegin \begin{multline}
  \DIFadd{C_j = }&\DIFadd{\sum_{i=1}^5 \left( \alpha^{DC(i)}_j \delta^{DC(i)}_j \right) +
         \sum_{i=1}^2 \left( \alpha^{CC(i)}_j \delta^{CC(i)}_j \right) +
         \sum_{i=1}^3 \left( \alpha^{TA(i)}_j \delta^{TA(i)}_j \right) +
         \sum_{i=1}^2 \left( \alpha^{OP(i)}_j \delta^{OP(i)}_j \right) + }\\
         &\DIFadd{\sum_{i=1}^{10} \left( \alpha^{NI(i)}_j \delta^{NI(i)}_j \right) +
         \epsilon_j
  }\end{multline}\DIFaddend 

   where, $\alpha_i$ are the coefficients of the regression, $\epsilon_i$ are the residuals and $\delta_i$ are the indicator variables that correspond to the processes utilized in the pipeline used to create the network $N_i$; for example, $\delta^{DC(1)}_i = 1$ if the DC(1) process was used in the generation of the network \DIFdelbegin \DIFdel{$N_i$ .
Here}\DIFdelend \DIFaddbegin \DIFadd{$N^i$.
}

   \DIFadd{Here,
   }\begin{enumerate}
     \item \DIFadd{$DC(i) \in$ \{CR, OR, DN, D2, DB\}
     }\item \DIFadd{$CC(i) \in$ \{remove bimera, uchime-denovo\}
     }\item \DIFadd{$TA(i) \in$ \{NaiveBayes(GG)}\DIFaddend , \DIFdelbegin \DIFdel{(i) DC(1}\DIFdelend \DIFaddbegin \DIFadd{NaiveBayes(SILVA), BLAST(NCBI}\DIFaddend )\DIFdelbegin \DIFdel{= "closed reference", DC(2) = "open reference", DC(3) = "de novo", DC(4) = "dada2", DC(5)= "deblur"; (ii)  TA(1) = "GreenGenes", TA(2) = "SILVA", TA(3) = "NCBI"; (iii)OP(1)= "no filtering", OP(2)= "filtering".
   }\DIFdelend \DIFaddbegin \DIFadd{\}
     }\item \DIFadd{$OP(i) \in$ \{Filter(on), Filter(off)\}
     }\item \DIFadd{$NI(i) \in$ \{}\acs{sparcc}\DIFadd{, propr, Spearman, Pearson, }\acs{spieceasi}\DIFadd{, }\acs{cozine}\DIFadd{, }\acs{harmonies}\DIFadd{, }\acs{spring}\DIFadd{, }\acs{mldm}\DIFadd{, FlashWeave\}
   }\end{enumerate}
\DIFaddend 

   The variance contributed by each step of the pipeline \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend calculated for every \DIFdelbegin \DIFdel{connection in the merged table }\DIFdelend \DIFaddbegin \DIFadd{component in $\mathbf{C}$ matrix }\DIFaddend through ANOVA using the Python statsmodels\DIFaddbegin \DIFadd{~\mbox{%DIFAUXCMD
\cite{seaboldStatsmodelsEconometricStatistical2010} }\hskip0pt%DIFAUXCMD
}\DIFaddend package and is shown in Figure~\DIFdelbegin \DIFdel{\ref{fig:figure2}B}\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:figure6}A}\DIFaddend .
  The total variance for the network \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend calculated by adding the variances for each connection \DIFdelbegin \DIFdel{.
  The PCA analysis is also performed on the merged table}\DIFdelend \DIFaddbegin \DIFadd{and normalizing with the degrees of freedom.
  The merged network table $\mathbf{\bar{N}}$ was used as the input to the PCA analysis to generate Figure~\ref{fig:figure6}B.
}

  \subsection*{\DIFadd{Synthetic interaction data}}
  \vspace{-5mm}
  \DIFadd{We generated synthetic interaction data using two methodologies previously used for benchmarking network inference methods.
}

  \DIFadd{The first method, ``seqtime''~\mbox{%DIFAUXCMD
\cite{faustSignaturesEcologicalProcesses2018}}\hskip0pt%DIFAUXCMD
, used generalized Lotka-Volterra (gLV) equations to model the microbial community dynamics and utilized the KlemmEguluz algorithm to generate a clique-based interaction network~\mbox{%DIFAUXCMD
\cite{Rottjers2018}}\hskip0pt%DIFAUXCMD
.
  We used the }\href{https://github.com/hallucigenia-sparsa/seqtime}{\DIFadd{seqtime}} \DIFadd{R package to simulate communities with number of species ($N$) varying from 10 }\DIFaddend to \DIFdelbegin \DIFdel{generate }\DIFdelend \DIFaddbegin \DIFadd{150 (10, 25, 50, 100, 150 and 200).
  The initial species concentrations were randomly sampled from a Poisson distribution and the simulation was rerun to generate a number of samples ($S$) varying from 50 to 500 (50, 100, 200, 500) for different communities.
  The abundance values of the species in the community at the end of the simulation time were used to create the OTU table.
}

  \DIFadd{The second method, ``NorTA'', used the Normal to Anything (NorTA) approach coupled with a given interaction network topology to generate the abundance distribution of the microbial community~\mbox{%DIFAUXCMD
\cite{Kurtz2015}}\hskip0pt%DIFAUXCMD
.
  We used the }\href{https://github.com/zdk123/SpiecEasi}{\DIFadd{spieceasi}} \DIFadd{R package~\mbox{%DIFAUXCMD
\cite{Kurtz2015} }\hskip0pt%DIFAUXCMD
to simulate communities with different network topologies (scale-free, cluster, block, Erdos-Renyi, band and hub) and target abundance distributions (Negative Binomial, Poisson, Zero-Inflated Negative Binomial).
  The OTU table was generated using the American Gut Project example in the spieceasi package (}\texttt{\DIFadd{amgut1.filt}}\DIFadd{) with the default parameter options.
}

  \DIFadd{For each method, we generated the OTU table depicting the abundances of species and used this as input to generate association networks using }\ac{micone} \DIFadd{pipeline.
  The interaction matrix was used as the source of expected (true) interactions and the associations predicted using }\ac{micone} \DIFadd{were the source of predicted interactions.
  Finally, for each dataset we evaluated the precision and sensitivity of the associations predicted by the individual network inference methods as well as the consensus (Figures \ref{fig:figure5}, \ref{fig:figure_s5}, and \ref{fig:figure_s6}).
}

  \subsection*{\DIFadd{Statistical analyses}}

  \subsubsection*{\DIFadd{DC step}}
  \vspace{-5mm}
  \DIFadd{In order to compare the representative sequences generated by the various methods in the DC step, we employed both the weighted~\mbox{%DIFAUXCMD
\cite{Lozupone2007} }\hskip0pt%DIFAUXCMD
(}\DIFaddend Figure~\ref{fig:figure2}\DIFdelbegin \DIFdel{C.
}\DIFdelend \DIFaddbegin \DIFadd{A) and unweighted UniFrac method~\mbox{%DIFAUXCMD
\cite{Lozupone2005} }\hskip0pt%DIFAUXCMD
(Figure~\ref{fig:figure2}B).
  The UniFrac distance metric (unique fraction metric) is a beta-diversity measure that computes the distance between two sets of taxa as the fraction of the branch length of the tree that leads to descendants from either one environment or the other, but not both~\mbox{%DIFAUXCMD
\cite{Lozupone2005}}\hskip0pt%DIFAUXCMD
.
  The weighted UniFrac distance metric takes into account the abundances of the representative sequences when calculating shared and unshared branch lengths, whereas the unweighted UniFrac distance metric does not and hence gives equal weights to each sequence.
  In Figure~\ref{fig:figure2} the distances between methods are the distance between the reference sequence distribution for a pair of methods averaged over every sample in the dataset.
  All UniFrac calculations were performed using the }\texttt{\DIFadd{scikit-bio}}\DIFadd{~\mbox{%DIFAUXCMD
\cite{thescikit-biodevelopmentteamScikitbioBioinformaticsLibrary2022} }\hskip0pt%DIFAUXCMD
v0.5.6 Python package.
}\DIFaddend 

  \DIFaddbegin \subsubsection*{\DIFadd{TA step}}
  \vspace{-5mm}
  \DIFadd{In Figure~\ref{fig:figure3}C, we used the Bray-Curtis distance metric to calculate the distance between the predicted (using the taxonomy databases in the TA step) and expected taxonomic distribution.
  The Bray-Curtis distance is used to quantify the compositional dissimilarity between two different taxonomic distributions defined by vectors $u$ and $v$.
  It is defined as:
  }$$\DIFadd{d = \frac{\sum_i{|u_i-v_i|}}{\sum_i{|u_i+v_i|}}}$$
  \DIFadd{The Bray-Curtis distance calculations were performed using the }\texttt{\DIFadd{scipy}}\DIFadd{~\mbox{%DIFAUXCMD
\cite{virtanenSciPyFundamentalAlgorithms2020} }\hskip0pt%DIFAUXCMD
v1.8.0 Python package.
}

  \subsubsection*{\DIFadd{NI step}}
  \vspace{-5mm}
  \DIFadd{In Figure~\ref{fig:figure5} we evaluated the accuracy and robustness of the inferred association networks (using the various network inference algorithms and the consensus methods) against the original interaction network used to create the taxonomic distribution.
  We used the following formulations of precision and sensitivity to calculate the accuracy of the predictions:
}

  \DIFadd{Precision = $\frac{TP}{TP + FP}$
}

  \DIFadd{Sensitivity = $\frac{TP}{FN + TP}$
}

  \DIFadd{where, TP - true positives, FP - false positives and FN - false negatives
}

  \DIFaddend \subsection*{Code and Data Availability}
  Pipeline: \href{https://github.com/segrelab/MiCoNE}{https://github.com/segrelab/MiCoNE} \\
  \DIFaddbegin \DIFadd{Documentation: }\href{https://micone.readthedocs.io}{\DIFadd{https://micone.readthedocs.io}} \\
  \DIFaddend Data and scripts: \href{https://github.com/segrelab/MiCoNE-pipeline-paper}{https://github.com/segrelab/MiCoNE-pipeline-paper} \DIFaddbegin \\
  \DIFadd{Synthetic data and scripts: }\href{https://github.com/segrelab/MiCoNE-synthetic-data}{\DIFadd{https://github.com/segrelab/MiCoNE-synthetic-data}}
\DIFaddend 



% ACKNOWLEDGMENTS
%!TEX root = ../main.tex

\section*{Acknowledgments}

We are grateful to members of the Segr lab for helpful discussions and for feedback on the manuscript. This work was partially funded by grants from the National Institutes of Health (National Institute of General Medical Sciences, award R01GM121950; National Institute of Dental and Craniofacial Research, award number R01DE024468; \DIFdelbegin \DIFdel{and }\DIFdelend National Institute on Aging, award number UH2AG064704\DIFaddbegin \DIFadd{; and National Cancer Institute, grant number R21CA260382}\DIFaddend ), the U.S. Department of Energy, Office of Science, Office of Biological \& Environmental Research through the Microbial Community Analysis and Functional Evaluation in Soils SFA Program (m-CAFEs) under contract number DE-AC02-05CH11231 to Lawrence Berkeley National Laboratory, the National Science Foundation (grants 1457695\DIFdelbegin \DIFdel{and }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend NSFOCE-BSF 1635070 \DIFdelbegin \DIFdel{), }\DIFdelend \DIFaddbegin \DIFadd{and the NSF Center for Chemical Currencies of a Microbial Planet) and }\DIFaddend the Human Frontiers Science Program (RGP0020/2016 \DIFdelbegin \DIFdel{), and the Boston University Interdisciplinary Biomedical Research Office}\DIFdelend \DIFaddbegin \DIFadd{and RGP0060/2021).
DK acknowledges support by the Kilachand Multicellular Design Program graduate fellowship}\DIFaddend .
KSK was supported by Simons Foundation Grant \#409704, by the Research Corporation for Science Advancement through Cottrell Scholar Award \#24010, by the Scialog grant \#26119, and by the Gordon and Betty Moore Foundation grant \#6790.08.

\section*{Contributions}
Designed the research project: DK, KK, DS, ZH, CDL. Performed analysis: DK, GB. Wrote the first draft of the manuscript: DK. Revised and wrote \DIFaddbegin \DIFadd{the }\DIFaddend final version of the manuscript: DK, DS, KK.


%--------------------------------------------------------%
%   REFERENCE LIST
%--------------------------------------------------------%
\newpage
\singlespacing
\printbibliography

%--------------------------------------------------------%
%   APPENDIX
%--------------------------------------------------------%

%!TEX root = ../main.tex

\begin{acronym}[XXXXXXXX]
    \acro{ngs}[NGS]{next-generation sequencing}
    \acro{mind}[MIND]{Microbial Interaction Network Database}
    \acro{micone}[MiCoNE]{Microbial Co-occurrence Network Explorer}
    \acro{otu}[OTU]{Operational Taxanomic Unit}
    \acro{esv}[ESV]{Exact Sequence Variant}
    \acro{pcr}[PCR]{Polymerase Chain Reaction}
    \acro{emp}[EMP]{Earth Microbiome Project}
    \acro{gg}[GG]{Greengenes}
    \acro{ncbi}[NCBI]{National Center for Biotechnology Information}
    \acro{comets}[COMETS]{Computation of Microbial Ecosystems in Time and Space}
    \acro{dada2}[DADA2]{Divisive Amplicon Denoising Algorithm 2}
    \acro{qiime1}[QIIME1]{Quantitative Insights Into Microbial Ecology 1}
    \acro{qiime2}[QIIME2]{Quantitative Insights Into Microbial Ecology 2}
    \acro{sparcc}[SparCC]{Sparse Correlations for Compositional data}
    \acro{spieceasi}[SpiecEasi]{Sparse InversE Covariance estimation for Ecological Association and Statistical Inference}
    \acro{mldm}[mLDM]{metagenomic Lognormal-Dirichlet-Multinomial}
    \DIFdelbegin %DIFDELCMD < \acro{magma}[MAGMA]{Microbial Association Graphical Model Analysis}
%DIFDELCMD <     %%%
\DIFdelend \DIFaddbegin \acro{cozine}[COZINE]{COmpositional Zero-Inflated Network Estimation}
    \acro{spring}[SPRING]{Semi-Parametric Rank-based approach for INference in Graphical model}
    \acro{harmonies}[HARMONIES]{Hybrid Approach for Microbiome Networks Inference via Exploiting Sparsity}
    \DIFaddend \acro{kegg}[KEGG]{Kyoto Encyclopedia of Genes and Genomes}
    \acro{fba}[FBA]{Flux Balance Analysis}
    \DIFaddbegin \acro{sp}[SP]{Sequence Processing}
    \DIFaddend \acro{dc}[DC]{Denoising and Clustering}
    \acro{ta}[TA]{Taxonomy Assignment}
    \acro{op}[OP]{OTU Processing}
    \acro{ni}[NI]{Network Inference}
\end{acronym}

%!TEX root = ../main.tex

\newpage
\section*{Tables and Figures}

% Tables

  \DIFdelbegin %DIFDELCMD < \begin{table}[h]
%DIFDELCMD <     %%%
\DIFdelendFL \DIFaddbeginFL \begin{table}[H]
    \DIFaddendFL \centering
    \DIFdelbeginFL %DIFDELCMD < \small
%DIFDELCMD <     \begin{tabular}{|c|c|c|}
%DIFDELCMD <       \hline
%DIFDELCMD <       %%%
\textbf{\DIFdelFL{Process}} %DIFAUXCMD
%DIFDELCMD < & %%%
\textbf{\DIFdelFL{Tool}} %DIFAUXCMD
%DIFDELCMD < & %%%
\textbf{\DIFdelFL{Parameters}} %DIFAUXCMD
%DIFDELCMD < \\
%DIFDELCMD <       \hline
%DIFDELCMD <       %%%
\DIFdelFL{Denoising and Clustering }%DIFDELCMD < & %%%
\DIFdelFL{Dada2/Deblur }%DIFDELCMD < & %%%
\DIFdelFL{default }%DIFDELCMD < \\
%DIFDELCMD <       %%%
\DIFdelFL{Taxonomy Assignment }%DIFDELCMD < & \ac{ncbi} %%%
\DIFdelFL{with Blast }%DIFDELCMD < & %%%
\DIFdelFL{RefSeq database }%DIFDELCMD < \\
%DIFDELCMD <       %%%
\DIFdelFL{OTU Processing }%DIFDELCMD < & %%%
\DIFdelFL{Based on statistical power }%DIFDELCMD < & %%%
\DIFdelFL{Dynamic cutoff }%DIFDELCMD < \\
%DIFDELCMD <       %%%
\DIFdelFL{Network Inference }%DIFDELCMD < & %%%
\DIFdelFL{Consensus method }%DIFDELCMD < & %%%
\DIFdelFL{- }%DIFDELCMD < \\
%DIFDELCMD <       \hline
%DIFDELCMD <     \end{tabular}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{\DIFdelbeginFL \DIFdelFL{Default }\DIFdelendFL 
      \DIFaddbeginFL \textbf{\DIFaddFL{Tools used in the }\ac{micone} \DIFaddFL{pipeline}}\DIFaddFL{.
      The }\DIFaddendFL tools \DIFdelbeginFL \DIFdelFL{and parameters }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{highlighted in gray are the defaults }\DIFaddendFL for the pipeline \DIFaddbeginFL \DIFaddFL{that are recommended based on the benchmarks with the mock and synthetic datasets.
      The consensus algorithm in the Network Inference (}\ac{ni}\DIFaddFL{) step incorporates all the modules (bootstrapping, direct association, and correlation-based) to generate the consensus network.
    }\DIFaddendFL }
    \DIFdelbeginFL %DIFDELCMD < \label{tab:default_options}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{tab:micone_tools}
  \DIFaddendFL \end{table}

% Figures

  \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \FloatBarrier
  %DIF >  \newpage
  %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=0.74\linewidth]{figure1.pdf}
  %DIF >  \end{figure}
  \DIFaddbeginFL \begin{figure}[H]
    \DIFaddendFL \centering
    \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=0.67\linewidth]{figure1.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
      \DIFdelbeginFL \textbf{\DIFdelFL{The workflow of the microbial co-occurrence analysis pipeline}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{The workflow of the }\ac{micone} \DIFaddFL{pipeline}}\DIFaddendFL .
      The steps \DIFaddbeginFL \DIFaddFL{of the workflow }\DIFaddendFL can be \DIFdelbeginFL \DIFdelFL{grouped }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{broken down }\DIFaddendFL into \DIFdelbeginFL \DIFdelFL{four }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{five }\DIFaddendFL major groups: \DIFaddbeginFL \textbf{\DIFaddFL{(SP)}} \textbf{\DIFaddFL{S}}\DIFaddFL{equence }\textbf{\DIFaddFL{P}}\DIFaddFL{rocessing, }\DIFaddendFL \textbf{(DC)} \textbf{D}enoising and \textbf{C}lustering, \textbf{(TA)} \textbf{T}axonomy \textbf{A}ssignment, \textbf{(OP)} \textbf{O}TU \DIFdelbeginFL \DIFdelFL{or }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{and }\DIFaddendFL ESV \textbf{P}rocessing, and \textbf{(NI)} \textbf{N}etwork \textbf{I}nference.
      Each step incorporates several processes \DIFaddbeginFL \DIFaddFL{(blue boxes)}\DIFaddendFL , each of which in turn \DIFdelbeginFL \DIFdelFL{have }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{has }\DIFaddendFL several \DIFdelbeginFL \DIFdelFL{alternate }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{alternative }\DIFaddendFL algorithms for the same task (indicated by the text to the right of the blue boxes).
      \DIFdelbeginFL \DIFdelFL{The text along the arrows }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Each arrow }\DIFaddendFL describes the data that is being passed from one step to another.
      \DIFdelbeginFL \DIFdelFL{For details on each process and data types, see Methods.
    }%DIFDELCMD < \MBLOCKRIGHTBRACE
%DIFDELCMD <     \label{fig:figure1}
%DIFDELCMD <   \end{figure}
%DIFDELCMD <   \FloatBarrier
%DIFDELCMD <   \newpage
%DIFDELCMD <   

%DIFDELCMD <   \begin{figure}[H]
%DIFDELCMD <     \centering
%DIFDELCMD <     \includegraphics[width=0.85\linewidth]{figure2.pdf}
%DIFDELCMD <   \end{figure}
%DIFDELCMD <   \begin{figure}[H]
%DIFDELCMD <     \centering
%DIFDELCMD <       \caption{
%DIFDELCMD <       %%%
\textbf{\DIFdelFL{The choice of database contributes to the most variance in the networks}}%DIFAUXCMD
\DIFdelFL{.
      }\textbf{\DIFdelFL{(A)}} %DIFAUXCMD
\DIFdelendFL The \DIFdelbeginFL \DIFdelFL{total relative variance in the networks contributed by the DC, TA and OP steps of the pipeline (right) and the linear model used }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{inputs }\DIFaddendFL to \DIFdelbeginFL \DIFdelFL{calculate }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{relative variance (left), see the Methods section for details.
      }\textbf{\DIFdelFL{(B)}} %DIFAUXCMD
\DIFdelFL{All combinations of inferred networks }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{pipeline }\DIFaddendFL are \DIFdelbeginFL \DIFdelFL{shown as points on a PCA plot.
      The color of the points corresponds to the taxonomy database}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{16S rRNA sequencing reads}\DIFaddendFL , \DIFdelbeginFL \DIFdelFL{the shape corresponds to the denoising/clustering method }\DIFdelendFL and the \DIFdelbeginFL \DIFdelFL{size corresponds to whether low abundance OTUs were removed or not.
      }\textbf{\DIFdelFL{(B inset)}} %DIFAUXCMD
\DIFdelFL{The }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{final output is the consensus }\DIFaddendFL network generated \DIFdelbeginFL \DIFdelFL{using DC=dada2, TA=GG, OP=no and NI=SPARCC and represents }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{from }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{particular point shown (big red square)}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{inferred co-occurrence networks}\DIFaddendFL .
      \DIFdelbeginFL \DIFdelFL{The plot clearly shows that the points can be separated based }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{For details }\DIFaddendFL on \DIFdelbeginFL \DIFdelFL{the TA step }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{each process }\DIFaddendFL and \DIFdelbeginFL \DIFdelFL{that }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{differences due to the DC and OP steps are not as significant}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{different outputs, see Methods}\DIFaddendFL .
    }
    \DIFdelbeginFL %DIFDELCMD < \label{fig:figure2}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure1}
  \DIFaddendFL \end{figure}
\DIFdelbegin %DIFDELCMD < \FloatBarrier
%DIFDELCMD <   \newpage
%DIFDELCMD <   %%%
\DIFdelend 



  \DIFdelbegin %DIFDELCMD < \begin{figure}
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \FloatBarrier
  %DIF >  \newpage
  %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=\textwidth]{figure2.pdf}
  %DIF >  \end{figure}
  \DIFaddbeginFL \begin{figure}[H]
    \DIFaddendFL \centering
    \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\textwidth]{figure3.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
      \DIFdelbeginFL \textbf{\DIFdelFL{The representative sequences generated by the different denoising/clustering methods are very similar but differ in the sequences that are in low abundance.}}
      %DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{The representative sequences generated by the different denoising and clustering methods differ in their identification of sequences that are low in abundance}}\DIFaddFL{.
      }\DIFaddendFL \textbf{(A)} The average weighted UniFrac distance between the representative sequences shows that the representative sequences and their compositions are fairly identical between the methods \DIFdelbeginFL \DIFdelFL{,
      }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{(with the exception of Deblur (DB) due to the low ESV count).
      }\DIFaddendFL \textbf{(B)} The relatively larger average unweighted UniFrac distance indicates that methods differ in their identification of sequences \DIFdelbeginFL \DIFdelFL{of low }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{that are lower in }\DIFaddendFL abundance\DIFaddbeginFL \DIFaddFL{.
      The number of }\ac{otu}\DIFaddFL{s or }\ac{esv}\DIFaddFL{s generated by the respective methods are provided in the parenthesis next to their names.
      The data used for the analysis in (A}\DIFaddendFL , \DIFaddbeginFL \DIFaddFL{B) were the samples from the fecal microbiome transplant (FMT) dataset~\mbox{%DIFAUXCMD
\cite{Kang2017}}\hskip0pt%DIFAUXCMD
, containing both healthy subjects and subjects with autism spectrum disorder (ASD).
      }\DIFaddendFL \textbf{(C, D)} The distributions of the average weighted \DIFaddbeginFL \DIFaddFL{and unweighted }\DIFaddendFL UniFrac distance between the \DIFdelbeginFL \DIFdelFL{expected }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{predicted }\DIFaddendFL sequence profile and the \DIFdelbeginFL \DIFdelFL{calculated }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{expected }\DIFaddendFL sequence profile in \DIFaddbeginFL \DIFaddFL{the }\DIFaddendFL mock datasets.
      \DIFdelbeginFL \textbf{\DIFdelFL{(D)}} %DIFAUXCMD
\DIFdelendFL The \DIFdelbeginFL \DIFdelFL{distributions of the }\DIFdelendFL average \DIFdelbeginFL \DIFdelFL{unweighted }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{weighted }\DIFaddendFL UniFrac \DIFdelbeginFL \DIFdelFL{distance }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{distances }\DIFaddendFL show that \DIFdelbeginFL \DIFdelFL{dada2 }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{de novo (DN) }\DIFaddendFL and \DIFdelbeginFL \DIFdelFL{Deblur }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{open reference (OR) }\DIFaddendFL were the \DIFdelbeginFL \DIFdelFL{best performing }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{best-performing }\DIFaddendFL methods in most of the datasets\DIFaddbeginFL \DIFaddFL{, while they are the worst-performing methods under the unweighted UniFrac metric}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{The good performance of dada2 (D2) under both distance metrics combined with its approach of identifying }\ac{esv}\DIFaddFL{s using de novo methods, prompts us to use it as the default method for the DC step.
      The data used for the analysis in (C, D) were the mock4, mock12, and mock16 datasets from mockrobiota~\mbox{%DIFAUXCMD
\cite{Bokulich2016}}\hskip0pt%DIFAUXCMD
.
    }\DIFaddendFL }
    \DIFdelbeginFL %DIFDELCMD < \label{fig:figure3}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure2}
  \DIFaddendFL \end{figure}
\DIFdelbegin %DIFDELCMD < \FloatBarrier
%DIFDELCMD <   \newpage
%DIFDELCMD <   %%%
\DIFdelend 


  %DIF >  \FloatBarrier
  %DIF >  \newpage
  %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=\textwidth]{figure3.pdf}
  %DIF >  \end{figure}
  \begin{figure}[H]
    \centering
    \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=0.9\textwidth,height=1.2\textwidth]{figure4.pdf}
%DIFDELCMD <   \end{figure}
%DIFDELCMD <   \begin{figure}[H]
%DIFDELCMD <     \centering
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
      \DIFdelbeginFL \textbf{\DIFdelFL{Taxonomic reference databases vary widely in terms of their taxonomy assignments.}}
      %DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{Taxonomic reference databases vary in terms of their taxonomy assignments below the Order level}}\DIFaddFL{.
      }\DIFaddendFL \textbf{(A)} The \DIFdelbeginFL \DIFdelFL{assignment }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{taxonomic assignments }\DIFaddendFL of the top 50 representative sequences \DIFdelbeginFL \DIFdelFL{to their respective taxonomies }\DIFdelendFL using the three different reference databases\DIFdelbeginFL \DIFdelFL{shows }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{.
      This result illustrates }\DIFaddendFL how the same sequences are assigned to different \DIFaddbeginFL \DIFaddFL{genera under different databases.
      A significant portion of the representative sequences are assigned to an ``unknown'' }\DIFaddendFL Genus \DIFaddbeginFL \DIFaddFL{in two of three databases (}\ac{gg} \DIFaddFL{and }\ac{ncbi}\DIFaddFL{)}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{The number of assigned genera for each database is displayed at the top of each column.
      }\DIFaddendFL \textbf{(B)} The \DIFdelbeginFL \DIFdelFL{percentage }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{number }\DIFaddendFL of \DIFdelbeginFL %DIFDELCMD < \ac{otu}%%%
\DIFdelFL{s }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{representative sequences }\DIFaddendFL assigned to the same taxonomic label when using different reference databases \DIFaddbeginFL \DIFaddFL{(for the top 100 sequences)}\DIFaddendFL .
      The \DIFdelbeginFL \DIFdelFL{percentage of }\DIFdelendFL mismatches \DIFdelbeginFL \DIFdelFL{decrease }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{are fewer }\DIFaddendFL at higher taxonomic levels\DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL but\DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL even at the \DIFdelbeginFL \DIFdelFL{Phylum }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Order }\DIFaddendFL level there exists \DIFdelbeginFL \DIFdelFL{around 10}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{greater than 51}\DIFaddendFL \% of mismatches\DIFaddbeginFL \DIFaddFL{, demonstrating the poor agreement in taxonomic labels assigned by the different databases}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{The data used for the analysis in (A, B) were samples (healthy and ASD) from the FMT dataset.
      }\DIFaddendFL \textbf{(C)} The Bray-Curtis dissimilarity between the \DIFdelbeginFL \DIFdelFL{expected }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{predicted }\DIFaddendFL taxonomy profile and \DIFdelbeginFL \DIFdelFL{calculated }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{expected }\DIFaddendFL taxonomy profile in the mock datasets shows that there is no singular best choice of database for every dataset\DIFaddbeginFL \DIFaddFL{, as all the databases show similar performances}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{The }\ac{gg} \DIFaddFL{database and the Naive Bayes classifier are chosen as the defaults for the TA step of }\ac{micone} \DIFaddFL{due to their popularity.
      The datasets used for the analysis in (C) were the mock datasets from mockrobiota.
    }\DIFaddendFL }
    \DIFdelbeginFL %DIFDELCMD < \label{fig:figure4}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure3}
  \DIFaddendFL \end{figure}
\DIFdelbegin %DIFDELCMD < \FloatBarrier
%DIFDELCMD <   \newpage
%DIFDELCMD <   %%%
\DIFdelend 


  %DIF >  \FloatBarrier
  %DIF >  \newpage
  %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=\textwidth]{figure4.pdf}
  %DIF >  \end{figure}
  \begin{figure}[H]
    \centering
    \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=0.9\textwidth]{figure5.pdf}
%DIFDELCMD <   \end{figure}
%DIFDELCMD <   \begin{figure}[H]
%DIFDELCMD <     \centering
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
      \DIFdelbeginFL \textbf{\DIFdelFL{Networks generated using different network inference methods show notable differences both in terms of edge-density and connectivity}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{Networks generated using different network inference methods show notable differences in terms of edge-density and connectivity}}\DIFaddendFL .
      \textbf{(A)} The \DIFdelbeginFL \DIFdelFL{six }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{nine }\DIFaddendFL different networks \DIFaddbeginFL \DIFaddFL{(excluding }\acs{mldm}\DIFaddFL{) }\DIFaddendFL generated by the different network inference methods\DIFaddbeginFL \DIFaddFL{.
      The nodes for each network (representing taxa) }\DIFaddendFL are \DIFdelbeginFL \DIFdelFL{very dissimilar}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{arranged in the same positions in a circular layout and the differences in the connections can be directly visualized and compared}\DIFaddendFL .
      The green links are positive associations and the orange links \DIFdelbeginFL \DIFdelFL{are }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{represent }\DIFaddendFL negative associations.
      \DIFaddbeginFL \DIFaddFL{The networks look dissimilar and vary widely in terms of connectivity, and it is notable that the correlation-based methods generally produce networks with higher edge-densities.
      }\DIFaddendFL A threshold of 0.3 was set for the \DIFaddbeginFL \DIFaddFL{correlation-based }\DIFaddendFL methods \DIFdelbeginFL \DIFdelFL{that infer pairwise correlations }\DIFdelendFL (\DIFdelbeginFL %DIFDELCMD < \ac{sparcc}%%%
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{sparcc}\DIFaddendFL , \DIFdelbeginFL \DIFdelFL{Spearman}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{propr}\DIFaddendFL , \DIFdelbeginFL \DIFdelFL{Pearson}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{spearman and pearson}\DIFaddendFL ) and \DIFdelbeginFL \DIFdelFL{no }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{a }\DIFaddendFL threshold \DIFaddbeginFL \DIFaddFL{of 0.01 }\DIFaddendFL was set for the \DIFdelbeginFL \DIFdelFL{other }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{direct association }\DIFaddendFL methods \DIFaddbeginFL \DIFaddFL{(flashweave, spieceasi, cozine, harmonies, and spring)}\DIFaddendFL .
      \textbf{(B)} The node overlap Upset plot \DIFdelbeginFL \DIFdelFL{~\mbox{%DIFAUXCMD
\cite{Lex} }\hskip0pt%DIFAUXCMD
}\DIFdelendFL indicates that all the networks have a large \DIFdelbeginFL \DIFdelFL{number }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{proportion }\DIFaddendFL of common nodes involved in connections \DIFaddbeginFL \DIFaddFL{(33 out of 68)}\DIFaddendFL .
      \DIFdelbeginFL \DIFdelFL{Whereas, }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Conversely }\DIFaddendFL \textbf{(C)}\DIFdelbeginFL \DIFdelFL{The }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, the }\DIFaddendFL edge overlap Upset plot shows that a very small fraction of these connections are actually shared \DIFaddbeginFL \DIFaddFL{(8 out of 202)}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{The data used in this analysis were the healthy stool samples from the FMT dataset.
      }\acs{mldm} \DIFaddFL{is not shown in the comparisons because the algorithm failed to converge for the particular network combination used here (default setting of the }\ac{micone} \DIFaddFL{pipeline).
    }\DIFaddendFL }
    \DIFdelbeginFL %DIFDELCMD < \label{fig:figure5}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure4}
  \DIFaddendFL \end{figure}
\DIFdelbegin %DIFDELCMD < \FloatBarrier
%DIFDELCMD <   \newpage
%DIFDELCMD <   %%%
\DIFdelend 


  \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \FloatBarrier
  %DIF >  \newpage
  %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=1.0\linewidth]{figure5.pdf}
  %DIF >  \end{figure}
  \DIFaddbeginFL \begin{figure}[H]
    \DIFaddendFL \centering
    \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=0.85\linewidth]{figure6.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
      \DIFdelbeginFL \textbf{\DIFdelFL{Network inference and taxonomic assignment have the highest influence on the inferred network structures.}}
      %DIFAUXCMD
\textbf{\DIFdelFL{(A)}} %DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{The associations generated by the scaled-sum consensus method show high precision in benchmarks using synthetic datasets}}\DIFaddFL{.
    }\DIFaddendFL The \DIFaddbeginFL \DIFaddFL{different points on the box plot show the precision of co-occurrence networks generated through individual }\DIFaddendFL network \DIFdelbeginFL \DIFdelFL{constructed using }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{inference methods and through consensus network construction approaches. Precision is estimated based on }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{default pipeline parameters }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{comparisons with two sets of synthetic benchmark datasets }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{DC=}%DIFDELCMD < \ac{dada2}%%%
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{``NorTA'' and ``seqtime''}\DIFaddendFL , \DIFdelbeginFL \DIFdelFL{TA=}%DIFDELCMD < \ac{ncbi}%%%
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{see Methods).
      The independent algorithms chosen for the comparison are the two best-performing correlation-based (propr}\DIFaddendFL , \DIFdelbeginFL \DIFdelFL{OP=on}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{sparcc) and direct association based (spieceasi}\DIFaddendFL , \DIFdelbeginFL \DIFdelFL{NI=}%DIFDELCMD < \ac{sparcc}%%%
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{flashweave}\DIFaddendFL ) \DIFdelbeginFL \DIFdelFL{is compared with networks generated when one of }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{methods.
      For consensus network inference, we used }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{steps use }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{scaled-sum (SS) and simple voting (SV) methods.
      A weight threshold of 0.1 and }\DIFaddendFL a \DIFdelbeginFL \DIFdelFL{different tool}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{p-value threshold of 0.05 was applied to each network before the calculation of precision}\DIFaddendFL .
      The \DIFdelbeginFL \DIFdelFL{common connections }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{overall best precision was consistently obtained by the scaled-sum consensus method for $p \geq 0.333$ on both benchmark datasets. Among all the individual network inference methods, spieceasi shows the best average precision.
      The simple voting method, when using the presence of edges in all inferred networks as a requirement }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{common }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{$p = 1.000$), also outperforms spieceasi on average precision.
      Therefore, we set the scaled-sum consensus method }\DIFaddendFL with \DIFaddbeginFL \DIFaddFL{$p = 0.333$ as }\DIFaddendFL the default \DIFaddbeginFL \DIFaddFL{tool for consensus }\DIFaddendFL network \DIFaddbeginFL \DIFaddFL{inference, since this option provides a good balance of precision and sensitivity (see also Figure~\ref{fig:figure_s5} and \ref{fig:figure_s6}}\DIFaddendFL )\DIFdelbeginFL \DIFdelFL{are }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{.
    }}
    \label{fig:figure5}
  \end{figure}


  %DIF >  \FloatBarrier
  %DIF >  \newpage
  %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=\textwidth]{figure6.pdf}
  %DIF >  \end{figure}
  \begin{figure}[H]
    \centering
      \caption{
      \textbf{\DIFaddFL{The choice of reference database has the largest impact on network variance}}\DIFaddFL{.
      }\textbf{\DIFaddFL{(A)}} \DIFaddFL{The percentage of variance }\DIFaddendFL in \DIFdelbeginFL \DIFdelFL{black}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the networks (from the  FMT dataset) contributed by the }\acf{dc}\DIFaddendFL , \DIFdelbeginFL \DIFdelFL{connections unique to }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{CC (chimera checking), }\acf{ta}\DIFaddFL{, }\acf{op} \DIFaddFL{and }\acf{ni} \DIFaddFL{steps of }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{network are colored purple }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{pipeline calculated using ANOVA on a linear model (see Methods).
      A weight threshold of 0.1 }\DIFaddendFL and \DIFdelbeginFL \DIFdelFL{connections }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{a p-value threshold of 0.05 were applied to each network before the analysis.
      The taxonomy database contributes most to the variance between the networks (65.4\%) followed by the filtering of the counts matrix (26.8\%) }\DIFaddendFL in the \DIFdelbeginFL \DIFdelFL{default network but not present }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{OP step.
    The variation due to the NI, DC and CC steps are much smaller }\DIFaddendFL in \DIFaddbeginFL \DIFaddFL{comparison (6.553\%, 0.648\%, and 0.003\% respectively).
      The negligible fraction labeled as }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{current network }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{residual is an artifact that arises when multiple steps }\DIFaddendFL are \DIFdelbeginFL \DIFdelFL{gray}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{changed at the same time}\DIFaddendFL .
      \textbf{(B)} \DIFdelbeginFL \DIFdelFL{The L1 distance between }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{All }\DIFaddendFL the \DIFaddbeginFL \DIFaddFL{inferred }\DIFaddendFL networks generated \DIFdelbeginFL \DIFdelFL{by changing one }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{from various combinations of tools are shown as points on a PCA plot.
      Each point on the PCA plot represents a network inferred using different combinations of tools and parameters that are available in the }\ac{micone} \DIFaddFL{pipeline.
      The color of the points corresponds to the tools used at each }\DIFaddendFL step of the \DIFdelbeginFL \DIFdelFL{default }\DIFdelendFL pipeline \DIFaddbeginFL \DIFaddFL{(DC, TA, OP, }\DIFaddendFL and \DIFaddbeginFL \DIFaddFL{NI).
      The points on }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{network generated using }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{PCA plot can be grouped based on }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{default parameters}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{TA step, but the extent of this separation decreases when the filtering is turned on in the OP step, confirming that the variability in the networks decreased upon filtering out the taxonomic entities at low abundance}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{Some algorithms, especially the direct association methods, at the NI step can also be seen to generate networks that are less variable compared to the others.
      The }\ac{dc} \DIFaddFL{step does not seem to have any correlation with the variation in the networks on the PCA plot.
    }\DIFaddendFL }
    \label{fig:figure6}
  \end{figure}
\DIFdelbegin %DIFDELCMD < \FloatBarrier
%DIFDELCMD <   \newpage
%DIFDELCMD < %%%
\DIFdelend 


  \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \FloatBarrier
  %DIF >  \newpage
  %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=1.0\linewidth]{figure7.pdf}
  %DIF >  \end{figure}
  \DIFaddbeginFL \begin{figure}[H]
    \DIFaddendFL \centering
    \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=17cm]{figure7.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
      \DIFaddbeginFL \textbf{\DIFaddFL{Comparison of networks generated from the control and ASD samples of the FMT dataset using the }\ac{micone} \DIFaddFL{pipeline}}\DIFaddFL{.
      }\DIFaddendFL The \DIFdelbeginFL \DIFdelFL{consensus }\DIFdelendFL networks \DIFaddbeginFL \DIFaddFL{for the control (left) and ASD (right) samples were }\DIFaddendFL generated using the default \DIFaddbeginFL \DIFaddFL{tools and parameters recommended by the }\ac{micone} \DIFaddendFL pipeline \DIFdelbeginFL \DIFdelFL{settings}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{as described in Table~\ref{tab:micone_tools}}\DIFaddendFL .
      \DIFdelbeginFL \textbf{\DIFdelFL{(A)}} %DIFAUXCMD
\DIFdelFL{Co-occurrence network of }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{There are 22 unique links in }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{Hard Palate microbiome generated from }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{network for control }\DIFaddendFL samples\DIFdelbeginFL \DIFdelFL{of healthy subjects }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, 12 unique links }\DIFaddendFL in \DIFdelbeginFL \DIFdelFL{a periodontal diseases study.
      }\textbf{\DIFdelFL{(B)}} %DIFAUXCMD
\DIFdelFL{Co-occurrence network of }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{Stool microbiome generated from samples of healthy }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{network for ASD }\DIFaddendFL subjects\DIFaddbeginFL \DIFaddFL{, and 7 edges }\DIFaddendFL in \DIFdelbeginFL \DIFdelFL{a fecal microbiome transplant study}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{common between both networks}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{The changes in these connections can serve as potential starting points for further experimental validations or literature surveys.
    }\DIFaddendFL }
    \label{fig:figure7}
  \end{figure}
%DIF < !TEX root = ../main.tex


  \DIFaddbegin \FloatBarrier
  \DIFaddend \newpage
  \DIFdelbegin \section*{\DIFdel{Supplementary}}
%DIFAUXCMD
\DIFdelend \DIFaddbegin \subsection*{\DIFadd{Supplementary Tables and Figures}}
\DIFaddend 

  \renewcommand{\thefigure}{S\arabic{figure}}
  \setcounter{figure}{0}

  \renewcommand{\thetable}{S\arabic{table}}
  \setcounter{table}{0}

  \DIFdelbegin %DIFDELCMD < \begin{table}[h]
%DIFDELCMD < \resizebox{\textwidth}{!}{
%DIFDELCMD < \begin{tabular}{lllll}
%DIFDELCMD < \hline
%DIFDELCMD < \textbf{Step}                             & \textbf{Task}                                            & \textbf{Tool}                          & \textbf{Parameter}                     & \textbf{Value}                                                                                           \\ \hline
%DIFDELCMD < \multirow{29}{*}{Denosing and Clustering} & \multicolumn{1}{c}{\multirow{9}{*}{Sequence Processing}} & \multirow{2}{*}{join\_reads}           & min\_overlap                           & 6                                                                                                        \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     &                                        & perc\_max\_diff                        & 8                                                                                                        \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     & \multirow{2}{*}{demultiplex\_illumina} & rev\_comp\_barcodes                    & False                                                                                                    \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     &                                        & rev\_comp\_mapping\_barcodes           & False                                                                                                    \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     & demultiplex\_454                       & -                                      & -                                                                                                        \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     & \multirow{4}{*}{trim\_filter\_fixed}   & seq\_sample\_size                      & 10,000                                                                                                   \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     &                                        & trunc\_q                               & 2                                                                                                        \\
%DIFDELCMD <                                           & \multicolumn{1}{c}{}                                     &                                        & max\_ee                                & 2                                                                                                        \\ \cline{2-5}
%DIFDELCMD <                                           & \multirow{3}{*}{Chimera Checking}                        & uchime                                 & -                                      & -                                                                                                        \\
%DIFDELCMD <                                           &                                                          & \multirow{2}{*}{remove\_bimera}        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & chimera\_method                        & consensus                                                                                                \\ \cline{2-5}
%DIFDELCMD <                                           & \multirow{17}{*}{Denoise Cluster}                        & \multirow{3}{*}{de\_novo}              & enable\_rev\_strand\_match             & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & suppress\_de\_novo\_chimera\_detection & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          & \multirow{4}{*}{closed\_reference}     & enable\_rev\_strand\_match             & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & suppress\_de\_novo\_chimera\_detection & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & reference\_sequences                   & 97\_otus.fasta                                                                                           \\
%DIFDELCMD <                                           &                                                          & \multirow{5}{*}{open\_reference}       & enable\_rev\_strand\_match             & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & suppress\_de\_novo\_chimera\_detection & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & reference\_sequences                   & 97\_otus.fasta                                                                                           \\
%DIFDELCMD <                                           &                                                          &                                        & picking\_method                        & uclust                                                                                                   \\
%DIFDELCMD <                                           &                                                          & \multirow{2}{*}{dada2}                 & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & big\_data                              & FALSE                                                                                                    \\
%DIFDELCMD <                                           &                                                          & \multirow{3}{*}{deblur}                & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & mind\_reads                            & 2                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & min\_size                              & 2                                                                                                        \\ \hline
%DIFDELCMD < \multirow{7}{*}{Taxonomy Assignment}      & \multirow{7}{*}{Assign}                                  & \multirow{3}{*}{naive\_bayes}          & confidence                             & 0.7                                                                                                      \\
%DIFDELCMD <                                           &                                                          &                                        & mem\_per\_core                         & 8G                                                                                                       \\
%DIFDELCMD <                                           &                                                          &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          & \multirow{4}{*}{blast}                 & max\_accepts                           & 10                                                                                                       \\
%DIFDELCMD <                                           &                                                          &                                        & perc\_identity                         & 0.8                                                                                                      \\
%DIFDELCMD <                                           &                                                          &                                        & evalue                                 & 0.001                                                                                                    \\
%DIFDELCMD <                                           &                                                          &                                        & min\_consensus                         & 0.51                                                                                                     \\ \hline
%DIFDELCMD < \multirow{12}{*}{OTU/ESV Processing}      & \multirow{5}{*}{Filter}                                  & \multirow{3}{*}{abundance}             & count\_thres                           & 500                                                                                                      \\
%DIFDELCMD <                                           &                                                          &                                        & prevalence\_thres                      & 0.05                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & abundance\_thres                       & 0.01                                                                                                     \\
%DIFDELCMD <                                           &                                                          & group                                  & tax\_levels                            & \begin{tabular}[c]{@{}l@{}}{[}'Phylum', 'Class', 'Order',\\ 'Family', 'Genus', 'Species'{]}\end{tabular} \\
%DIFDELCMD <                                           &                                                          & partition                              & -                                      & -                                                                                                        \\ \cline{2-5}
%DIFDELCMD <                                           & \multirow{6}{*}{Transform}                               & \multirow{6}{*}{normalize}             & count\_thres                           & 500                                                                                                      \\
%DIFDELCMD <                                           &                                                          &                                        & axis                                   & sample                                                                                                   \\
%DIFDELCMD <                                           &                                                          &                                        & prevalence\_thres                      & 0.05                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & abundace\_thres                        & 0.01                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & rm\_sparse\_obs                        & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & rm\_sparse\_samples                    & True                                                                                                     \\ \cline{2-5}
%DIFDELCMD <                                           & Export                                                   & biom2tsv                               & -                                      & -                                                                                                        \\ \hline
%DIFDELCMD < \multirow{17}{*}{Network Inference}       & \multirow{4}{*}{Bootstrap}                               & \multirow{3}{*}{resample}              & bootstraps                             & 1000                                                                                                     \\
%DIFDELCMD <                                           &                                                          &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & filter\_flag                           & True                                                                                                     \\
%DIFDELCMD <                                           &                                                          & pvalue                                 & ncpus                                  & 1                                                                                                        \\ \cline{2-5}
%DIFDELCMD <                                           & \multirow{12}{*}{Correlation}                            & \multirow{2}{*}{sparcc}                & iterations                             & 50                                                                                                       \\
%DIFDELCMD <                                           &                                                          &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          & pearson                                & -                                      & -                                                                                                        \\
%DIFDELCMD <                                           &                                                          & spearman                               & -                                      & -                                                                                                        \\
%DIFDELCMD <                                           &                                                          & \multirow{5}{*}{spieceasi}             & method                                 & mb                                                                                                       \\
%DIFDELCMD <                                           &                                                          &                                        & ncpus                                  & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & nreps                                  & 50                                                                                                       \\
%DIFDELCMD <                                           &                                                          &                                        & nlambda                                & 20                                                                                                       \\
%DIFDELCMD <                                           &                                                          &                                        & lambda\_min\_ratio                     & 1e-2                                                                                                     \\
%DIFDELCMD <                                           &                                                          & \multirow{2}{*}{mldm}                  & z\_mean                                & 1                                                                                                        \\
%DIFDELCMD <                                           &                                                          &                                        & max\_iteration                         & 1500                                                                                                     \\
%DIFDELCMD <                                           &                                                          & magma                                  & -                                      & -                                                                                                        \\ \cline{2-5}
%DIFDELCMD <                                           & Network                                                  & make\_network                          & -                                      & -                                                                                                        \\ \hline
%DIFDELCMD < \end{tabular}
%DIFDELCMD < }
%DIFDELCMD < %%%
\DIFdelendFL \DIFaddbeginFL \begin{table}[H]
    \centering
    \DIFaddendFL \caption{
      \DIFaddbeginFL \textbf{\DIFaddFL{Table of global network metrics for networks inferred from all possible combinations of tools}}\DIFaddFL{.
      In each row, one tool in a particular step is kept constant, and the metric is calculated for every possible combination of tools for the other steps of the pipeline.
      Therefore, each row shows the grouped average metric for each tool in every step of the pipeline.
      }\DIFaddendFL The \DIFdelbeginFL \DIFdelFL{default parameters used }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{network inference methods show the most variation }\DIFaddendFL in the \DIFdelbeginFL \DIFdelFL{various }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{global network metrics compared to }\DIFaddendFL tools \DIFaddbeginFL \DIFaddFL{in other steps }\DIFaddendFL of the pipeline\DIFaddbeginFL \DIFaddFL{.
    }\DIFaddendFL }
    \DIFdelbeginFL %DIFDELCMD < \label{tab:all_parameters}
%DIFDELCMD < %%%
\DIFdelendFL \DIFaddbeginFL \label{tab:network_metrics}
  \DIFaddendFL \end{table}


    \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <   %%%
\DIFdelendFL %DIF >  \begin{figure}[H]
    %DIF >    \centering
    %DIF >    \includegraphics[width=1.0\linewidth]{figure_s1.pdf}
    %DIF >  \end{figure}
    \DIFaddbeginFL \begin{figure}[H]
      \DIFaddendFL \centering
        \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{figureS1.pdf}
%DIFDELCMD <   %%%
\DIFdelendFL \caption{
          \DIFdelbeginFL \textbf{\DIFdelFL{Comparison of various denoising and clustering algorithms used in the pipeline}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{The t-SNE plot of all the inferred networks clusters the networks based on the taxonomy reference database used}}\DIFaddendFL .
          \DIFdelbeginFL \DIFdelFL{(A, B) Correlation of }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Each point on }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{abundances }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{t-SNE plot represents a network inferred using different combinations }\DIFaddendFL of \DIFdelbeginFL \DIFdelFL{the taxa }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{tools and parameters }\DIFaddendFL that are \DIFaddbeginFL \DIFaddFL{available }\DIFaddendFL in \DIFdelbeginFL \DIFdelFL{common between }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{count matrices created by two different methods}\DIFdelendFL \DIFaddbeginFL \ac{micone} \DIFaddFL{pipeline}\DIFaddendFL .
          \DIFaddbeginFL \DIFaddFL{The points are colored by the tools and parameters used in }\ac{dc} \DIFaddFL{step }\DIFaddendFL (A)\DIFdelbeginFL \DIFdelFL{The worst correlation }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, }\ac{ta} \DIFaddFL{step }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{least similar methods) is between open-reference and dada2.
    (}\DIFdelendFL B)\DIFdelbeginFL \DIFdelFL{The best correlation }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, }\ac{op} \DIFaddFL{step }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{most similar methods}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{C}\DIFaddendFL ) \DIFdelbeginFL \DIFdelFL{is between open-reference }\DIFdelendFL and \DIFdelbeginFL \DIFdelFL{denovo.
    }\DIFdelendFL \DIFaddbeginFL \ac{ni} \DIFaddFL{step }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{C}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{D}\DIFaddendFL )\DIFdelbeginFL \DIFdelFL{A heatmap showing the $\mathrm{R}^2$ }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{.
          The separation }\DIFaddendFL of \DIFdelbeginFL \DIFdelFL{all pairwise comparisons of }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{methods}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{points based on taxonomy reference database shows that the points cluster based on reference database in high-dimensional space}\DIFaddendFL .
        }
      \DIFdelbeginFL %DIFDELCMD < \label{fig:figureS1}
%DIFDELCMD < %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure_s1}
    \DIFaddendFL \end{figure}
    %DIF >  \FloatBarrier
    %DIF >  \newpage

    \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \begin{figure}[H]
    %DIF >    \centering
    %DIF >    \includegraphics[width=1.0\linewidth]{figure_s2.pdf}
    %DIF >  \end{figure}
    \DIFaddbeginFL \begin{figure}[H]
      \DIFaddendFL \centering
        \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{figureS2.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
          \DIFdelbeginFL \textbf{\DIFdelFL{Heatmaps showing the weighted and unweighted unifrac distances for the hard palate dataset analysis}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{The UniFrac distance between the 1000 most abundant representative sequences is higher than that when all sequences are considered}}\DIFaddendFL .
          \DIFaddbeginFL \DIFaddFL{Each value is the average UniFrac distance between the reference sequences generated by the various methods in the }\ac{dc} \DIFaddFL{step }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{A}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{similar to Figure~\ref{fig:figure2}}\DIFaddendFL )\DIFaddbeginFL \DIFaddFL{.
          There is an increase in both }\DIFaddendFL weighted \DIFdelbeginFL \DIFdelFL{unifrac distances }\DIFdelendFL and \DIFdelbeginFL \DIFdelFL{(B) }\DIFdelendFL unweighted \DIFdelbeginFL \DIFdelFL{unifrac }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{UniFrac }\DIFaddendFL distances \DIFdelbeginFL \DIFdelFL{between }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{compared to when all }\DIFaddendFL the representative sequences \DIFaddbeginFL \DIFaddFL{are considered.
          This shows that the 1000 most abundant representative sequences }\DIFaddendFL generated by \DIFdelbeginFL \DIFdelFL{different denoising and clustering algorithms}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the DC methods are not as similar to each other}\DIFaddendFL .
          \DIFdelbeginFL \DIFdelFL{These results }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{And since the weighted UniFrac is much smaller than the unweighted UniFrac distance, we can conclude that those reference sequences that }\DIFaddendFL are \DIFaddbeginFL \DIFaddFL{present }\DIFaddendFL in \DIFdelbeginFL \DIFdelFL{agreement with }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{stool microbiome dataset}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{middle of the abundance distribution (considering all sequences) are dissimilar}\DIFaddendFL .
        }
      \DIFdelbeginFL %DIFDELCMD < \label{fig:figureS2}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure_s2}
    \DIFaddendFL \end{figure}
    %DIF >  \FloatBarrier
    %DIF >  \newpage

    \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \begin{figure}[H]
    %DIF >    \centering
    %DIF >    \includegraphics[width=1.0\linewidth]{figure_s3.pdf}
    %DIF >  \end{figure}
    \DIFaddbeginFL \begin{figure}[H]
      \DIFaddendFL \centering
        \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{figureS3.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
          \DIFdelbeginFL \textbf{\DIFdelFL{The distributions of the average weighted UniFrac distance between the expected sequence profile and the calculated sequence profile in the synthetic datasets}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{The weighted and unweighted UniFrac distances are small for the representative sequences generated using remove bimera and uchime for each denoising method}}\DIFaddendFL .
          \DIFdelbeginFL \DIFdelFL{We observe no significant difference between }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{With }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{various }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{exception of de novo and open reference under the unweighted UniFrac metric, all the other }\DIFaddendFL methods \DIFdelbeginFL \DIFdelFL{on }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{have high similarity, implying that }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{synthetic datasets used }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{two chimera checking methods, uchime and remove bimera, produce similar outputs.
          This is especially true }\DIFaddendFL for \DIFdelbeginFL \DIFdelFL{this study}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the }\ac{dada2} \DIFaddFL{and Deblur methods which are the recommended denoising methods in the }\ac{micone} \DIFaddFL{pipeline}\DIFaddendFL .
          \DIFaddbeginFL \DIFaddFL{Therefore, remove bimera is recommended as the default chimera method if one is using }\ac{dada2} \DIFaddFL{and uchime-denovo when one is using Deblur, since these methods were developed for these respective algorithms (}\ac{qiime2} \DIFaddFL{uses uchime-denovo in the Deblur workflow).
        }\DIFaddendFL }
      \DIFdelbeginFL %DIFDELCMD < \label{fig:figureS3}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure_s3}
    \DIFaddendFL \end{figure}
    %DIF >  \FloatBarrier
    %DIF >  \newpage

    %DIF <    \begin{figure}[h]
%DIF >  \begin{figure}[H]
    %   \centering
    %DIF <      \includegraphics[width=0.9\linewidth]{pdf/all_denoise_reg.pdf}
%DIF <      \caption*{All pairwise correlations comparing the similarity between different denoising and clustering methods}
%DIF <      \label{fig:figureS4}
%DIF >    \includegraphics[width=1.0\linewidth]{figure_s4.pdf}
    % \end{figure}
    \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD <   \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL \DIFaddbeginFL \begin{figure}[H]
      \DIFaddendFL \centering
        \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{figureS4.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
          \DIFdelbeginFL \textbf{\DIFdelFL{(A)}} %DIFAUXCMD
\DIFdelFL{Taxonomy composition }\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{The pairwise comparison of assignments generated using different databases for all representative sequences has a higher proportion of mismatches}}\DIFaddFL{.
          The comparison made here is similar to Figure~\ref{fig:figure3}B, but instead }\DIFaddendFL of the \DIFdelbeginFL \DIFdelFL{20 most abundant genera predicted for }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{top 100 taxonomic entities (by abundance), all }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{stool microbiome dataset generated using different taxonomy references }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{assignments from one database are matched with those from the other two }\DIFaddendFL databases\DIFdelbeginFL \DIFdelFL{: Greengenes, SILVA and NCBI}\DIFdelendFL .
          The \DIFdelbeginFL \DIFdelFL{legend shows }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{higher percentage of mismatches implies that }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{common and }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{assigned taxonomies in }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{unique genera among the taxonomy assignments}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{more abundant sequences (top 100) match more consistently}\DIFaddendFL .
        }
      \DIFdelbeginFL %DIFDELCMD < \label{fig:figureS4}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure_s4}
    \DIFaddendFL \end{figure}
    %DIF >  \FloatBarrier
    %DIF >  \newpage

  \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=1.0\linewidth]{figure_s5.pdf}
  %DIF >  \end{figure}
  \DIFaddbeginFL \begin{figure}[H]
    \DIFaddendFL \centering
      \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{figureS5.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
        \DIFaddbeginFL \textbf{\DIFaddFL{The precision and sensitivity of the inferred networks on the ``NorTA'' synthetic interaction data}}\DIFaddFL{.
        }\DIFaddendFL The \DIFdelbeginFL \DIFdelFL{bray-curtis dissmilarity between }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{different consensus methods used are scaled-sum (SS) and simple voting (SV) methods.
        Pearson and Spearman methods are not used in }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{expected taxonomic composition }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{calculation of the consensus.
        Among all the independent network inference methods, }\acs{spieceasi} \DIFaddFL{has the best average precision (0.944), but the overall best precision was consistently obtained by the scaled-sum method (0.956, 0.985, }\DIFaddendFL and \DIFdelbeginFL \DIFdelFL{generated taxonomic composiion for }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{1.000).
        The simple voting method when using }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{synthetic datasets}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{presence of edges in all inferred networks as a requirement ($p = 1.000$), also outperforms }\acs{spieceasi} \DIFaddFL{on average precision (0.969)}\DIFaddendFL .
        \DIFaddbeginFL \DIFaddFL{Although }\acs{spieceasi} \DIFaddFL{has a higher sensitivity, if the goal of network inference is to obtain the list of associations that have a high probability of existing in the real microbial community, then the consensus methods perform better.
      }\DIFaddendFL }
    \DIFdelbeginFL %DIFDELCMD < \label{fig:figureS5}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure_s5}
  \DIFaddendFL \end{figure}
  %DIF >  \FloatBarrier
  %DIF >  \newpage

  %DIF <    \begin{figure}[h]
%DIF >  \begin{figure}[H]
  %   \centering
  %DIF <      \includegraphics[width=\linewidth]{figureS6.pdf}
%DIF <      \caption{
%DIF <        Calculation of presence threshold that is applied on the OTU table in the OTU processing (OP) step of the pipeline.
%DIF <        This presence threhold $p_t$ is dependent on the number of samples in the dataset and the required correlation stength threshold.
%DIF <    }
%DIF <      \label{fig:figureS6}
%DIF >    \includegraphics[width=1.0\linewidth]{figure_s6.pdf}
  % \end{figure}
  \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD <   \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL \DIFaddbeginFL \begin{figure}[H]
    \DIFaddendFL \centering
      \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{figureS8.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{
        \DIFaddbeginFL \textbf{\DIFaddFL{The precision and sensitivity of the inferred networks on the ``seqtime'' synthetic interaction data.}}
        \DIFaddendFL The \DIFdelbeginFL \DIFdelFL{similarity between }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{different consensus methods used are scaled-sum (SS) and simple voting (SV) methods.
        Pearson and Spearman methods are not used in }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{networks generated using }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{calculation of }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{different }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{consensus.
        Among all the independent }\DIFaddendFL network inference \DIFdelbeginFL \DIFdelFL{algorithms for stool dataset }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{methods, }\acs{spieceasi} \DIFaddFL{has the best average precision }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{A}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{0.624}\DIFaddendFL )\DIFdelbeginFL \DIFdelFL{and }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, but }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{hard palate dataset }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{overall best precision was consistently obtained by the scaled-sum method }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{B}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{0.688, 0.820, and 1.000}\DIFaddendFL ).
        The \DIFdelbeginFL \DIFdelFL{similarity between }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{simple voting method when using }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{various methods was found to vary with }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{presence of edges in all inferred networks as a requirement ($p = 1.000$), also outperforms }\acs{spieceasi} \DIFaddFL{on average precision (0.692).
        These results show that }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{dataset used}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{scaled-sum method is not only much better suited for inferring robust and accurate interactions from count data generated from network topologies (NorTA), but it is also capable of accurately extracting real associations from Lotka-Volterra simulations}\DIFaddendFL .
      }
    \DIFdelbeginFL %DIFDELCMD < \label{fig:figureS8}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure_s6}
  \DIFaddendFL \end{figure}
  %DIF >  \FloatBarrier
  %DIF >  \newpage

  \DIFdelbegin %DIFDELCMD < \begin{figure}[h]
%DIFDELCMD <     %%%
\DIFdelendFL %DIF >  \begin{figure}[H]
  %DIF >    \centering
  %DIF >    \includegraphics[width=1.0\linewidth]{figure_s7.pdf}
  %DIF >  \end{figure}
  \DIFaddbeginFL \begin{figure}[H]
    \DIFaddendFL \centering
    \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=0.8\linewidth]{pdf/denoise_network.pdf}
%DIFDELCMD <     %%%
\DIFdelendFL \caption{\DIFdelbeginFL \DIFdelFL{A }\DIFdelendFL 
      \DIFaddbeginFL \textbf{\DIFaddFL{Sensitivity analysis of the default settings of the }\ac{micone} \DIFaddFL{pipeline}}\DIFaddFL{.
      }\textbf{\DIFaddFL{(A)}} \DIFaddFL{The }\DIFaddendFL network \DIFaddbeginFL \DIFaddFL{constructed using the default pipeline parameters (DC=}\ac{dada2}\DIFaddFL{, CC=remove bimera, TA=}\ac{gg}\DIFaddFL{, OP=Filter(on), NI=scaled-sum consensus) is compared against networks generated when one of the steps uses a different tool.
      The layout is created by fixing the positions of all the nodes from all networks and then drawing only the relevant edges, making the connections directly comparable.
      The edges colored green are positive associations and those in red are negative associations.
      We observe that changing the TA and OP steps leads to the creation of the most number of unique edges.
      }\textbf{\DIFaddFL{(B)}} \DIFaddFL{The dot plot }\DIFaddendFL showing \DIFdelbeginFL \DIFdelFL{union }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the fraction of nodes }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{A}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{left}\DIFaddendFL ) and \DIFdelbeginFL \DIFdelFL{intersection }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{edges }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{B}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{right}\DIFaddendFL ) \DIFdelbeginFL \DIFdelFL{of }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{in common between the default network and the }\DIFaddendFL networks generated \DIFdelbeginFL \DIFdelFL{using different denoising }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{by changing one step of the default pipeline.
      The low value of the common fraction for TA }\DIFaddendFL and \DIFdelbeginFL \DIFdelFL{clustering tools on }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{OP steps shows that these steps induce }\DIFaddendFL the \DIFdelbeginFL \DIFdelFL{Stool dataset}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{biggest changes in nodes and edges}\DIFaddendFL .
      \DIFaddbeginFL \DIFaddFL{The NI step is not shown in this analysis because the consensus methods use edges from the individual network inference methods and a comparison would be biased.
    }\DIFaddendFL }
    \DIFdelbeginFL %DIFDELCMD < \label{fig:figureS5}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \label{fig:figure_s7}
  \DIFaddendFL \end{figure}
  %DIF >  \FloatBarrier
  %DIF >  \newpage
\DIFaddbegin 


%DIF > !TEX root = ../main.tex

\newpage
\section*{\DIFadd{Supplementary Text}}

  \subsection*{\DIFadd{Processing the FMT data}}

    \subsubsection*{\DIFadd{Data download and pre-processing}}
    \DIFadd{The main biological dataset used in this study was the collection of 16S rRNA sequencing reads from stool samples (healthy and autistic individuals) for a fecal microbiome transplant study~\mbox{%DIFAUXCMD
\cite{Kang2017}}\hskip0pt%DIFAUXCMD
.
    The data containing the 16S sequencing reads (V4 region) was downloaded from Qiita~\mbox{%DIFAUXCMD
\cite{gonzalezQiitaRapidWebenabled2018} }\hskip0pt%DIFAUXCMD
(study ID: 10532).
    Only runs 2, 3, and 4 were used for the subsequent analysis as these runs consisted of paired-end sequencing data, and run 1 contained single-end data.
    The sample metadata was updated to contain only BMI, sex, height, weight, and experimental group.
    This was necessary as two of the network inference algorithms (}\acs{mldm} \DIFadd{and FlashWeave) required information about environmental heterogeneity.
    However, these environmental correlations were not included in the current analyses.
}

    \subsubsection*{\DIFadd{Processing using the }\ac{micone} \DIFadd{pipeline}}
    \DIFadd{The data was then processed using the }\ac{micone} \DIFadd{pipeline starting at the }\ac{sp} \DIFadd{step and ending at the }\ac{ni} \DIFadd{step with the consensus algorithm.
    The configuration files (main.nf and nextflow.config) used to run the }\ac{micone} \DIFadd{pipeline as well the details of the pipeline execution (dag, report, timeline and trace) are in the "runs/FMT" directory of the data and scripts repository (}\href{https://github.com/segrelab/MiCoNE-pipeline-paper}{\DIFadd{https://github.com/segrelab/MiCoNE-pipeline-paper}}\DIFadd{)
    The results of the pipeline execution for reproducing the analyses in the manuscript are stored on Zenodo.
}

  \subsection*{\DIFadd{Processing the mock data}}

    \subsubsection*{\DIFadd{Data download and pre-processing}}
    \DIFadd{The mock datasets, mock4, mock12 and mock16 used for this study, were obtained from mockrobiota~\mbox{%DIFAUXCMD
\cite{Bokulich2016}}\hskip0pt%DIFAUXCMD
.
    Mock 4 is a mock community composed of 21 bacterial strains represented in equal abundances in two replicate samples, and the same strains represented in uneven abundances in two other replicate samples.
    Mock 12 is composed of 27 bacterial strains containing closely related taxa, the members of which were chosen in part for their well-separated 16S rRNA gene sequences. Some pairs of strains differ by as little as one nucleotide, but all the strains are distinguishable over the sequenced region of the 16S rRNA gene.
    Mock 16 is a mock community composed of even amounts of purified genomic DNA from 49 bacteria and 10 archaea.
    The datasets did not require any preprocessing and could be directly used as input to the pipeline
}

    \subsubsection*{\DIFadd{Processing using the }\ac{micone} \DIFadd{pipeline}}
    \DIFadd{The data was processed using the }\ac{micone} \DIFadd{pipeline starting at the }\ac{sp} \DIFadd{step and ending at the }\ac{op} \DIFadd{step with the filtered taxonomic tables as the final output.
    The configuration files (main.nf and nextflow.config) used to run the }\ac{micone} \DIFadd{pipeline as well the details of the pipeline execution (dag, report, timeline and trace) are in the "runs/mock*" directory of the data and scripts repository (}\href{https://github.com/segrelab/MiCoNE-pipeline-paper}{\DIFadd{https://github.com/segrelab/MiCoNE-pipeline-paper}}\DIFadd{)
    The results of the pipeline execution for reproducing the analyses in the manuscript are stored on Zenodo.
}

    \subsubsection*{\DIFadd{Interpretation of Unifrac results in the DC step}}
  \DIFadd{In Figure \ref{fig:figure_s3}, we observe that both the weighted and unweighted UniFrac distances are increased for the top 1000 representative sequences, implying that the top representative sequences generated by the different methods are not as similar to each other.
  Therefore, since the weighted UniFrac distances are lower than the unweighted distances, we conclude that the representative sequences in the middle range of the abundance distribution are those that must be the most similar between the methods.
}

    \DIFadd{Open-reference and de novo clustering methods perform the best under the weighted UniFrac metric and the worst (marginally) under the unweighted UniFrac metric (Figure~\ref{fig:figure2}C and \ref{fig:figure2}D).
    This result can be attributed to the large number of low abundance representative sequences that are generated by these methods.
    Deblur performs poorly under weighted Unifrac and although its performance on the mock4 dataset is the best under unweighted UniFrac, its performance on the other datasets is average.
    The Deblur method returns a very small number of representative sequences (2388) and this could account for the reason for the high dissimilarity with the other methods as well as irregular performance on the mock data.
}

  \subsection*{\DIFadd{Synthetic interaction data}}

    \subsubsection*{\DIFadd{Data generation}}
    \DIFadd{The synthetic interaction data for the study were generated using two methods.
    The first method, ``seqtime''~\mbox{%DIFAUXCMD
\cite{faustSignaturesEcologicalProcesses2018} }\hskip0pt%DIFAUXCMD
utilized generalized Lotka-Volterra (gLV) equations to model the microbial community dynamics and made use of the KlemmEguluz algorithm to generate clique-based interaction networks~\mbox{%DIFAUXCMD
\cite{Rottjers2018}}\hskip0pt%DIFAUXCMD
.
    We used the seqtime R package to simulate communities with different numbers of species and samples (see Methods for details).
    The second method, ``NorTA'' used the Normal to Anything (NorTA) approach coupled with a given interaction network topology to generate the abundance distribution of the microbial community~\mbox{%DIFAUXCMD
\cite{Kurtz2015}}\hskip0pt%DIFAUXCMD
.
    We used the spieceasi R package to simulate communities with different abundance distributions and network topologies (see Methods for details).
    The scripts to generate these datasets can be found in the synthetic data and scripts repository (}\href{https://github.com/segrelab/MiCoNE-synthetic-data}{\DIFadd{https://github.com/segrelab/MiCoNE-synthetic-data}}\DIFadd{)
}

    \subsubsection*{\DIFadd{Processing using the }\ac{micone} \DIFadd{pipeline}}
    \DIFadd{The data was processed using the }\ac{micone} \DIFadd{pipeline using only the }\ac{ni} \DIFadd{step with the consensus networks as the final output.
    The configuration files (main.nf and nextflow.config) used to run the }\ac{micone} \DIFadd{pipeline as well the details of the pipeline execution (dag, report, timeline and trace) are in the "runs/norta" and "runs/seqtime" directories of the data and scripts repository (}\href{https://github.com/segrelab/MiCoNE-pipeline-paper}{\DIFadd{https://github.com/segrelab/MiCoNE-pipeline-paper}}\DIFadd{)
    The results of the pipeline execution for reproducing the analyses in the manuscript are stored on Zenodo.
}


  \subsection*{\DIFadd{Network metrics}}

  \DIFadd{In Table~\ref{tab:network_metrics} we show various global network metrics calculated for each tool in the pipeline.
  All the networks that make use of a particular tool are grouped together, and the following average metrics are calculated for each group:
  }\begin{enumerate}
    \item \DIFadd{The average shortest path length describes the average of all the shortest paths in the graph. No number is reported if the graph is not connected, therefore, the results indicate that none of the networks that make use of }\acs{harmonies}\DIFadd{, }\acs{cozine}\DIFadd{, }\acs{spring}\DIFadd{, }\acs{spieceasi} \DIFadd{and Pearson are connected.
    }\item \DIFadd{The average clustering is the average clustering coefficient of the graph. The closer the value is to 1.0, the more densely connected is the graph. We can observe that the networks that use correlation-based methods have the highest values while the direct association based methods have the lowest.
    }\item \DIFadd{The number of connected components is the highest for the direct association based methods and the lowest for the correlation-based methods. In the case of propr, all the networks have only one giant component.
    }\item \DIFadd{The modularity metric is the modularity over all partitions in a graph calculated using a label propagation algorithm~\mbox{%DIFAUXCMD
\cite{cordascoCommunityDetectionSemisynchronous2010}}\hskip0pt%DIFAUXCMD
. Positive values imply that there are more edges between vertices of the same type than we would expect by chance, and negative implies that there are less. The networks inferred by }\acs{mldm} \DIFadd{report very few edges, and skew the average modularity scores. This could also be an artifact of incomplete convergence of the }\acs{mldm} \DIFadd{algorithm for some combinations.
    }\item \DIFadd{Node connectivity refers to the minimum number of nodes that must be removed from the graph to make it disconnected. We observe that only the networks generated using propr have a high value since most of these networks are connected.
    }\item \DIFadd{Degree assortativity coefficient measures the similarity of connections in the graph with respect to the node degree. Again we observe that the direct association based methods have a negative degree of assortativity, meaning that there are many hubs in these networks. The correlation-based methods have positive values implying that in these networks nodes with similar degrees attach to one another.
  }\end{enumerate}
  \DIFadd{A weight threshold of 0.1 and a p-value threshold of 0.05 were applied to each network before the analysis.
  All the metrics were calculated using the }\texttt{\DIFadd{networkx}} \DIFadd{Python package~\mbox{%DIFAUXCMD
\cite{hagbergExploringNetworkStructure2008}}\hskip0pt%DIFAUXCMD
.
}

  \subsection*{\DIFadd{p-value merging}}

  \DIFadd{Fisher~\mbox{%DIFAUXCMD
\cite{fisher_224a_1948} }\hskip0pt%DIFAUXCMD
proposed that for $k$ independent p-values, each generated by $k$ different methods and denoted by $\bar{P}^i$ (notations are same as used in the "Consensus network and p-value merging" subsection of the Methods), the following will hold true for the statistic $\Psi$:
  }\begin{equation*}
    \DIFadd{\begin{aligned}
      \Psi &= \sum_{i=1}^k -2 \log \left( \bar{P}^i \right) \\
        \Psi &\sim \chi^2_{2k}
    \end{aligned}
  }\end{equation*}

  \DIFadd{Brown~\mbox{%DIFAUXCMD
\cite{brown_400_1975} }\hskip0pt%DIFAUXCMD
extended Fisher's method to dependent p-values by using a re-scaled $\chi^2$ distribution:
  }\begin{equation*}
    \DIFadd{\Psi \sim c \chi^2_{2f}
  }\end{equation*}
  \DIFadd{where, $f$ is the degrees of freedom and $c$ is the scale factor and are given by:
  }\begin{equation*}
    \DIFadd{f = \frac{\mathrm{E}[\Psi]^2}{\mathrm{Var}[\Psi]} ~~~\text{and}~~~ c = \frac{\mathrm{Var}[\Psi]}{2\mathrm{E}[\Psi]} = \frac{k}{f}
  }\end{equation*}

  \DIFadd{We can calculate $E[\Psi]$ and $Var[\Psi]$ under the null hypothesis that the data are drawn from a multivariate Gaussian with some covariance matrix.
  We then use these values to parametrize a $\chi^2$ distribution from which the p-value corresponding to $\frac{\psi}{c}$ can be calculated.
  Furthermore, Brown showed that $\mathrm{E}[\Psi]$ and $\mathrm{Var}[\Psi]$ can be calculated via the following numerical approximation:
  }\begin{equation*}
    \DIFadd{\mathrm{E}[\Psi] = 2k ~~~\text{and}~~~ \mathrm{Var}[\Psi] = 4k + 2\sum_{i<j} \mathrm{Cov}\left( -2\log(\bar{P}^i), -2\log(\bar{P}^j) \right)
  }\end{equation*}

  \DIFadd{The above formulation was improved by Kost and McDermott~\mbox{%DIFAUXCMD
\cite{kost_combining_2002} }\hskip0pt%DIFAUXCMD
by further fitting a third-order polynomial to approximate the covariance
  }\begin{equation}
    \DIFadd{\mathrm{Cov}\left( -2\log(\bar{P}^i), -2\log(\bar{P}^j) \right) \approx 3.263 \rho_{ij} + 0.710 \rho_{ij}^2 + 0.027 \rho_{ij}^3
    \label{eqn:suppl_covariance-pvalues}
  }\end{equation}
  \DIFadd{where, $\rho_{ij}$ is the correlation between method $i$ and method $j$
}

  \DIFadd{Using $E[\Psi]$ and $Var[\Psi]$ we then fit a $\chi^2$ distribution with the parameters $c$ and $f$.
  Note that since, in general, $f$ will not be an integer, this should be understood as a Gamma distribution with a shape parameter $f$, as mentioned by Brown~\mbox{%DIFAUXCMD
\cite{brown_400_1975}}\hskip0pt%DIFAUXCMD
.
  Using this, we calculate the test $\psi$ and compute the p-value from the CDF of the $\chi^2$ distribution, given in Equation~\ref{eqn:suppl_pvalue-combined}.
  Therefore, the final combined p-value~\mbox{%DIFAUXCMD
\cite{Poole_Gibbs_Shmulevich_Bernard_Knijnenburg_2016} }\hskip0pt%DIFAUXCMD
is then given by:
  }\begin{equation}
    \DIFadd{\begin{aligned*}
        }& \DIFadd{\hat{P}_j = 1.0 - \Phi_{2f}\left( \psi / c \right) }\\
        \DIFadd{\text{where},~ }&\DIFadd{\psi = -2 \sum_{i=1}^k \log(\bar{P}^i_j) ~~~\text{and}~~~ \Phi_{2f} = \mathrm{CDF}\left( \chi^2_{2f} \right)
    \end{aligned*}
    \label{eqn:suppl_pvalue-combined}
  }\end{equation}

  \DIFadd{The p-value merging and consensus method in }\ac{micone} \DIFadd{(see Methods) uses Equation~\ref{eqn:suppl_covariance-pvalues} to estimate the covariance of the p-values and Equation~\ref{eqn:suppl_pvalue-combined} to merge the p-values (obtained from bootstrapping) from the different correlation methods.
  Note that we do not use Pearson and Spearman methods in the p-value merging step and these algorithms are only used for demonstration and comparison.
  The combined p-values are used to threshold for significance in the correlation-based networks during the consensus network step.
}

  %DIF >  TODO: Should we mention the MIND database here?
  \subsection*{\DIFadd{The JSON network format and network exports}}

    \DIFadd{The default format }\ac{micone} \DIFadd{uses for storing the network files is the JSON (JavaScript Object Notation) format.
    The custom JSON schema we have designed is able to store all network-related information pertaining to nodes, links, and the metadata related to the links and datasets.
    Additionally, }\ac{micone} \DIFadd{also supports exporting of networks into a variety of other formats such as edge lists, GML, and Cytoscape formats.
    Since we make use of }\texttt{\DIFadd{networkx}}\DIFadd{~\mbox{%DIFAUXCMD
\cite{hagbergExploringNetworkStructure2008} }\hskip0pt%DIFAUXCMD
for the export functionality, networks can be exported to all formats supported by the package.
    However, not all the corresponding metadata will be exported appropriately, as most formats do not support this additional metadata.
    The details of the format and information about importing/exporting it and other network formats can be found in the }\ac{micone} \DIFadd{documentation.
}

  \subsection*{\DIFadd{Supplementary discussion}}

  \DIFadd{It is worth pointing out some additional, more specific, conclusions stemming from the individual steps of our analysis.
  The different denoising/clustering methods differ mostly in their identification of sequences that are in low abundance.
  Hence, they do not have much of an impact on the inferred co-occurrence networks when the sequences of low abundance are removed (Figure~\ref{fig:figure1}).
  Comparison of inferred and expected reference sequences and their abundances in mock community datasets has allowed us to identify }\ac{dada2} \DIFadd{as the method which best recapitulates the expected sequence composition.
  For the chimera checking module, we suggest using the remove bimera method since it was developed in conjunction with }\ac{dada2} \DIFadd{and its performance does not significantly differ from uchime-denovo.
  For the current work we have decided to focus on the tools most widely used at the time of the analysis.
  Some tools which were not as widely used (e.g. dbOTU3~\mbox{%DIFAUXCMD
\cite{Olesen2017}}\hskip0pt%DIFAUXCMD
) as well as older popular methods like mothur~\mbox{%DIFAUXCMD
\cite{Schloss2009} }\hskip0pt%DIFAUXCMD
have not been included in the study but could be added into the pipelines in future updated analyses.
}

  \DIFadd{The choice of taxonomy database was found to be the most important factor in the inference of microbial co-occurrence networks, contributing $65.4\%$ of the total variance.
  The frequent changes in the taxonomy nomenclature coupled with the frequency of updates to the various 16S reference databases create inherent differences \mbox{%DIFAUXCMD
\cite{Balvociute2017} }\hskip0pt%DIFAUXCMD
in taxonomy hierarchies in these databases.
  Our analysis revealed that no particular reference database performs better than the others across the different mock dataset benchmarks.
  The default reference database in the pipeline is the }\ac{gg} \DIFadd{reference database along with the ``Naive Bayes'' classifier as the query tool.
  The reason for our choice stems from the popularity of the }\ac{gg} \DIFadd{database~\mbox{%DIFAUXCMD
\cite{parkEvaluation16SRRNA2018} }\hskip0pt%DIFAUXCMD
in taxonomic studies, which would enable easy comparison across datasets.
  However, for newer studies we recommend using SILVA database because of its size and taxonomic comprehensiveness~\mbox{%DIFAUXCMD
\cite{iiRESCRIPtReproducibleSequence2021} }\hskip0pt%DIFAUXCMD
and since }\ac{gg} \DIFadd{has not been updated since 2013.
  Additionally, a particular database might be more appropriate than the rest based on specific requirements.
  For example, in order to generate a dataset that is compatible with the }\ac{mind} \DIFadd{platform~\mbox{%DIFAUXCMD
\cite{huResourceComparisonIntegration2022} }\hskip0pt%DIFAUXCMD
}\ac{ncbi} \DIFadd{is the most appropriate choice as it guarantees compatibility of taxonomic hierarchy and therefore comparability with other datasets.
  Furthermore, we also enable users to use custom databases~\mbox{%DIFAUXCMD
\cite{Ritari2015,iiRESCRIPtReproducibleSequence2021} }\hskip0pt%DIFAUXCMD
with the BLAST and Naive Bayes classifiers that are incorporated into the pipeline (from }\ac{qiime2}\DIFadd{).
  We suggest that that choice of the database should be made based on possible reported or inferred biases in the representation of given biomes in a specific databases~\mbox{%DIFAUXCMD
\cite{Balvociute2017,iiRESCRIPtReproducibleSequence2021}}\hskip0pt%DIFAUXCMD
, as choosing taxon-specific databases have also been observed to compromise classification~\mbox{%DIFAUXCMD
\cite{rmarcelinoUseTaxonspecificReference2020}}\hskip0pt%DIFAUXCMD
.
}

  \DIFadd{The }\ac{op} \DIFadd{step of the pipeline is second in its contribution to total network variance.
  This can be attributed to the large number of nodes that are added to the final networks when the filtering is turned off.
  Additionally, a very large number of nodes also decreases the accuracy of the network inference algorithms for the same sample size~\mbox{%DIFAUXCMD
\cite{peschelNetCoMiNetworkConstruction2020} }\hskip0pt%DIFAUXCMD
and increases the computational complexity~\mbox{%DIFAUXCMD
\cite{tackmannRapidInferenceDirect2019}}\hskip0pt%DIFAUXCMD
.
  We observe that filtering out taxa that are present in low abundances in all samples increases the proportion of taxa in common between taxonomy tables generated using different reference databases (Figure~\ref{fig:figure_s4}), providing another reason for filtering.
  We also observe that the reduction in the number of taxa leads to a better agreement in the networks inferred through different methods (Figure~\ref{fig:figure1}).
  Moreover, filtering is necessary in order to increase the power in tests of significance when the number of taxa is much greater than the number of samples.
}

  \pagenumbering{goggle}
  \thispagestyle{empty}
    \begin{table}[H]
    \centering
    \begin{tabular}{|c|c|c|c|}
      \hline
      \textbf{\DIFaddFL{Workflow step}} & \textbf{\DIFaddFL{Module/Condition}} & \textbf{\DIFaddFL{Tool/Parameter}} & \textbf{\DIFaddFL{References/Value}} \\
      \hline
      \multirow{7}{*}{Denoising and Clustering} & \multirow{5}{*}{Denoise and Cluster} & \DIFaddFL{Closed reference }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{rognesVSEARCHVersatileOpen2016,bolyenReproducibleInteractiveScalable2019} }\hskip0pt%DIFAUXCMD
}\\
                                                &  & \DIFaddFL{Open reference }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{rognesVSEARCHVersatileOpen2016,bolyenReproducibleInteractiveScalable2019} }\hskip0pt%DIFAUXCMD
}\\
                                                &  & \DIFaddFL{De novo }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{rognesVSEARCHVersatileOpen2016,bolyenReproducibleInteractiveScalable2019} }\hskip0pt%DIFAUXCMD
}\\
                                                &  & \rowcolor{lightgray} \DIFaddFL{Dada2 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Callahan2016} }\hskip0pt%DIFAUXCMD
}\\
                                                &  & \DIFaddFL{Deblur }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Amir2017,bolyenReproducibleInteractiveScalable2019} }\hskip0pt%DIFAUXCMD
}\\
                                                \cline{2-4}
                                                & \multirow{2}{*}{Chimera checking} & \DIFaddFL{Uchime-denovo }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{rognesVSEARCHVersatileOpen2016,bolyenReproducibleInteractiveScalable2019} }\hskip0pt%DIFAUXCMD
}\\
                                                & & \rowcolor{lightgray} \DIFaddFL{Remove bimera }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Callahan2016} }\hskip0pt%DIFAUXCMD
}\\
                                                \hline
      \multirow{5}{*}{Taxonomy assignment} &  \multirow{2}{*}{Query tool} & \DIFaddFL{Blast }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{camachoBLASTArchitectureApplications2009,bokulichOptimizingTaxonomicClassification2018} }\hskip0pt%DIFAUXCMD
}\\
                                           &  & \rowcolor{lightgray} \DIFaddFL{Naive bayes classifier }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{bokulichOptimizingTaxonomicClassification2018} }\hskip0pt%DIFAUXCMD
}\\
                                           \cline{2-4}
                                           & \multirow{3}{*}{Database} & \rowcolor{lightgray} \DIFaddFL{Greengenes 13\_8 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{DeSantis2006} }\hskip0pt%DIFAUXCMD
}\\
                                           & & \DIFaddFL{SILVA 138 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Quast2012} }\hskip0pt%DIFAUXCMD
}\\
                                           & & \DIFaddFL{NCBI RefSeq (Oct 2021) }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Sayers2009} }\hskip0pt%DIFAUXCMD
}\\
      \hline
      \multirow{6}{*}{OTU processing} & \multirow{3}{*}{Filter(off)} & \DIFaddFL{Prevalence threshold      }& \DIFaddFL{2 / n\_samples }\\
                                      & & \DIFaddFL{Abundance threshold       }& \DIFaddFL{0.001          }\\
                                      & & \DIFaddFL{Observation sum threshold }& \DIFaddFL{10             }\\ \cline{2-4}
                                      & \multirow{3}{*}{Filter(on)}  & \rowcolor{lightgray} \DIFaddFL{Prevalence threshold      }& \DIFaddFL{0.05           }\\
                                      & & \rowcolor{lightgray} \DIFaddFL{Abundance threshold       }& \DIFaddFL{0.01           }\\
                                      & & \rowcolor{lightgray} \DIFaddFL{Observation sum threshold }& \DIFaddFL{100            }\\
      \hline
      \multirow{12}{*}{Network Inference} & \multirow{2}{*}{Bootstrapping}& \DIFaddFL{fastspar\_bootstraps v1.0 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Watts2018} }\hskip0pt%DIFAUXCMD
}\\
                                          & & \DIFaddFL{fastspar\_pvalues v1.0 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Watts2018} }\hskip0pt%DIFAUXCMD
}\\
                                          \cline{2-4}
                                          & \multirow{6}{*}{Direct association} & \acs{spieceasi} \DIFaddFL{v1.1.2 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Kurtz2015} }\hskip0pt%DIFAUXCMD
}\\
                                          & & \DIFaddFL{FlashWeave.jl v0.18.1 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{tackmannRapidInferenceDirect2019} }\hskip0pt%DIFAUXCMD
}\\
                                          & & \acs{cozine} \DIFaddFL{v1.0 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{haCompositionalZeroinflatedNetwork2020a} }\hskip0pt%DIFAUXCMD
}\\
                                          & & \acs{harmonies} \DIFaddFL{v1.0 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{jiangHARMONIESHybridApproach2020} }\hskip0pt%DIFAUXCMD
}\\
                                          & & \acs{spring} \DIFaddFL{v1.0.4 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{yoonMicrobialNetworksSPRING2019} }\hskip0pt%DIFAUXCMD
}\\
                                          & & \acs{mldm} \DIFaddFL{v1.1 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Yang2017} }\hskip0pt%DIFAUXCMD
}\\
                                          \cline{2-4}
                                          & \multirow{2}{*}{Correlation-based} & \acs{sparcc} \DIFaddFL{(FastSpar v1.0) }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{Watts2018} }\hskip0pt%DIFAUXCMD
}\\
                                          & & \DIFaddFL{Pearson }& \DIFaddFL{- }\\
                                          & & \DIFaddFL{Spearman }& \DIFaddFL{- }\\
                                          & & \DIFaddFL{propr v2.1.2 }& \DIFaddFL{\mbox{%DIFAUXCMD
\cite{quinnProprRpackageIdentifying2017} }\hskip0pt%DIFAUXCMD
}\\ \cline{2-4}
                                          & \multirow{2}{*}{Consensus algorithm} & \rowcolor{lightgray} \DIFaddFL{scaled-sum }& \DIFaddFL{0.333 }\\
                                          & & \DIFaddFL{simple voting }& \DIFaddFL{1.000 }\\
      \hline
    \end{tabular}
  \end{table}

  \thispagestyle{empty}
  \begin{landscape}
\begin{table}[H]
  \centering
  \small
  \begin{tabular}{|l|p{1cm}|p{1cm}|p{1cm}|p{2cm}|p{1.5cm}|p{1.5cm}|p{1.5cm}|p{1.5cm}|p{1.5cm}|p{1.5cm}|p{1.5cm}|}
\hline
\DIFaddFL{DC }& \DIFaddFL{CC            }& \DIFaddFL{TA                }& \DIFaddFL{OP          }& \DIFaddFL{NI         }& \DIFaddFL{Average shortest path length }& \DIFaddFL{Average clustering }& \DIFaddFL{No. of connected components }& \DIFaddFL{Modularity }& \DIFaddFL{Node connectivity }& \DIFaddFL{Degree assortativity coefficient }\\ \hline
\DIFaddFL{DB }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{2.066                        }& \DIFaddFL{0.265              }& \DIFaddFL{27.088                      }& \DIFaddFL{0.387      }& \DIFaddFL{1.868             }& \DIFaddFL{0.070                            }\\
\DIFaddFL{OR }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{1.904                        }& \DIFaddFL{0.273              }& \DIFaddFL{24.868                      }& \DIFaddFL{-17.060    }& \DIFaddFL{1.877             }& \DIFaddFL{0.099                            }\\
\DIFaddFL{D2 }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{2.263                        }& \DIFaddFL{0.267              }& \DIFaddFL{28.228                      }& \DIFaddFL{0.326      }& \DIFaddFL{1.789             }& \DIFaddFL{0.078                            }\\
\DIFaddFL{DN }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{2.065                        }& \DIFaddFL{0.272              }& \DIFaddFL{25.132                      }& \DIFaddFL{0.302      }& \DIFaddFL{1.754             }& \DIFaddFL{0.093                            }\\
\DIFaddFL{CR }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{1.937                        }& \DIFaddFL{0.276              }& \DIFaddFL{24.289                      }& \DIFaddFL{-1.865     }& \DIFaddFL{2.079             }& \DIFaddFL{0.097                            }\\
\DIFaddFL{-  }& \DIFaddFL{remove bimera }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{2.070                        }& \DIFaddFL{0.272              }& \DIFaddFL{25.968                      }& \DIFaddFL{-7.439     }& \DIFaddFL{1.863             }& \DIFaddFL{0.086                            }\\
\DIFaddFL{-  }& \DIFaddFL{uchime        }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{2.066                        }& \DIFaddFL{0.269              }& \DIFaddFL{25.874                      }& \DIFaddFL{0.275      }& \DIFaddFL{1.884             }& \DIFaddFL{0.089                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{NCBI              }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{2.218                        }& \DIFaddFL{0.262              }& \DIFaddFL{22.784                      }& \DIFaddFL{-3.683     }& \DIFaddFL{1.495             }& \DIFaddFL{0.100                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{SILVA             }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{1.872                        }& \DIFaddFL{0.275              }& \DIFaddFL{30.716                      }& \DIFaddFL{0.415      }& \DIFaddFL{2.674             }& \DIFaddFL{0.074                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{GG                }& \DIFaddFL{-           }& \DIFaddFL{-          }& \DIFaddFL{2.086                        }& \DIFaddFL{0.274              }& \DIFaddFL{24.263                      }& \DIFaddFL{-7.478     }& \DIFaddFL{1.453             }& \DIFaddFL{0.088                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{On          }& \DIFaddFL{-          }& \DIFaddFL{1.975                        }& \DIFaddFL{0.252              }& \DIFaddFL{18.493                      }& \DIFaddFL{-7.194     }& \DIFaddFL{1.880             }& \DIFaddFL{0.073                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{Off         }& \DIFaddFL{-          }& \DIFaddFL{2.193                        }& \DIFaddFL{0.291              }& \DIFaddFL{34.174                      }& \DIFaddFL{0.431      }& \DIFaddFL{1.867             }& \DIFaddFL{0.104                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{propr      }& \DIFaddFL{1.650                        }& \DIFaddFL{0.528              }& \DIFaddFL{1.000                       }& \DIFaddFL{0.000      }& \DIFaddFL{10.583            }& \DIFaddFL{0.036                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{harmonies  }&                              & \DIFaddFL{0.108              }& \DIFaddFL{82.633                      }& \DIFaddFL{0.757      }& \DIFaddFL{0.000             }& \DIFaddFL{0.340                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{flashweave }& \DIFaddFL{3.878                        }& \DIFaddFL{0.079              }& \DIFaddFL{1.800                       }& \DIFaddFL{0.458      }& \DIFaddFL{0.383             }& \DIFaddFL{-0.032                           }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{cozine     }&                              & \DIFaddFL{0.090              }& \DIFaddFL{82.167                      }& \DIFaddFL{0.694      }& \DIFaddFL{0.000             }& \DIFaddFL{-0.072                           }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{spring     }&                              & \DIFaddFL{0.092              }& \DIFaddFL{11.767                      }& \DIFaddFL{0.566      }& \DIFaddFL{0.000             }& \DIFaddFL{-0.035                           }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{sparcc     }& \DIFaddFL{1.747                        }& \DIFaddFL{0.525              }& \DIFaddFL{1.033                       }& \DIFaddFL{0.009      }& \DIFaddFL{5.500             }& \DIFaddFL{-0.011                           }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{spieceasi  }&                              & \DIFaddFL{0.076              }& \DIFaddFL{39.383                      }& \DIFaddFL{0.717      }& \DIFaddFL{0.000             }& \DIFaddFL{-0.031                           }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{pearson    }&                              & \DIFaddFL{0.457              }& \DIFaddFL{23.883                      }& \DIFaddFL{0.551      }& \DIFaddFL{0.000             }& \DIFaddFL{0.478                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{spearman   }& \DIFaddFL{1.875                        }& \DIFaddFL{0.551              }& \DIFaddFL{1.450                       }& \DIFaddFL{0.044      }& \DIFaddFL{1.117             }& \DIFaddFL{0.170                            }\\
\DIFaddFL{-  }& \DIFaddFL{-             }& \DIFaddFL{-                 }& \DIFaddFL{-           }& \DIFaddFL{mldm       }& \DIFaddFL{2.760                        }& \DIFaddFL{0.130              }& \DIFaddFL{2.267                       }& \DIFaddFL{-75.648    }& \DIFaddFL{0.433             }& \DIFaddFL{-0.028                           }\\ \hline
\end{tabular}
\end{table}
\end{landscape}
\DIFaddend 



%--------------------------------------------------------%
%   END DOCUMENT
%--------------------------------------------------------%

\end{document}
