%!TEX root = ../main.tex

\begin{abstract}
  {
    \noindent
    Microbes tend to organize into communities consisting of hundreds of species involved in complex interactions with each other.
    16S ribosomal RNA (16S rRNA) amplicon profiling provides snapshots that reveal the phylogenies and abundance profiles of these microbial communities.
    These snapshots, when collected from multiple samples, have the potential to reveal which microbes co-occur, providing a glimpse into the network of associations in these communities.
    The inference of networks from 16S data is prone to statistical artifacts.
    There are many tools for performing each step of the 16S analysis workflow, but the extent to which these steps affect the final network is still unclear.
    In this study, we perform a meticulous analysis of each step of a pipeline that can convert 16S sequencing data into a network of microbial associations.
    Through this process, we map how different choices of algorithms and parameters affect the co-occurrence network, and estimate steps which contribute most significantly to the variance. We further determine the tools and parameters that generate the most accurate and robust co-occurrence networks, based on comparison with mock and synthetic datasets.
    Ultimately, we develop a standardized pipeline that follows these default tools and parameters, but that can also help explore the outcome of any other combination of choices.
    We envisage that this pipeline for processing 16S sequencing data into networks of microbial co-occurrences could be used for integrating multiple data-sets, and for generating comparative analyses and consensus networks that can help understand and control microbial community assembly in different biomes.
    The pipeline is available as a Python package at \href{https://github.com/segrelab/MiCoNE}{https://github.com/segrelab/MiCoNE}
  }
\end{abstract}

% Insert keywords here
\keywords{Microbiome, 16S rRNA, Pipeline, Interaction, Denoising, Taxonomy, Network Inference, Correlations, Qiime, Co-occurrence, Networks}
