%!TEX root = ../main.tex

\begin{abstract}
  {
    \noindent
    Microbes tend to organize into communities consisting of hundreds of species involved in complex interactions with each other.
    16S ribosomal RNA (16S rRNA) gene profiling provides snapshots that reveal the phylogenies and abundance distributions of these microbial communities.
    These snapshots, when collected from multiple samples, have the potential to reveal which microbes co-occur, providing a glimpse into the network of inter-dependencies underlying these communities.
    The inference of networks from 16S data is prone to statistical artifacts, but the extent to which the different steps of the workflow affect the resultant network is still unclear.
    In this study, we perform a meticulous analysis of each step of a pipeline that processes 16S sequencing data into a network of microbial associations.
    Through this process, we determine the tools and parameters that generate the most accurate and robust co-occurrence.
    Ultimately, we develop a standardized pipeline that follows these default tools and parameters, but that can also help explore the outcome of any other combination of choices.
    We envisage that this standard pipeline for processing 16S sequencing data into networks of microbial co-occurrences could be used for integrating multiple data-sets, and for generating comparative analyses and consensus networks useful for detecting disease-related patterns. \todo[size=\footnotesize]{NOTE: Abstract constrained to 200 words} \\
    The pipeline is available as a Python package and a docker container at \href{https://github.com/dileep-kishore/mindpipe}{https://github.com/dileep-kishore/mindpipe}
  }
\end{abstract}

% Insert keywords here
\keywords{Microbiome, 16S rRNA, Pipeline, Interaction, Network, Correlations, Qiime, Co-occurrence}
