%!TEX root = ../main.tex

\section*{Discussion}

  \subsection*{Why \ac{micone}?}

  Co-occurrence associations in microbial communities help identify important interdependencies that may drive or reflect microbial community structure and organization.
  A myriad of tools and methods have been developed for different parts of the workflow for inference of co-occurrence networks from 16S rRNA data.
  Our analysis shows that networks generated using different combinations of tools and approaches can be significantly different from each other, highlighting the need for a clear assessment of the source of variability and for tools that provide the most robust and accurate results.
  Our newly developed integrated software for the inference of co-occurrence networks from 16S rRNA data, \ac{micone}, is a freely customizable and user-friendly pipeline that allows users to compare networks generated by multiple possible combinations of tools and parameters.
  Importantly, in addition to revisiting the test cases presented in this work, users will be able to explore the effect of various tool combinations on their own datasets of interest.
  The \ac{micone} pipeline is built in a modular fashion; its plug-and-play architecture enables users to add new tools and steps, either using existing packages that were not examined in the present work or those developed in the future.
  The \ac{micone} Python package provides functions and methods to perform detailed analysis of the count matrices and the co-occurrence networks.
  The inferred networks are exported to a custom JSON format (see Supplementary) by default, but can also be exported to Cytoscape~\cite{shannonCytoscapeSoftwareEnvironment2003}, GML~\cite{himsoltGMLPortableGraph2010}, and many other popular formats via the \ac{micone} Python package.

  While several tools/workflows such as \ac{qiime2}~\cite{bolyenReproducibleInteractiveScalable2019} and NetCoMi~\cite{peschelNetCoMiNetworkConstruction2020} can be used to generate co-occurrence networks from 16S sequencing data, no single tool exist that integrates the complete process of inferring microbial interaction networks from 16S sequencing reads.
  \ac{micone} is unique as it offers this functionality packaged in a workflow that can be run locally, on the compute cluster or the cloud.

  \subsection*{The default pipeline and recommended tools}

  Through \ac{micone}, in addition to transparently revealing the dependence of co-occurrence networks on tool and parameter choices, we take advantage of our spectrum of computational options and the availability of mock and synthetic datasets, to suggest a default standard setting.
  Additionally, we develop a consensus approach, that can reliably generate networks that are robust across multiple tool/parameter choices.
  An important caveat related to these results is the fact that due to the lack of a universal standard for microbial interaction data, our conclusions are based on the specific datasets used in our analysis.
  While our datasets comprise several mock and synthetic datasets that cover a fairly diverse range of abundance distributions and network topologies, datasets that have drastically different distributions may require a re-assessment of the best settings through our pipeline.

  The extending discussion behind our choices and selection of default settings for the DC, TA and OP steps of the pipeline can be found in the supplementary discussion section (see Supplementary).
  The networks generated by different network inference methods show considerable differences in edge-density and connectivity.
  One reason for this is the underlying assumptions regarding sparsity, distribution and compositionality, that the algorithms make.
  The consensus network created by merging the networks inferred from the different network inference methods enables the creation of a network whose links have evidence based on multiple inference algorithms.
  We have developed two consensus algorithms to support this effort, the simple voting method and the scaled-sum method.
  Ultimately, we find that the scaled-sum method along with p-value merging (applied to the correlation-based methods) performs the best on synthetic datasets, therefore this is the default for the \ac{ni} step of the pipeline.
  The consensus algorithm not only leads to the creation of a network that has a higher number of true positives, it also decreases the variability in the networks due to choosing a different tool or parameter in any of the previous steps of the pipeline (Figure~\ref{fig:figure7}B).
  This provides the means to obtain a short list of associations, each of which, have a high probability of being present in the real interaction network
  This small set of associations could be further tested and validated experimentally to verify their plausibility.

  \subsection*{Future directions}

  Future work building upon our current results could enhance the network inference process in multiple ways.
  The current analyses make use of one fecal microbiome transplant dataset with healthy and autistic samples, three mock community datasets and several datasets generated by two synthetic interaction methods.
  Incorporating datasets from a broad spectrum of biomes with varying microbial distributions into \ac{micone} will likely increase the robustness and generalizability of the results from these analyses.

  The network analyses in this study are primarily at the Genus level, wherein the lowest resolution of a node is a Genus and if an entity cannot be resolved to the Genus level and next lowest taxonomic level is used (e.g. Family).
  As a consequence, two entities belonging to the same lineage where one entity is resolved to the Genus level and another is resolved to the Family level are treated as two different nodes in the network.
  Therefore, the development of a metric of overlap to compare nodes with shared lineages within and across networks could provide more biologically and phylogentically relevant comparisons.

  Futher benchmarking of co-occurrence networks could be pursued through the utilization of literature-based interactions~\cite{lima-mendezDeterminantsCommunityStructure2015a} or biological benchmark interaction data~\cite{sungGlobalMetabolicInteraction2017a}.
  Furthermore, \ac{micone} could be extended to enable the processing of metagenomics sequencing data, facilitating the analysis of a much larger and diverse range of datasets and domains of life.

  Although in the current analysis, we only use default parameter values recommended by the tool creators, the \ac{micone} pipeline could be used in the future to explore any combinations of parameters and to optimize these values for improved network inference.
  Overall, it is important to keep in mind that there likely is no ``best method'' for the various steps of 16S data analysis, especially for the network inference step.
  Therefore, \ac{micone} is intended to help researchers to identify the methods and algorithms that are most suitable for their datasets in an easy-to-use and reproducible manner.

  We envision that \ac{micone}, and its underlying tools and databases, will be increasingly useful towards building large comparative analyses across studies.
  It allows for rapid, configurable and reproducible inference of microbial networks and further for the formulation of hypotheses about the role of these interactions on community composition and stability.
  These comparative analyses will require coupled network analysis and visualization tools (such as \ac{mind}~\cite{huResourceComparisonIntegration2022}), and will require a more systematic access to datasets, shared in accordance to FAIR standards~\cite{pachecoFAIRRepresentationsMicrobial2022}.
