%!TEX root = ../main.tex

\section*{Discussion}

% General statements
Co-occurrence associations in microbial communities help identify important interactions that drive microbial community structure and organization.
Our analysis shows that networks generated using different combinations of tools and approaches can look significantly different from each other, highlighting the importance of a clear assessment of the source of variability and of tools that provide the most robust and accurate results.
Our newly developed integrated software for the inference of co-occurrence networks from 16S rRNA data, \ac{micone}, constitutes a freely customizable and user friendly pipeline that allows users to easily test combinations of tools and to compare networks generated by multiple possible choices (refer to the Methods section).
Importantly, in addition to revisiting the test cases presented in this work, users will be able to explore the effect of various tool combinations on their own datasets of interest.
The \ac{micone} pipeline is built in a modular fashion.
Its plug-and-play architecture will make it possible for users to add new tools and steps, either from existing packages, or from those that were not examined in the present work and those that will be developed in the future.

The main outcome of this work is thus two-fold: on one hand we transparently reveal the dependence of co-occurrence networks on tool and parameter choices, making it possible to more rigorously assess and compare existing networks.
On the other hand, we take advantage of our spectrum of computational options and the availability of mock and synthetic datasets, to suggest a default standard setting, and a consensus approach, likely to yield networks that are robust across multiple tool/parameter choices.

An important caveat related to this last point is the fact that our conclusions are based on the specific datasets used in our analysis.
While our datasets cover a relatively broad spectrum of biomes and sequencing pipelines, datasets that have drastically different distributions may require a re-assessment of the best settings through our pipeline.

It is worth pointing out some additional more specific conclusions stemming from the individual steps of our analysis.

The different denoising/clustering methods differ mostly in their identification of sequences that are in low abundances.
Hence, they do not have much of an impact on the inferred co-occurrence networks when the sequences of low abundance are removed.
However, comparison of inferred and expected reference sequences and their abundances in mock community datasets has allowed us to identify \ac{dada2} as the method which best recapitulates the expected sequence composition.
There are newer recently published methods like dbOTU3~\cite{Olesen2017} as well as older popular methods like mothur~\cite{Schloss2009} that have not been included in the study, and are possible avenues for future investigation.

The choice of taxonomy database was found to be the most important factor in the inference of a microbial co-occurrence network, contributing $\sim20\%$ of the total variance.
The frequent changes in the taxonomy nomenclature coupled with the frequency of updates to the various 16S reference databases create inherent differences \cite{Balvociute2017} in taxonomy hierarchies in these databases.
Our analysis did not reveal one particular reference database to perform better in all scenarios. We suggest that that choice of the database should be made based on possible reported or inferred biases in the representation of given biomes in a specific databases \cite{Balvociute2017}.
The default reference database in the pipeline is the \ac{ncbi} 16S RefSeq database as it is more frequently updated and is most compatible with the blast+ query tool.
We also enable users to use custom databases \cite{Griffen2011,Ritari2015} with the blast and naive bayes classifiers that are incorporated into the pipeline (from \ac{qiime2}).

Filtering out taxa that are present in low abundances in all samples did not increase (in most datasets tested) the proportion of taxa in common between taxonomy tables generated using different reference databases.
However, we do observe that the reduction in the number of taxa leads to better agreement in the networks inferred through different methods.
Moreover, filtering is necessary in order to increase the power in tests of significance when the number of taxa is much greater than the number of samples.

The networks generated by different network inference methods show considerable differences in edge-density and connectivity.
One reason for this is the underlying assumptions regarding sparsity, distribution and compositionality that the algorithms make.
The consensus network created by merging the networks inferred using the different network inference methods enables the creation of a network whose links have evidence based on multiple inference algorithms.

% \hl{Other factors that play an important role in network inference} \\
% Differences in environments of the samples could lead to the inference of spurious interactions [ref].

Exploring the effects of these combinations of methods on the resultant networks is difficult and inconvenient since different tools differ in their input and output formats and require inter-converting between the various formats.
The pipeline facilitates this comparative exploration by providing a variety of modules for inter-conversion between various formats, and by allowing easy incorporation of new tools as modules.

We envision that \ac{micone}, and the underlying tools and databases that help process amplicon sequencing data into co-occurrence networks, will be increasingly useful towards building large comparative analyses across studies. By having a unified transparent tool to compute networks, it will be possible to reprocess available 16S datasets to obtain networks that are directly comparable to each other.
Furthermore, even in the analysis of published networks across studies and processing methods, \ac{micone} could help understand underlying biases of each network, which could in turn be taken into account upon making cross-study comparisons.  

% TODO: Uses in MIND (?)
