%!TEX root = ../main.tex

\section*{Discussion}

  \subsection*{Why \ac{micone}?}

  Co-occurrence associations in microbial communities help identify important interactions that drive microbial community structure and organization.
  Our analysis shows that networks generated using different combinations of tools and approaches can look significantly different from each other, highlighting the importance of a clear assessment of the source of variability and of tools that provide the most robust and accurate results.
  Our newly developed integrated software for the inference of co-occurrence networks from 16S rRNA data, \ac{micone}, constitutes a freely customizable and user friendly pipeline that allows users to easily test combinations of tools and to compare networks generated by multiple possible choices (see Methods).
  Importantly, in addition to revisiting the test cases presented in this work, users will be able to explore the effect of various tool combinations on their own datasets of interest.
  The \ac{micone} pipeline is built in a modular fashion; its plug-and-play architecture will make it possible for users to add new tools and steps, either from existing packages, or from packages that were not examined in the present work, as well as those developed in the future.
  The easy-to-use nature, the configurability and modular nature of the \ac{micone} package enables users to start and end the pipeline at any point in the worflow, enabling users to run parts of the pipeline in isolation.
  The \ac{micone} Python package provides functions and methods to perform detailed analysis of the count matrices and the inferred networks.
  % TODO: Ref and supplementary
  Additionally, the inferred networks by default are created using a custom JSON format (see Supplementary), but are interchangeable to Cytoscape[ref], GML[ref], and many other popular formats via the \ac{micone} Python package.

  % TODO: Refs
  There have been several tools/workflows such as \ac{qiime2}~\cite{bolyenReproducibleInteractiveScalable2019} and NetCoMi[ref] that have been developed for analysis of 16S sequencing data and to infer networks from these datasets.
  However, there has been no single tool that completely integrates the process of inferring microbial interaction networks from 16S sequencing reads.
  \ac{micone} is unique as it offers this functionality packaged in a simple workflow that can be run locally, on the compute cluster or the cloud.
  For an end user who is interested in identifying the microbial interactions present in a specific sequencing dataset, \ac{micone} offers a fast and reproducible way to compare and test novel hypotheses related to interactions.

  \subsection*{The default pipeline and suggested tools}

  The main outcome of this work is two-fold: on one hand we transparently reveal the dependence of co-occurrence networks on tool and parameter choices, and built a tool for this purpose, enabling to more rigorously assess and compare existing networks.
  On the other hand, we take advantage of our spectrum of computational options and the availability of mock and synthetic datasets, to suggest a default standard setting, and a consensus approach, likely to yield networks that are robust across multiple tool/parameter choices.

  An important caveat related to this last point is the fact that due to the lack of a universal standard for microbial interaction data, our conclusions are based on the specific datasets used in our analysis.
  While our datasets cover several mock and synthetic datasets that cover a fairly diverse range of abundance distributions and network topologies, datasets that have drastically different distributions may require a re-assessment of the best settings through our pipeline.

  It is worth pointing out some additional more specific conclusions stemming from the individual steps of our analysis.
  The different denoising/clustering methods differ mostly in their identification of sequences that are in low abundances.
  Hence, they do not have much of an impact on the inferred co-occurrence networks when the sequences of low abundance are removed.
  However, comparison of inferred and expected reference sequences and their abundances in mock community datasets has allowed us to identify \ac{dada2} as the method which best recapitulates the expected sequence composition.
  For the chimera checking module, we suggest using the remove bimera method since it was developed specifically for \ac{dada2} and performs equally as well as uchime.
  For the current work we have decided to focus on the tools most widely used at the time of the analysis.
  Some tools there were not as widely used (e.g. dbOTU3~\cite{Olesen2017}) as well as older popular methods like mothur~\cite{Schloss2009} have not been included in the study, but could be added into the pipelines in future updated analyses.

  The choice of taxonomy database was found to be the most important factor in the inference of a microbial co-occurrence network, contributing $\sim65.4\%$ of the total variance.
  The frequent changes in the taxonomy nomenclature coupled with the frequency of updates to the various 16S reference databases create inherent differences \cite{Balvociute2017} in taxonomy hierarchies in these databases.
  Our analysis revealed that no particular reference database performs better than the others across all scenarios. We suggest that that choice of the database should be made based on possible reported or inferred biases in the representation of given biomes in a specific databases \cite{Balvociute2017}.
  The default reference database in the pipeline is the \ac{gg} reference database along with the NaiveBayes classifier as the query tool.
  The reason for our choice stem from the popularity of the \ac{gg} database in taxonomic studies, which would enable easy comparision across datasets.
  We also enable users to use custom databases \cite{Ritari2015} with the blast+ and naive bayes classifiers that are incorporated into the pipeline (from \ac{qiime2}).

  The \ac{op} step of the pipeline the step that second in its contribution to total network variance.
  This can be attributed to the large number of nodes that are added to the final networks when filtering is turned off.
  % TODO: Refs
  Additionally, the large number of nodes also greatly decrease the accuracy of the network inference algorithms [ref] as well as increase the computational complexity.
  % TODO: Supplementary work - Figure S5
  We observe that filtering out taxa that are present in low abundances in all samples did not necessarily increase the proportion of taxa in common between taxonomy tables generated using different reference databases.
  % TODO: Supplementary work - Figure S2
  However, we do observe that the reduction in the number of taxa leads to better agreement in the networks inferred through different methods.
  Moreover, filtering is necessary in order to increase the power in tests of significance when the number of taxa is much greater than the number of samples.

  The networks generated by different network inference methods show considerable differences in edge-density and connectivity.
  One reason for this is the underlying assumptions regarding sparsity, distribution and compositionality that the algorithms make.
  The consensus network created by merging the networks inferred from the different network inference methods enables the creation of a network whose links have evidence based on multiple inference algorithms.
  We have developed two consensus algorithms to support this effort, the simple voting method and the scaled-sum method.
  Ultimately, we find that the scaled-sum method along with pvalue merging (applied to the correlation-based methods) performs the best on synthetic datasets, therefore this is the default for the \ac{ni} step of the pipeline.

  Exploring the effects of these combinations of methods on the resultant networks is difficult and inconvenient since different tools differ in their input and output formats and require inter-converting between the various formats.
  The pipeline facilitates this comparative exploration by providing a variety of modules for inter-conversion between various formats, and by allowing easy incorporation of new tools as modules.
  Since there exists no ``best method'' for the various steps of 16S data analysis, especially for microbial network inference, \ac{micone} is intended to help researchers to identify the methods and algorithms that are most suitable for their datasets in a easy-to-use and reproducible manner.
  We envision that \ac{micone}, and the underlying tools and databases that help process amplicon sequencing data into co-occurrence networks, will be increasingly useful towards building large comparative analyses across studies.
  By having a unified transparent tool to compute networks, it will be possible to reprocess available 16S datasets to obtain networks that are directly comparable to each other.
  Furthermore, even in the analysis of published networks across studies and processing methods, \ac{micone} could help understand underlying biases of each network, which could in turn be taken into account upon making cross-study comparisons.

  \subsection*{Future directions}

  The current analyses include one fecal microbiome transplant dataset with healthy and autistic samples, three mock community datasets and several datasets generated by two synthetic interaction methods.
  The analyses presented in this study are specific to the datasets utilized, and due to the lack of a gold standard reference dataset for microbial interaction data, it is not possible to generalize these results to datasets with widely different microbiome profiles.
  Incorporating datasets from a broad spectrum of biomes with varying microbial distributions into \ac{micone} will likely increase the robustness and generalizability of the results from these analyses.
  Furthermore, expanding the scope of the \ac{micone} pipeline to also include metagenomics sequencing data will enable an even broader access to a diverse range of datasets as well as domains of life that can be incorporated into the analysis.

  Although the pipeline currently supports changing of parameter values for the most commonly used parameters for several tools, we only use the those default parameter values that are recommended by the tool creators.
  The exploration of the effects of parameter values on the inferred association networks might help optimize these values for better network inference, but these are not part of this current study.
  Additionally, despite incorporating a wide range of tools and algorithsm into the \ac{micone} pipeline, there are several other tools that were omitted due time and computational constraints.
  However, the modular nature of the \ac{micone} pipeline enables easy addition of new tools and methods at any step of the workflow.

  % TODO: Should we move the last paragraph of the previous section here?

  % TODO: Health and disease
  In summary, we believe that \ac{micone} is a useful tool for the exploration of microbial networks generated from 16S data, allowing for rapid, configurable and reproducible inference of microbial networks and further formulation of hypotheses about the role of these interactions on community composition and stability.
