%!TEX root = ../main.tex

\section*{Introduction}

Microbial communities are ubiquitous and play an important role in marine and terrestrial environments, urban ecosystems, and human health \cite{lima-mendezDeterminantsCommunityStructure2015a,Thompson2017,royo-llonchCompendium530Metagenomeassembled2021,tedersooFungalBiogeographyGlobal2014,dankoGlobalMetagenomicMap2021,mclellanMicrobiomeUrbanWaters2015,HumanMicrobiomeProjectConsortium2012}.
  These microbial communities, or microbiomes, often comprise several hundreds of different microbial strains interacting with each other and their environment, often through complex metabolic and signaling relationships~\cite{zelezniakMetabolicDependenciesDrive2015,Ghoul2016,coyteUnderstandingCompetitionCooperation2019,DSouza2018}.
  Understanding how these interconnections shape community structure and function is a fundamental challenge in microbial ecology, and has applications in the study of microbial ecosystems across different biomes.
  With the advancement in DNA sequencing technologies~\cite{huNextgenerationSequencingTechnologies2021,buermansNextGenerationSequencing2014,Narihiro2017},  more information can be extracted from these microbial community samples than ever before.
  In particular, high-throughput sequencing, including metagenomic sequencing and sequencing of 16S rRNA gene amplicons (hereafter referred to as 16S data) of microbial communities, can help detect, identify and quantify a large portion of the constitutive microorganisms of a microbiome \cite{ju16SRRNAGene2015,Jovel2016,quinceShotgunMetagenomicsSampling2017,sedlarBioinformaticsStrategiesTaxonomy2017}.
  These advances have led to large-scale data collection efforts involving terrestrial~\cite{Thompson2017,gilbertMeetingReportTerabase2010,tedersooFungalBiogeographyGlobal2014}, marine~\cite{lima-mendezDeterminantsCommunityStructure2015a,royo-llonchCompendium530Metagenomeassembled2021} and human-associated microbiota~\cite{HumanMicrobiomeProjectConsortium2012,proctorIntegrativeHumanMicrobiome2019,Lloyd-Price2016}.

 This wealth of information has the potential to help us understand how communities assemble and operate.
 In particular, a powerful tool for translating microbiome composition data into knowledge is the construction of association (co-occurrence or correlation) networks, in which  microbial taxa are represented by nodes, and frequent co-occurrences (or negative correlations) across datasets are encoded as edges between nodes.
 While the relationship between directly measured interactions~\cite{lubbeExometabolomicAnalysisCrossFeeding2017,Jian2020,Hsu2019} and statistically inferred co-occurrence is still poorly understood \cite{Zuniga2017,Rottjers2018}, a significant amount of effort has gone into estimating correlations from large microbiome sequence datasets~\cite{faustMicrobialCooccurrenceRelationships2012,leeCrosskingdomCooccurrenceNetworks2022,faustMicrobialInteractionsNetworks2012a,maEarthMicrobialCooccurrence2020a}.

 The importance of these networks is two-fold: first, they can serve as maps that help identify hubs of keystone species \cite{Menon2018,Rottjers2018}, and the community response to environmental perturbations or underlying host conditions \cite{Gilbert2016}; second, they can serve as a bridge towards building mechanistic models of microbial communities, greatly enhancing our capacity to understand and control them.
 For example, multiple studies have shown the importance of specific microbial associations in the healthy microbiome \cite{Lloyd-Price2016,Wu2016,HumanMicrobiomeProjectConsortium2012} and their role in dysbiosis \cite{Wang2017,Gilbert2016,Belizario2015}.
 In the context of terrestrial biogeochemistry, co-occurrence networks were shown to help understand microbiome assembly \cite{fiererEmbracingUnknownDisentangling2017}, and  the response of microbial communities to environmental perturbations \cite{Jiao2019}.

One of the most frequently used avenues for inferring co-occurrence networks is the parsing and analysis of 16S sequencing data \cite{Rottjers2018,Friedman2012}.
Numerous software tools and pipelines have been developed to analyze 16S sequencing data, with a strong emphasis on the known limitations of this method, including resolution, sequencing depth, compositional nature, sequencing errors, and copy number variations \cite{Bharti2019,pollockMadnessMicrobiomeAttempting2018}.
Popular methods for different phases of the analysis of 16S data include tools for: (i) quality checking and trimming the sequencing reads; (ii) denoising and clustering the trimmed reads \cite{Caporaso2010,Callahan2016,Amir2017}; (iii) assigning taxonomy to the denoised reads \cite{bokulichOptimizingTaxonomicClassification2018}; (iv) processing and transforming the taxonomy count matrices \cite{Weiss2015}; and (v) inferring the co-occurrence network \cite{Watts2018,Kurtz2015,tackmannRapidInferenceDirect2019}.
Different specific algorithms are often aggregated into popular online platforms (like MG-RAST~\cite{keeganMGRASTMetagenomicsService2016}, Qiita~\cite{gonzalezQiitaRapidWebenabled2018}) and software packages (such as \ac{qiime2}~\cite{bolyenReproducibleInteractiveScalable2019}).
The different methods and tools can lead to vastly different inferences of community compositions and co-occurrence networks \cite{Golob2017,Weiss2016}, making it difficult to reliably compare networks across different publications and studies.
 This difference is partially due to the focus of existing platforms on \ac{otu} or \ac{esv} generation and not on the effects of upstream statistical methods on the inferred co-occurrence networks.
 Furthermore, no organized framework currently exists that can systematically analyze and compare each step in the pipeline for processing amplicons into co-occurrence networks.

 In this study, we present a standardized 16S data analysis pipeline called \ac{micone} that produces robust and reproducible co-occurrence networks from 16S sequence data of microbial communities, and enable users to interactively explore how the network would change upon using different alternative tools and parameters at each step.
 Our pipeline is coupled to an online integrative tool for the organization, visualization, and analysis of inter-microbial networks called \ac{mind}~\cite{huResourceComparisonIntegration2022}, which is available at \href{http://microbialnet.org/}{http://microbialnet.org/}.
Through a systematic comparative analysis, we determine which steps of the \ac{micone} pipeline have the largest influence on the final network, and which choice seems to have the optimal agreement with the tested mock and synthetic datasets.
These steps together with our default settings ensure better reproducibility and easier comparison of co-occurrence networks across datasets.
We expect that our tool will also be useful for benchmarking future alternative methods, and for ensuring a transparent evaluation of the possible biases introduced by the use of specific tools.
